From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Tue, 31 Oct 2017 17:12:13 +0900
Subject: [PATCH] meson: do not include man/meson.build if xsltproc not found

Fixes #7232.
---
 man/meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/man/meson.build b/man/meson.build
index 7d28e6ba1ad5..e376bb8365b8 100644
--- a/man/meson.build
+++ b/man/meson.build
@@ -32,7 +32,7 @@ custom_entities_ent = configure_file(
 man_pages = []
 html_pages = []
 source_xml_files = []
-foreach tuple : manpages
+foreach tuple : want_man or want_html ? manpages : []
         stem = tuple[0]
         section = tuple[1]
         aliases = tuple[2]
@@ -115,8 +115,8 @@ systemd_index_xml = custom_target(
         output : 'systemd.index.xml',
         command : [make_man_index_py, '@OUTPUT@'] + nonindex_xml_files)
 
-foreach tuple : [['systemd.directives', '7', systemd_directives_xml],
-                 ['systemd.index',      '7', systemd_index_xml]]
+foreach tuple : want_man or want_html ? [['systemd.directives', '7', systemd_directives_xml],
+                                         ['systemd.index',      '7', systemd_index_xml]] : []
         stem = tuple[0]
         section = tuple[1]
         xml = tuple[2]
