diff -urN freetype-2.1.9-orig/builds/unix/freetype-config.in freetype-2.1.9/builds/unix/freetype-config.in
--- freetype-2.1.9-orig/builds/unix/freetype-config.in	2004-04-17 20:58:43.000000000 +0200
+++ freetype-2.1.9/builds/unix/freetype-config.in	2006-02-27 22:11:53.000000000 +0100
@@ -118,7 +118,7 @@
   if test "$enable_shared" = "yes" ; then
     eval "rpath=\"$hardcode_libdir_flag_spec\""
   fi
-  libs="-lfreetype @LIBZ@"
+  libs="${wl}-lfreetype @LIBZ@"
   if test "$libdir" != "/usr/lib" ; then
     echo -L$libdir $rpath $libs
   else
