From 72cbfd16d1b4f42b661ee97936c4e7f5f6db4fef Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Mon, 6 Jul 2009 12:26:06 +0200
Subject: [PATCH 2/2] [Configure] let environment override RANLIB and AR

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 Configure |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/Configure b/Configure
index 0fe11a5..07cbf2f 100755
--- a/Configure
+++ b/Configure
@@ -1025,7 +1025,16 @@ my $shared_cflag = $fields[$idx_shared_cflag];
 my $shared_ldflag = $fields[$idx_shared_ldflag];
 my $shared_extension = $fields[$idx_shared_extension];
 my $ranlib = $fields[$idx_ranlib];
+# Allow environment RANLIB to override ranlib...
+if($ENV{RANLIB}) {
+    $ranlib = $ENV{RANLIB};
+}
 my $arflags = $fields[$idx_arflags];
+my $ar = "ar";
+# Allow environment AR to override ar...
+if($ENV{AR}) {
+    $ar = $ENV{AR};
+}
 
 if ($fips)
 	{
@@ -1533,6 +1542,8 @@ while (<IN>)
 	s/^PROCESSOR=.*/PROCESSOR= $processor/;
 	s/^RANLIB=.*/RANLIB= $ranlib/;
 	s/^ARFLAGS=.*/ARFLAGS= $arflags/;
+	s/^AR=ar/AR= $ar/;
+	s/^ARD=ar/ARD= $ar/;
 	s/^PERL=.*/PERL= $perl/;
 	s/^KRB5_INCLUDES=.*/KRB5_INCLUDES=$withargs{"krb5-include"}/;
 	s/^LIBKRB5=.*/LIBKRB5=$withargs{"krb5-lib"}/;
@@ -1600,6 +1611,7 @@ print "RMD160_OBJ_ASM=$rmd160_obj\n";
 print "PROCESSOR     =$processor\n";
 print "RANLIB        =$ranlib\n";
 print "ARFLAGS       =$arflags\n";
+print "AR            =$ar\n";
 print "PERL          =$perl\n";
 print "KRB5_INCLUDES =",$withargs{"krb5-include"},"\n"
 	if $withargs{"krb5-include"} ne "";
-- 
1.6.3.3

