From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Fri, 25 Jan 2013 18:03:04 +0100
Subject: [PATCH] makefile: remove host path from list of header search dirs

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 Makefile      |    2 +-
 test/Makefile |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index e5fae3d..02f7b7e 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ YACC = bison -y
 
 CFLAGS ?= -g -Wall -Werror -Wshadow -O2 -pipe -fno-strict-aliasing
 
-override CFLAGS += -I. -I${INCLUDEDIR}
+override CFLAGS += -I.
 
 CHECKOBJS = y.tab.o lex.yy.o queue.o module_compiler.o parse_util.o \
 	    policy_define.o
diff --git a/test/Makefile b/test/Makefile
index 0731e89..51d0f63 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -7,7 +7,7 @@ LIBDIR=$(PREFIX)/lib
 INCLUDEDIR ?= $(PREFIX)/include
 
 CFLAGS ?= -g -Wall -W -Werror -O2 -pipe
-override CFLAGS += -I$(INCLUDEDIR)
+override CFLAGS +=
 
 LDLIBS=-lfl -lselinux $(LIBDIR)/libsepol.a -L$(LIBDIR)
 
