From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Thu, 26 Apr 2012 15:07:30 +0200
Subject: [PATCH] HACK: support/apxs: don't execute httpd

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 support/apxs.in |   24 ++++++++++++------------
 1 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/support/apxs.in b/support/apxs.in
index 135544c..cbefb3a 100644
--- a/support/apxs.in
+++ b/support/apxs.in
@@ -199,18 +199,18 @@ if ($0 =~ m:support/apxs$:) {
     ($httpd = $0) =~ s:support/apxs$::;
 }
 
-unless (-x "$httpd") {
-	error("$httpd not found or not executable");
-	exit 1;
-}
-
-unless (grep /mod_so/, `. $envvars && $httpd -l`) {
-    error("Sorry, no shared object support for Apache");
-    error("available under your platform. Make sure");
-    error("the Apache module mod_so is compiled into");
-    error("your server binary `$httpd'.");
-    exit 1;
-}
+#unless (-x "$httpd") {
+#	error("$httpd not found or not executable");
+#	exit 1;
+#}
+
+#unless (grep /mod_so/, `. $envvars && $httpd -l`) {
+#    error("Sorry, no shared object support for Apache");
+#    error("available under your platform. Make sure");
+#    error("the Apache module mod_so is compiled into");
+#    error("your server binary `$httpd'.");
+#    exit 1;
+#}
 
 sub get_config_vars{
     my ($file, $rh_config) = @_;
