diff -urN clock-1.5.1/Makefile clock-1.5.1-ptx/Makefile
--- clock-1.5.1/Makefile	1998-10-06 01:39:57.000000000 +0200
+++ clock-1.5.1-ptx/Makefile	2005-12-27 15:20:07.000000000 +0100
@@ -14,8 +14,8 @@
 ARCH = intel
 
 install = install
-cc = gcc
-ld = gcc
+CC ?= gcc
+LD ?= gcc
 
 cc_flags = -O2 -DVERSION=\"$(VERSION)\"
 ld_flags = -lc
@@ -23,10 +23,10 @@
 all: clock
 
 clock: clock.o
-	$(ld) clock-$(ARCH).o -o clock $(ld_flags)
+	$(LD) clock-$(ARCH).o -o clock $(ld_flags)
 
 clock.o: clock-$(ARCH).c
-	$(cc) $(cc_flags) -c clock-$(ARCH).c
+	$(CC) $(cc_flags) -c clock-$(ARCH).c
 
 clean:
 	@rm -f clock clock-$(ARCH).o
