From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 27 May 2016 20:20:16 +0200
Subject: [PATCH] check for libsystemd instead of libsystemd-login

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f4a0c4177fbf..4bf2c05d989c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -160,7 +160,7 @@ AC_ARG_ENABLE([systemd],
               [enable_systemd=auto])
 if test "$enable_systemd" != "no"; then
   PKG_CHECK_MODULES(SYSTEMD,
-                    [libsystemd-login],
+                    [libsystemd],
                     have_systemd=yes,
                     have_systemd=no)
   if test "$have_systemd" = "yes"; then
