diff -urN klibc-1.1.1.orig/klibc/Makefile klibc-1.1.1/klibc/Makefile
--- klibc-1.1.1.orig/klibc/Makefile	2005-09-06 22:49:34.000000000 +0200
+++ klibc-1.1.1/klibc/Makefile	2005-11-22 17:34:52.000000000 +0100
@@ -184,8 +184,6 @@
 		$(INSTALLROOT)$(INSTALLDIR)/$(CROSS)lib
 	$(INSTALL_EXEC) klibc-`cat $(SOLIB).hash`.so \
 		$(INSTALLROOT)$(INSTALLDIR)/$(CROSS)lib
-	$(INSTALL_EXEC) klibc-`cat $(SOLIB).hash`.so \
-		$(INSTALLROOT)/$(SHLIBDIR)
 
 ifneq ($(wildcard $(DIR).*.d),)
 include $(wildcard $(DIR).*.d)
diff -urN klibc-1.1.1.orig/Makefile klibc-1.1.1/Makefile
--- klibc-1.1.1.orig/Makefile	2005-09-06 22:49:33.000000000 +0200
+++ klibc-1.1.1/Makefile	2005-11-22 17:33:14.000000000 +0100
@@ -61,7 +61,7 @@
 	  mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(CROSS)include/$$d ; \
 	  for r in $(KRNLSRC)/include $(KRNLOBJ)/include $(KRNLOBJ)/include2 ; do \
 	    [ ! -d $$r/$$d ] || \
-	      cp -rfL $$r/$$d/. $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/$$d/. ; \
+	      cp -rfd $$r/$$d/. $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/$$d/. ; \
 	  done ; \
 	done
 	cd $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include && ln -sf asm-$(ARCH) asm
