--- etherwake-1.09.orig/Makefile
+++ etherwake-1.09/Makefile
@@ -0,0 +1,12 @@
+CFLAGS 	= -Wall
+INSTALL = /usr/bin/install
+CC 	= gcc
+
+all: 
+	$(CC) $(CFLAGS) -o etherwake ether-wake.c
+
+clean:
+	rm etherwake
+	
+install:
+	$(INSTALL) etherwake $(DESTDIR)/usr/sbin
--- etherwake-1.09.orig/debian/changelog
+++ etherwake-1.09/debian/changelog
@@ -0,0 +1,86 @@
+etherwake (1.09-1) unstable; urgency=low
+
+  * New Upstream
+
+ -- Alain Schroeder <alain@debian.org>  Sun, 10 Jul 2005 20:53:02 +0200
+
+etherwake (1.08-1) unstable; urgency=low
+
+  * New Upstream.
+    * Beautified Help Texts.
+    * Cleans (PF_INET,SOCK_PACKET) problems (Closes: #171051)
+  * Renamed ether-wake.8 to etherwake.8 (Closes: #185134)
+  * Revisioned the manpage (Closes: #179290)
+  * Cleaned my helptext from multiline strings, Thanks to Daniel Schepler
+    <schepler@math.berkeley.edu> (Closes: #194909)
+  * Upped Standart Version to 3.5.10.0
+
+ -- Alain Schroeder <alain@debian.org>  Fri,  6 Jun 2003 21:02:11 +0200
+
+etherwake (1.06-2) unstable; urgency=low
+
+  * Reamped description to explain what WOL is (Closes: #135420).
+  * Reamped manpage a bit.
+  * Upped Standart Version to 3.5.6.0.
+
+ -- Alain Schroeder <alain@debian.org>  Sun, 24 Feb 2002 00:45:07 +0100
+
+etherwake (1.06-1) unstable; urgency=low
+
+  * New Upstream.
+
+ -- Alain Schroeder <alain@parkautomat.net>  Fri, 15 Feb 2002 19:15:07 +0100
+
+etherwake (1.05-6) unstable; urgency=low
+
+  * Fix etherwake to use sockets correctly - removes "uses obsolete
+    (PF_INET,SOCK_PACKET)" message. (Closes: #111681)
+
+ -- Alain Schroeder <alain@debian.org>  Sat, 22 Dec 2001 21:51:06 +0100
+
+etherwake (1.05-5) unstable; urgency=low
+
+  * Improve description again. (Closes: #124604)
+
+ -- Alain Schroeder <alain@debian.org>  Sat, 22 Dec 2001 20:22:16 +0100
+
+etherwake (1.05-4) unstable; urgency=low
+
+  * Removed more typos in description. (Again thanks to Carlos Valdivia
+    Yagüe)
+
+ -- Alain Schroeder <alain@debian.org>  Thu, 10 May 2001 15:55:48 +0200
+ 
+etherwake (1.05-3) unstable; urgency=low
+
+  * Removed typo in description. (Closes: #95946)
+
+ -- Alain Schroeder <alain@debian.org>  Tue,  1 May 2001 21:14:41 +0200
+
+etherwake (1.05-2) unstable; urgency=low
+
+  * Upload with diff. It was incidentally uploaded as a native 
+    Debian package.
+  * Fixed issues with new libc. The old sys/socket.h did not include
+    bits/uio.h, which it does now. This unfortunately causes that
+    ether-wake wasn´t build anymore. I just commented specific part as
+    a quick fix for now. It seems that everything works even with the old
+    libc, closes #87014.
+
+ -- Alain Schroeder <alain@debian.org>  Fri, 23 Feb 2001 00:27:02 +0100
+
+etherwake (1.05-1) unstable; urgency=low
+
+  * New upstream (help was added).
+  * Modified manpage to reflect upstream changes.
+  * Changed binary name to original.
+
+ -- Alain Schroeder <alain@debian.org>  Tue, 20 Feb 2001 11:39:03 +0100
+
+etherwake (1.03-1) unstable; urgency=low
+
+  * Added a manpage.
+  * Added a Makefile.
+  * Initial Release.
+
+ -- Alain Schroeder <alain@debian.org>  Thu,  7 Dec 2000 10:00:24 +0100
--- etherwake-1.09.orig/debian/copyright
+++ etherwake-1.09/debian/copyright
@@ -0,0 +1,12 @@
+This package was debianized by Alain Schroeder <alain@debian.org> on
+Thu,  7 Dec 2000 10:00:24 +0100.
+
+It was downloaded from http://www.scyld.com/expert/wake-on-lan.html
+
+Upstream Author: Donald Becker <becker@scyld.com>
+
+Copyright:
+
+  You are free to distribute this software under the terms of the GNU 
+  General Public License. The full text of this license can be found 
+  in the file /usr/share/common-licenses/GPL.
--- etherwake-1.09.orig/debian/control
+++ etherwake-1.09/debian/control
@@ -0,0 +1,18 @@
+Source: etherwake
+Section: net
+Priority: optional
+Maintainer: Alain Schroeder <alain@debian.org>
+Build-Depends: debhelper (>> 3.0)
+Standards-Version: 3.6.2
+
+Package: etherwake
+Architecture: any
+Depends: ${shlibs:Depends}
+Description: A little tool to send magic Wake-on-LAN packets
+ You can wake up WOL compliant Computers which have been powered down to
+ sleep mode or start WOL compliant Computers with a BIOS feature.
+ .
+ WOL is an abbreviation for Wake-on-LAN. It is a standard that allows you 
+ to turn on a computer from another location over a network connection.
+ .
+ etherwake also supports WOL passwords.
--- etherwake-1.09.orig/debian/dirs
+++ etherwake-1.09/debian/dirs
@@ -0,0 +1 @@
+usr/sbin
--- etherwake-1.09.orig/debian/rules
+++ etherwake-1.09/debian/rules
@@ -0,0 +1,86 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+export DH_VERBOSE=1
+
+# This is the debhelper compatability version to use.
+export DH_COMPAT=2
+
+configure: configure-stamp
+configure-stamp:
+	dh_testdir
+	# Add here commands to configure the package.
+	
+
+	touch configure-stamp
+
+build: configure-stamp build-stamp
+build-stamp:
+	dh_testdir
+
+	# Add here commands to compile the package.
+	$(MAKE)
+	#/usr/bin/docbook-to-man debian/etherwake.sgml > etherwake.1
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp configure-stamp
+
+	# Add here commands to clean up after the build process.
+	-$(MAKE) clean
+
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+#	Fix below...
+#	mkdir -p debian/tmp/usr/sbin
+
+	# Add here commands to install the package into debian/tmp.
+	$(MAKE) install DESTDIR=`pwd`/debian/etherwake
+
+# Build architecture-independent files here.
+binary-indep: build install
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build install
+#	dh_testversion
+	dh_testdir
+	dh_testroot
+#	dh_installdebconf	
+	dh_installdocs
+	dh_installexamples
+	dh_installmenu
+#	dh_installemacsen
+#	dh_installpam
+#	dh_installinit
+	dh_installcron
+	dh_installman
+	dh_installinfo
+#	dh_undocumented
+	dh_installchangelogs 
+	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+	# You may want to make some executables suid here.
+#	dh_suidregister
+#	dh_makeshlibs
+	dh_installdeb
+#	dh_perl
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install configure
--- etherwake-1.09.orig/debian/postinst
+++ etherwake-1.09/debian/postinst
@@ -0,0 +1,47 @@
+#! /bin/sh
+# postinst script for etherwake
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postinst> `configure' <most-recently-configured-version>
+#        * <old-postinst> `abort-upgrade' <new version>
+#        * <conflictor's-postinst> `abort-remove' `in-favour' <package>
+#          <new-version>
+#        * <deconfigured's-postinst> `abort-deconfigure' `in-favour'
+#          <failed-install-package> <version> `removing'
+#          <conflicting-package> <version>
+# for details, see /usr/share/doc/packaging-manual/
+#
+# quoting from the policy:
+#     Any necessary prompting should almost always be confined to the
+#     post-installation script, and should be protected with a conditional
+#     so that unnecessary prompting doesn't happen if a package's
+#     installation fails and the `postinst' is called with `abort-upgrade',
+#     `abort-remove' or `abort-deconfigure'.
+
+case "$1" in
+    configure)
+
+    ;;
+
+    abort-upgrade|abort-remove|abort-deconfigure)
+
+    ;;
+
+    *)
+        echo "postinst called with unknown argument \`$1'" >&2
+        exit 0
+    ;;
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+
+exit 0
+
+
--- etherwake-1.09.orig/debian/postrm
+++ etherwake-1.09/debian/postrm
@@ -0,0 +1,36 @@
+#! /bin/sh
+# postrm script for etherwake
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postrm> `remove'
+#        * <postrm> `purge'
+#        * <old-postrm> `upgrade' <new-version>
+#        * <new-postrm> `failed-upgrade' <old-version>
+#        * <new-postrm> `abort-install'
+#        * <new-postrm> `abort-install' <old-version>
+#        * <new-postrm> `abort-upgrade' <old-version>
+#        * <disappearer's-postrm> `disappear' <r>overwrit>r> <new-version>
+# for details, see /usr/share/doc/packaging-manual/
+
+case "$1" in
+       purge|remove|upgrade|failed-upgrade|abort-install|abort-upgrade|disappear)
+
+
+        ;;
+
+    *)
+        echo "postrm called with unknown argument \`$1'" >&2
+        exit 0
+
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+
+
--- etherwake-1.09.orig/debian/preinst
+++ etherwake-1.09/debian/preinst
@@ -0,0 +1,42 @@
+#! /bin/sh
+# preinst script for etherwake
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <new-preinst> `install'
+#        * <new-preinst> `install' <old-version>
+#        * <new-preinst> `upgrade' <old-version>
+#        * <old-preinst> `abort-upgrade' <new-version>
+#
+# For details see /usr/share/doc/packaging-manual/
+
+case "$1" in
+    install|upgrade)
+#        if [ "$1" = "upgrade" ]
+#        then
+#            start-stop-daemon --stop --quiet --oknodo  \
+#                --pidfile /var/run/etherwake.pid  \
+#                --exec /usr/sbin/etherwake 2>/dev/null || true
+#        fi
+    ;;
+
+    abort-upgrade)
+    ;;
+
+    *)
+        echo "preinst called with unknown argument \`$1'" >&2
+        exit 0
+    ;;
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+
+exit 0
+
+
--- etherwake-1.09.orig/debian/manpages
+++ etherwake-1.09/debian/manpages
@@ -0,0 +1 @@
+etherwake.8
--- etherwake-1.09.orig/debian/prerm
+++ etherwake-1.09/debian/prerm
@@ -0,0 +1,37 @@
+#! /bin/sh
+# prerm script for etherwake
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <prerm> `remove'
+#        * <old-prerm> `upgrade' <new-version>
+#        * <new-prerm> `failed-upgrade' <old-version>
+#        * <conflictor's-prerm> `remove' `in-favour' <package> <new-version>
+#        * <deconfigured's-prerm> `deconfigure' `in-favour'
+#          <package-being-installed> <version> `removing'
+#          <conflicting-package> <version>
+# for details, see /usr/share/doc/packaging-manual/
+
+case "$1" in
+    remove|upgrade|deconfigure)
+#       install-info --quiet --remove /usr/info/etherwake.info.gz
+        ;;
+    failed-upgrade)
+        ;;
+    *)
+        echo "prerm called with unknown argument \`$1'" >&2
+        exit 0
+    ;;
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+
+exit 0
+
+
