diff -urN ipkg-0.99.154.orig/ipkg_conf.c ipkg-0.99.154/ipkg_conf.c
--- ipkg-0.99.154.orig/ipkg_conf.c	2005-05-30 22:06:10.000000000 +0200
+++ ipkg-0.99.154/ipkg_conf.c	2005-12-21 13:32:05.000000000 +0100
@@ -180,6 +180,13 @@
         sprintf (lists_dir,"%s",IPKG_CONF_LISTS_DIR);
      }
 
+     if (args->offline_root) {
+	     char *tmp = malloc(strlen(lists_dir) + strlen(args->offline_root) + 1);
+	     sprintf_alloc(&tmp, "%s/%s",args->offline_root,lists_dir);
+	     free(lists_dir);
+	     lists_dir = tmp;
+     }
+
      pending_dir = malloc(strlen(lists_dir)+strlen("/pending")+5);
      snprintf(pending_dir,strlen(lists_dir)+strlen("/pending") ,"%s%s",lists_dir,"/pending");
 
