From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 10 Dec 2012 16:57:53 +0100
Subject: [PATCH] HACK: configure.in: remove uint32_t sanity check

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/configure.in |   19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/src/configure.in b/src/configure.in
index 152313b..9af93da 100644
--- a/src/configure.in
+++ b/src/configure.in
@@ -3163,25 +3163,6 @@ AC_CHECK_SIZEOF([long])
 AC_CHECK_SIZEOF([time_t])
 AC_CHECK_SIZEOF([off_t])
 
-dnl Make sure that uint32_t is really 32 bits unsigned.
-AC_MSG_CHECKING([uint32_t is 32 bits])
-AC_TRY_RUN([
-#ifdef HAVE_STDINT_H
-# include <stdint.h>
-#endif
-#ifdef HAVE_INTTYPES_H
-# include <inttypes.h>
-#endif
-main() {
-  uint32_t nr1 = (uint32_t)-1;
-  uint32_t nr2 = (uint32_t)0xffffffffUL;
-  if (sizeof(uint32_t) != 4 || nr1 != 0xffffffffUL || nr2 + 1 != 0) exit(1);
-  exit(0);
-}],
-AC_MSG_RESULT(ok),
-AC_MSG_ERROR([WRONG!  uint32_t not defined correctly.]),
-AC_MSG_ERROR([could not compile program using uint32_t.]))
-
 dnl Check for memmove() before bcopy(), makes memmove() be used when both are
 dnl present, fixes problem with incompatibility between Solaris 2.4 and 2.5.
 
