--- SDL-1.2.10.orig/sdl.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ SDL-1.2.10/sdl.pc.in	2006-06-08 11:34:15.000000000 +0200
@@ -0,0 +1,14 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include/SDL
+
+Name: sdl
+Description: SDL 1.2.10 
+Version: 1.2.10
+Requires:
+Conflicts:
+Libs: -L${libdir} @SDL_LIBS@ @SHARED_SYSTEM_LIBS@
+Libs.private: @SDL_STATIC_LIBS@ @STATIC_SYSTEM_LIBS@
+Cflags: -I${includedir} @SDL_CFLAGS@
+
