---
 Makefile.in |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

Index: b/Makefile.in
===================================================================
--- a/Makefile.in
+++ b/Makefile.in
@@ -214,10 +214,9 @@ srcdir = @srcdir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 themedir = @themedir@
-top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libs src docs misc scripts
+SUBDIRS = src docs misc scripts
 ACLOCAL_AMFLAGS = -I m4
 LIBFREETYPE2_SOURCE_INTERNAL = libs/freetype-2.3.5
 LIBJPEG_SOURCE_INTERNAL = libs/jpeg-6b
