diff --git a/ethtool-util.h b/ethtool-util.h
index 5572771..8ae7e47 100644
--- a/ethtool-util.h
+++ b/ethtool-util.h
@@ -5,11 +5,7 @@
 
 #include <sys/types.h>
 
-/* hack, so we may include kernel's ethtool.h */
-typedef unsigned long long __u64;
-typedef __uint32_t __u32;         /* ditto */
-typedef __uint16_t __u16;         /* ditto */
-typedef __uint8_t __u8;           /* ditto */
+#include "ethtool-copy.h"
 
 /* historical: we used to use kernel-like types; remove these once cleaned */
 typedef unsigned long long u64;
@@ -17,8 +13,6 @@ typedef __uint32_t u32;         /* ditto */
 typedef __uint16_t u16;         /* ditto */
 typedef __uint8_t u8;           /* ditto */
 
-#include "ethtool-copy.h"
-
 /* National Semiconductor DP83815, DP83816 */
 int natsemi_dump_regs(struct ethtool_drvinfo *info, struct ethtool_regs *regs);
 int natsemi_dump_eeprom(struct ethtool_drvinfo *info,
