From: Jon Ringle <jon@ringle.org>
Subject: define "__socklen_t_defined"

Port from 1.5.15

Index: klibc-1.5.25/usr/include/sys/socket.h
===================================================================
--- klibc-1.5.25.orig/usr/include/sys/socket.h
+++ klibc-1.5.25/usr/include/sys/socket.h
@@ -152,6 +152,7 @@ struct sockaddr {
 	char            sa_data[14];    /* 14 bytes of protocol address */
 };
 typedef int socklen_t;
+#define __socklen_t_defined 1
 struct msghdr {
 	void *msg_name;
 	int msg_namelen;
