From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Wed, 2 Sep 2015 18:20:10 +0200
Subject: [PATCH] auto/endianness: add mechanism allowing to force result

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 auto/endianness | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/auto/endianness b/auto/endianness
index 1b552b6b37b8..0f2394cfb6a1 100644
--- a/auto/endianness
+++ b/auto/endianness
@@ -32,7 +32,7 @@ ngx_test="$CC $CC_TEST_FLAGS $CC_AUX_FLAGS \
 eval "$ngx_test >> $NGX_AUTOCONF_ERR 2>&1"
 
 if [ -x $NGX_AUTOTEST ]; then
-    if $NGX_AUTOTEST >/dev/null 2>&1; then
+    if $NGX_AUTOTEST >/dev/null 2>&1 || test "$ngx_force_have_little_endian" = "yes"; then
         echo " little endian"
         have=NGX_HAVE_LITTLE_ENDIAN . auto/have
     else
