From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Sat, 9 Jan 2010 19:17:59 +0100
Subject: [PATCH] cppunit-config.in: add SYSROOT support

Forwarded: not-needed
Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 cppunit-config.in |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/cppunit-config.in b/cppunit-config.in
index 51e8b58..9ee6fc9 100644
--- a/cppunit-config.in
+++ b/cppunit-config.in
@@ -81,14 +81,14 @@ if test "$echo_exec_prefix" = "yes"; then
 fi
 
 if test "$echo_cflags" = "yes"; then
-      if test "$includedir" != "/usr/include" ; then
-            echo -I$includedir
+      if test "${SYSROOT}$includedir" != "/usr/include" ; then
+            echo -I${SYSROOT}$includedir
       fi
 fi
 
 if test "$echo_libs" = "yes"; then
-      if test @libdir@ != /usr/lib ; then
-            my_linker_flags="-L@libdir@"
+      if test ${SYSROOT}@libdir@ != /usr/lib ; then
+            my_linker_flags="-L${SYSROOT}@libdir@"
       fi
       echo ${my_linker_flags} -lcppunit @LIBADD_DL@
 fi      
