--- jpeg-6b/configure.orig	2006-03-27 14:44:54.000000000 +0200
+++ jpeg-6b/configure	2006-03-27 14:51:04.000000000 +0200
@@ -613,6 +613,7 @@
 echo $ac_n "checking whether the C compiler ($CC $CFLAGS $LDFLAGS) works""... $ac_c" 1>&6
 echo "configure:615: checking whether the C compiler ($CC $CFLAGS $LDFLAGS) works" >&5
 
+ac_cv_prog_cc_cross=yes
 ac_ext=c
 # CFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
 ac_cpp='$CPP $CPPFLAGS'
@@ -620,33 +621,33 @@
 ac_link='${CC-cc} -o conftest $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 cross_compiling=$ac_cv_prog_cc_cross
 
-cat > conftest.$ac_ext <<EOF
-#line 625 "configure"
-#include "confdefs.h"
-main(){return(0);}
-EOF
-if { (eval echo configure:629: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest; then
-  ac_cv_prog_cc_works=yes
-  # If we can't run a trivial program, we are probably using a cross compiler.
-  if (./conftest; exit) 2>/dev/null; then
-    ac_cv_prog_cc_cross=no
-  else
-    ac_cv_prog_cc_cross=yes
-  fi
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  ac_cv_prog_cc_works=no
-fi
-rm -fr conftest*
-
-echo "$ac_t""$ac_cv_prog_cc_works" 1>&6
-if test $ac_cv_prog_cc_works = no; then
-  { echo "configure: error: installation or configuration problem: C compiler cannot create executables." 1>&2; exit 1; }
-fi
-echo $ac_n "checking whether the C compiler ($CC $CFLAGS $LDFLAGS) is a cross-compiler""... $ac_c" 1>&6
-echo "configure:649: checking whether the C compiler ($CC $CFLAGS $LDFLAGS) is a cross-compiler" >&5
-echo "$ac_t""$ac_cv_prog_cc_cross" 1>&6
+#cat > conftest.$ac_ext <<EOF
+##line 625 "configure"
+##include "confdefs.h"
+#main(){return(0);}
+#EOF
+#if { (eval echo configure:629: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest; then
+#  ac_cv_prog_cc_works=yes
+#  # If we can't run a trivial program, we are probably using a cross compiler.
+#  if (./conftest; exit) 2>/dev/null; then
+#    ac_cv_prog_cc_cross=no
+#  else
+#    ac_cv_prog_cc_cross=yes
+#  fi
+#else
+#  echo "configure: failed program was:" >&5
+#  cat conftest.$ac_ext >&5
+#  ac_cv_prog_cc_works=no
+#fi
+#rm -fr conftest*
+
+#echo "$ac_t""$ac_cv_prog_cc_works" 1>&6
+#if test $ac_cv_prog_cc_works = no; then
+#  { echo "configure: error: installation or configuration problem: C compiler cannot create executables." 1>&2; exit 1; }
+#fi
+#echo $ac_n "checking whether the C compiler ($CC $CFLAGS $LDFLAGS) is a cross-compiler""... $ac_c" 1>&6
+#echo "configure:649: checking whether the C compiler ($CC $CFLAGS $LDFLAGS) is a cross-compiler" >&5
+#echo "$ac_t""$ac_cv_prog_cc_cross" 1>&6
 cross_compiling=$ac_cv_prog_cc_cross
 
 echo $ac_n "checking whether we are using GNU C""... $ac_c" 1>&6
