From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 15 Mar 2012 08:55:26 +0100
Subject: [PATCH] cmake: allow specifying the preferred Python2 version

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 cmake/Modules/FindPythonInterpWithDebug.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Modules/FindPythonInterpWithDebug.cmake b/cmake/Modules/FindPythonInterpWithDebug.cmake
index 7722d956917e..06c5aff02580 100644
--- a/cmake/Modules/FindPythonInterpWithDebug.cmake
+++ b/cmake/Modules/FindPythonInterpWithDebug.cmake
@@ -1,4 +1,4 @@
-find_program(PYTHON_EXECUTABLE NAMES python2.7 python2.6 python2.5)
+find_program(PYTHON_EXECUTABLE NAMES ${Python_PREFERRED_VERSION} python2.7 python2.6 python2.5)
 
 if (NOT PYTHON_EXECUTABLE)
     find_package(PythonInterp REQUIRED)
