---
 Zend/zend_strtod.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: php-4.4.4/Zend/zend_strtod.c
===================================================================
--- php-4.4.4.orig/Zend/zend_strtod.c
+++ php-4.4.4/Zend/zend_strtod.c
@@ -129,7 +129,8 @@ typedef unsigned long int uint32_t;
  * byte and word endianness. The byte order is still little endian
  * but the word order is big endian.
  */
-#define IEEE_BIG_ENDIAN
+// Hmm... 
+// #define IEEE_BIG_ENDIAN
 #endif
 
 #ifdef __vax__
