From: Denis Oliver Kropp <dok@directfb.org>
Subject: Hack for pxafb which crashed the kernel

Signed-off-by: Denis Oliver Kropp <dok@directfb.org>

---

diff -pur DirectFB-1.0.1/systems/fbdev/fbdev.c DirectFB-1.0.1-hacked/systems/fbdev/fbdev.c
--- DirectFB-1.0.1/systems/fbdev/fbdev.c	2007-08-19 23:52:06.000000000 +0200 
+++ DirectFB-1.0.1-hacked/systems/fbdev/fbdev.c	2007-11-23 19:58:47.000000000 +0100 
@@ -2449,7 +2449,10 @@ fbdev_ioctl_call_handler( int   caller,
                ioctl( dfb_fbdev->vt->fd, KDSETMODE, KD_GRAPHICS );
      }
 
-     ret = ioctl( dfb_fbdev->fd, call_arg, call_ptr );
+     if (call_arg != FBIOPUT_VSCREENINFO)
+          ret = ioctl( dfb_fbdev->fd, call_arg, call_ptr );
+     else
+          ret = 0;
 
      if (dfb_config->vt) {
           if (call_arg == FBIOPUT_VSCREENINFO) {
