---
 autogen.sh |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

Index: autogen.sh
===================================================================
--- autogen.sh.orig
+++ autogen.sh
@@ -104,7 +104,4 @@ aclocal \
   && ${LIBTOOLIZE} --force --copy \
   && autoheader --force \
   && automake --add-missing --foreign --copy  --include-deps \
-  && autoconf --force \
-  && echo Now running configure to configure gpsd \
-  && echo "./configure $@" \
-  && ./configure $@
+  && autoconf --force
