--- nfs-utils-1.0.1/config.mk.in	2004-02-04 23:00:30.000000000 +0100
+++ nfs-utils-1.0.1-ptx/config.mk.in	2004-02-04 23:01:22.000000000 +0100
@@ -55,15 +55,14 @@
 INSTALL		= install
 MAN2PS		= groff -Tps -man
 
-AFLAGS		= -I$(TOP)support/include \
-		  -Wall $(ARCHFLAGS) -pipe
+AFLAGS		= -Wall $(ARCHFLAGS) -pipe
 ifdef KERNEL_INCDIR
 AFLAGS	       += -I$(KERNEL_INCDIR)
 endif
 
 ALL-CFLAGS	= $(AFLAGS) $(CCOPTS) -DVERSION="\"$(VERSION)\""
 CC		= @CC@
-CFLAGS		= @CFLAGS@ $(ALL-CFLAGS)
+CFLAGS		= -I$(TOP)support/include @CFLAGS@ $(ALL-CFLAGS)
 CC_FOR_BUILD	= @CC_FOR_BUILD@
 CFLAGS_FOR_BUILD= -O2 $(ALL-CFLAGS)
 
