From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 17 Jul 2014 17:47:27 +0200
Subject: [PATCH] fix glibc version check

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 libbb/unarchive.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libbb/unarchive.c b/libbb/unarchive.c
index d583767fe9e1..e2e016b846d4 100644
--- a/libbb/unarchive.c
+++ b/libbb/unarchive.c
@@ -237,7 +237,7 @@ extract_archive(FILE *src_stream, FILE *out_stream,
 		 * file pointed to, so dont try and change the date or mode, lchown does
 		 * does the right thing, but isnt available in older versions of libc */
 		if (S_ISLNK(file_entry->mode)) {
-#if (__GLIBC__ > 2) && (__GLIBC_MINOR__ > 1)
+#if (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 1)
 			lchown(full_name, file_entry->uid, file_entry->gid);
 #endif
 		} else {
