From edec345978a5f2bb283ad35bab594ff75ea5b4c1 Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Sun, 10 Jan 2010 01:04:21 +0100
Subject: [PATCH] xft-config: add SYSROOT support

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 xft-config.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/xft-config.in b/xft-config.in
index 549b73a..6f25ecc 100644
--- a/xft-config.in
+++ b/xft-config.in
@@ -84,14 +84,14 @@ if test "$echo_exec_prefix" = "yes" ; then
 fi
 
 if test "$echo_cflags" = "yes" ; then
- cflags="-I${includedir} ${freetypecflags} ${fontconfigcflags} ${xrendercflags}"
+ cflags="-I${SYSROOT}${includedir} ${freetypecflags} ${fontconfigcflags} ${xrendercflags}"
  echo $cflags
 fi
 
 if test "$echo_libs" = "yes" ; then
  libs="-lXft -lX11 ${freetypelibs} ${fontconfiglibs} ${xrenderlibs}"
- if test "${libdir}" != "/usr/lib" ; then
-  echo -L${libdir} $libs
+ if test "${SYSROOT}${libdir}" != "/usr/lib" ; then
+  echo -L${SYSROOT}${libdir} $libs
  else
   echo $libs
  fi
-- 
1.6.6

