diff -ur samba-3.0.33-orig/source/configure samba-3.0.33/source/configure
--- samba-3.0.33-orig/source/configure	2008-11-20 15:13:46.000000000 +0100
+++ samba-3.0.33/source/configure	2009-01-02 15:34:24.000000000 +0100
@@ -34672,18 +34672,9 @@
     *linux*)
        # glibc <= 2.3.2 has a broken getgrouplist
        if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+  linux_getgrouplist_ok=maybe
 else
   cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
 
 #include <unistd.h>
 #include <sys/utsname.h>
@@ -34735,6 +34726,7 @@
 fi
 
 
+
        if test x"$linux_getgrouplist_ok" = x"yes"; then
 
 cat >>confdefs.h <<\_ACEOF
@@ -34742,6 +34734,13 @@
 _ACEOF
 
        fi
+       if test x"$linux_getgrouplist_ok" = x"maybe"; then
+
+cat >>confdefs.h <<\_ACEOF
+#define HAVE_GETGROUPLIST 1
+_ACEOF
+
+       fi
        ;;
     *)
 
