Subject: Make the buildsystem be gracious to us.

Signed-off-by: Juergen Beisert juergen@kreuzholzen.de

---
 Makefile      |    2 +-
 misc/Makefile |    6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

Index: misc/Makefile
===================================================================
--- misc/Makefile.orig
+++ misc/Makefile
@@ -1,10 +1,14 @@
 SSOBJ=ss.o ssfilter.o
 LNSTATOBJ=lnstat.o lnstat_util.o
 
-TARGETS=ss nstat ifstat rtacct arpd lnstat
+TARGETS=ss nstat ifstat rtacct lnstat
 
 include ../Config
 
+ifdef BUILD_ARPD
+TARGETS += arpd
+endif
+
 all: $(TARGETS)
 
 ss: $(SSOBJ) $(LIBUTIL)
Index: Makefile
===================================================================
--- Makefile.orig
+++ Makefile
@@ -23,7 +23,7 @@ ADDLIB+=ipx_ntop.o ipx_pton.o
 CC = gcc
 HOSTCC = gcc
 CCOPTS = -D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall
-CFLAGS = $(CCOPTS) -I../include $(DEFINES)
+CFLAGS = $(CROSS_CPPFLAGS) $(CCOPTS) -I../include $(DEFINES)
 YACCFLAGS = -d -t -v
 
 LDLIBS += -L../lib -lnetlink -lutil
