From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Wed, 2 Dec 2015 11:49:11 +0100
Subject: [PATCH] remove bogus 'inline'

It breaks compiling with gcc-5.x

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 crc32.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/crc32.c b/crc32.c
index bae73847d5d3..810fcc0d054a 100644
--- a/crc32.c
+++ b/crc32.c
@@ -72,7 +72,6 @@ static const unsigned long crctab[256] = {
   0xA2F33668, 0xBCB4666D, 0xB8757BDA, 0xB5365D03, 0xB1F740B4
 };
 
-inline
 unsigned long crc32(  const void* buffer, 
 		      unsigned long length, 
 		      unsigned long crc)
