From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 14 Jun 2012 17:56:38 +0200
Subject: [PATCH] set and use LDFLAGS

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 Makefile.in |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 1cda327..5e47a08 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -2,6 +2,7 @@ SHELL		= @SHELL@
 CC		= @CC@
 CPPFLAGS	= @CPPFLAGS@
 CFLAGS		= @CFLAGS@
+LDFLAGS		= @LDFLAGS@
 LIBS		= @LIBS@
 EXEEXT		= @EXEEXT@
 srcdir		= @srcdir@
@@ -55,7 +56,7 @@ LOCALEDEF_OBJS	= charmap.o charmap-dir.o ld-address.o ld-collate.o \
 all:		localedef$(EXEEXT)
 
 localedef$(EXEEXT): $(LOCALEDEF_OBJS)
-		$(CC) -o $@ $(LOCALEDEF_OBJS) $(LIBS)
+		$(CC) -o $@ $(LOCALEDEF_OBJS) $(LDFLAGS) $(LIBS)
 
 clean:
 		rm -f locale$(EXEEXT) $(LOCALEDEF_OBJS)
