--- xc/programs/Xserver1/Imakefile	2003-12-03 04:59:06.000000000 +0000
+++ xc/programs/Xserver/Imakefile	2004-02-16 15:36:33.000000000 +0000
@@ -776,17 +776,17 @@
 XCOMM server with Keith's C&T driver
 XCOMM
 
-       VESADIR = $(KDRIVE)/vesa
-          VESA = $(VESADIR)/LibraryTargetName(vesa)
+      FBDEVDIR = $(KDRIVE)/fbdev
+         FBDEV = $(FBDEVDIR)/LibraryTargetName(fbdev)
       CHIPSDIR = $(KDRIVE)/chips
          CHIPS = $(CHIPSDIR)/LibraryTargetName(chips)
 
         KDDIRS = StdKdDirs
   
    CHIPSDIRS = $(STDDIRS) $(KDDIRS) \
-		 $(FBDEVDIR) $(VESADIR) $(SHADOWDIR) $(LAYERDIR) $(CHIPSDIR)
+		 $(FBDEVDIR) $(SHADOWDIR) $(LAYERDIR) $(CHIPSDIR)
    
-   CHIPSLIBS = PreFbLibs $(CHIPS) $(VESA) $(LAYER) KdLibs FbPostFbLibs
+   CHIPSLIBS = PreFbLibs $(CHIPS) $(FBDEV) $(LAYER) KdLibs FbPostFbLibs
 CHIPSSYSLIBS = StdKdSysLibs
 
 #if HasParallelMake
