---
 Src/configure |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

Index: Frodo-4.1b/Src/configure
===================================================================
--- Frodo-4.1b.orig/Src/configure
+++ Frodo-4.1b/Src/configure
@@ -6806,6 +6806,12 @@ echo "${ECHO_T}SVGAlib" >&6
     TARGET=svgalib
     CFLAGS="$CFLAGS -D__svgalib__"
     LIBS="$LIBS -lvga"
+  elif [ x"$HAVE_SDL" = "xyes" ]; then
+    echo "$as_me:6817: result: SDL" >&5
+  echo "${ECHO_T}SDL" >&6
+    TARGET=sdl
+    CFLAGS="$CFLAGS $SDL_CFLAGS -DHAVE_SDL"
+    LIBS="$LIBS $SDL_LIBS"
   else
     echo "$as_me:6810: result: Ummm..." >&5
 echo "${ECHO_T}Ummm..." >&6
@@ -6813,12 +6819,6 @@ echo "${ECHO_T}Ummm..." >&6
 echo "$as_me: error: Neither X nor SVGAlib found, don't know what target to use." >&2;}
    { (exit 1); exit 1; }; }
   fi
-elif [ x"$HAVE_SDL" = "xyes" ]; then
-  echo "$as_me:6817: result: SDL" >&5
-echo "${ECHO_T}SDL" >&6
-  TARGET=sdl
-  CFLAGS="$CFLAGS $SDL_CFLAGS -DHAVE_SDL"
-  LIBS="$LIBS $SDL_LIBS"
 else
   echo "$as_me:6823: result: X Window System" >&5
 echo "${ECHO_T}X Window System" >&6
