From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 30 Mar 2018 16:35:56 +0200
Subject: [PATCH] fix parallel building

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 60681422ddfc..2df737816742 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -159,7 +159,7 @@ $(OBJDIR_NORMAL_LIB): $(OBJDIR_TSTAMP) $(CONFIG_H) $(OBJS)
 	@echo ""
 	@echo $(NORMAL_LIB) created in $(OBJDIR)
 
-$(ELFDIR_ELF_LIB): $(ELFDIR) $(CONFIG_H) $(ELFOBJS) $(VERSION_SCRIPT)
+$(ELFDIR_ELF_LIB): $(ELFDIR_TSTAMP) $(CONFIG_H) $(ELFOBJS) $(VERSION_SCRIPT)
 	-$(RM) $(ELFDIR_ELF_LIB)
 	cd $(ELFDIR) && $(ELF_LINK_CMD) -o $(ELFLIB_BUILD_NAME) $(OFILES) $(LDFLAGS) $(ELF_DEP_LIBS)
 	if [ $(ELFLIB_MAJOR) != $(ELFLIB_BUILD_NAME) ]; then \
