From: unknown author <unknown.author@example.com>
Date: Tue, 8 Feb 2011 15:10:31 +0100
Subject: [PATCH] python: don't add rpaths in setup.py

We don't add rpaths.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 setup.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/setup.py b/setup.py
index c556209..66a97bd 100644
--- a/setup.py
+++ b/setup.py
@@ -1077,7 +1077,6 @@ class PyBuildExt(build_ext):
             exts.append(Extension('_bsddb', ['_bsddb.c'],
                                   depends = ['bsddb.h'],
                                   library_dirs=dblib_dir,
-                                  runtime_library_dirs=dblib_dir,
                                   include_dirs=db_incs,
                                   libraries=dblibs))
         else:
@@ -1187,7 +1186,6 @@ class PyBuildExt(build_ext):
                                   include_dirs=["Modules/_sqlite",
                                                 sqlite_incdir],
                                   library_dirs=sqlite_libdir,
-                                  runtime_library_dirs=sqlite_libdir,
                                   extra_link_args=sqlite_extra_link_args,
                                   libraries=["sqlite3",]))
         else:
