From: Ladislav Michl <ladis@linux-mips.org>
Date: Tue, 3 Aug 2010 13:00:00 +0200
Subject: [PATCH] make mkubifs optional

Signed-off-by: Ladislav Michl <ladis@linux-mips.org>
---
 Makefile |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 5de9382..462a8f9 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,11 @@ ifeq ($(WITHOUT_LZO), 1)
   CPPFLAGS += -DWITHOUT_LZO
 endif
 
-SUBDIRS = ubi-utils mkfs.ubifs
+SUBDIRS = ubi-utils
+
+ifneq ($(WITHOUT_MKUBIFS), 1)
+  SUBDIRS += mkfs.ubifs
+endif
 
 TARGETS = ftl_format flash_erase flash_eraseall nanddump doc_loadbios \
 	ftl_check mkfs.jffs2 flash_lock flash_unlock flash_info \
-- 
1.7.1

