---
 configure |    6 ++++++
 1 file changed, 6 insertions(+)

Index: mod_python-3.3.1/configure
===================================================================
--- mod_python-3.3.1.orig/configure
+++ mod_python-3.3.1/configure
@@ -2709,6 +2709,7 @@ else
   # check Apache version
   echo "$as_me:$LINENO: checking Apache version" >&5
 echo $ECHO_N "checking Apache version... $ECHO_C" >&6
+  if test "x$cross_compiling" = "xno"; then
   HTTPD="`${APXS} -q SBINDIR`/`${APXS} -q TARGET`"
   ver=`$HTTPD -v | awk '/version/ {print $3}' | awk -F/ '{print $2}'`
   echo "$as_me:$LINENO: result: $ver" >&5
@@ -2734,6 +2735,11 @@ echo $ECHO_N "checking for Apache includ
   AP_INCLUDES="-I`${APXS} -q INCLUDEDIR`"
   echo "$as_me:$LINENO: result: $AP_INCLUDES" >&5
 echo "${ECHO_T}$AP_INCLUDES" >&6
+  else
+    { echo "$as_me:$LINENO: result: cross_compiling, assuming recent new version..." >&5
+echo "${ECHO_T}cross_compiling, assuming recent new version..." >&6; }
+  fi
+
 
     if test "`uname`" = "SunOS"; then
     echo "$as_me:$LINENO: checking for gcc on Solaris possible missing _eprintf problem" >&5
