---
 glib/gnulib/vasnprintf.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: glib-2.14.5/glib/gnulib/vasnprintf.c
===================================================================
--- glib-2.14.5.orig/glib/gnulib/vasnprintf.c
+++ glib-2.14.5/glib/gnulib/vasnprintf.c
@@ -560,7 +560,7 @@ vasnprintf (char *resultbuf, size_t *len
 		      break;
 
 		    case 'c':
-# ifdef HAVE_WINT_T
+# if defined(HAVE_WINT_T) && defined(MB_CUR_MAX)
 		      if (type == TYPE_WIDE_CHAR)
 			tmp_length = MB_CUR_MAX;
 		      else
@@ -569,7 +569,7 @@ vasnprintf (char *resultbuf, size_t *len
 		      break;
 
 		    case 's':
-# ifdef HAVE_WCHAR_T
+# if defined(HAVE_WCHAR_T) && defined(MB_CUR_MAX)
 		      if (type == TYPE_WIDE_STRING)
 			tmp_length =
 			  local_wcslen (a.arg[dp->arg_index].a.a_wide_string)
