From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 14 Jun 2018 11:41:00 +0200
Subject: [PATCH] configure: make atk-bridge optional

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure.ac                | 12 +++++++++---
 gtk/a11y/gtkaccessibility.c |  4 ++--
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index fef0ecebf3e2..bae5d7f5cbae 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1393,14 +1393,20 @@ AC_SUBST(GDK_DEP_CFLAGS)
 # Check for Accessibility Toolkit flags
 ########################################
 
+ATK_PACKAGES="atk"
+
 if test x$enable_x11_backend = xyes; then
-   ATK_PACKAGES="atk atk-bridge-2.0"
-else
-   ATK_PACKAGES="atk"
+  PKG_CHECK_MODULES(ATK_BRIDGE, "atk-bridge-2.0", have_atk_bridge=yes, have_atk_bridge=no)
+  if test "x$have_atk_bridge" = "xyes"; then
+    ATK_PACKAGES="$ATK_PACKAGES atk-bridge-2.0"
+    GDK_WINDOWING="$GDK_WINDOWING
+#define GDK_WINDOWING_ATK_BRIDGE"
+  fi
 fi
 
 PKG_CHECK_MODULES(ATK, $ATK_PACKAGES)
 
+
 GTK_PACKAGES="atk >= atk_required_version cairo >= cairo_required_version cairo-gobject >= cairo_required_version gdk-pixbuf-2.0 >= gdk_pixbuf_required_version gio-2.0 >= glib_required_version"
 GTK_PRIVATE_PACKAGES="$ATK_PACKAGES $WAYLAND_PACKAGES $MIR_PACKAGES epoxy >= epoxy_required_version"
 if test "x$enable_x11_backend" = xyes -o "x$enable_wayland_backend" = xyes; then
diff --git a/gtk/a11y/gtkaccessibility.c b/gtk/a11y/gtkaccessibility.c
index 7f0e5207d915..81b2ed0d55b3 100644
--- a/gtk/a11y/gtkaccessibility.c
+++ b/gtk/a11y/gtkaccessibility.c
@@ -37,7 +37,7 @@
 #include <gtk/gtktogglebutton.h>
 #include <gtk/gtkaccessible.h>
 
-#ifdef GDK_WINDOWING_X11
+#ifdef GDK_WINDOWING_ATK_BRIDGE
 #include <atk-bridge.h>
 #endif
 
@@ -988,7 +988,7 @@ _gtk_accessibility_init (void)
   _gtk_accessibility_override_atk_util ();
   do_window_event_initialization ();
 
-#ifdef GDK_WINDOWING_X11
+#ifdef GDK_WINDOWING_ATK_BRIDGE
   atk_bridge_adaptor_init (NULL, NULL);
 #endif
 
