From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 29 Oct 2011 19:06:44 +0200
Subject: [PATCH] install to DESTDIR

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 6fcb8f7..332c180 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -7,7 +7,7 @@ srcdir		= @srcdir@
 VPATH		= @srcdir@
 INSTALL_PREFIX	= @prefix@
 INSTALL		= @INSTALL@
-INSTALL_SBIN	= $(INSTALL_PREFIX)/sbin
+INSTALL_SBIN	= $(DESTDIR)$(INSTALL_PREFIX)/sbin
 MISSING_SOURCES	= @LOCAL_MISSING_SOURCES@
 MISSING_OBJECTS = $(MISSING_SOURCES:.c=.o)
 CC		= @CC@
