---
 main.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: fbv-1.0b/main.c
===================================================================
--- fbv-1.0b.orig/main.c
+++ fbv-1.0b/main.c
@@ -201,13 +201,21 @@ int show_image(char *filename)
 	unsigned char * alpha = NULL;
 	
 	int x_size, y_size, screen_width, screen_height;
-	int x_pan, y_pan, x_offs, y_offs, refresh = 1, c, ret = 1;
+	int x_pan = 0;
+	int y_pan = 0;
+	int x_offs = 0;
+	int y_offs = 0;
+	int refresh = 1;
+	int c;
+	int ret = 1;
 	int delay = opt_delay, retransform = 1;
 	
 	int transform_stretch = opt_stretch, transform_enlarge = opt_enlarge, transform_cal = (opt_stretch == 2),
 	    transform_iaspect = opt_ignore_aspect, transform_rotation = 0;
 	
 	struct image i;
+
+	memset(&i, 0, sizeof(struct image));
 	
 #ifdef FBV_SUPPORT_GIF
 	if(fh_gif_id(filename))
