Index: client.mk
===================================================================
RCS file: /cvsroot/mozilla/client.mk,v
retrieving revision 1.198.10.1.4.1
retrieving revision 1.198.10.1.8.1
diff -u -r1.198.10.1.4.1 -r1.198.10.1.8.1
--- client.mk	2 Oct 2003 23:15:42 -0000	1.198.10.1.4.1
+++ client.mk	27 Oct 2003 00:59:57 -0000	1.198.10.1.8.1
@@ -52,17 +52,17 @@
 #
 # For branches, uncomment the MOZ_CO_TAG line with the proper tag,
 # and commit this file on that tag.
-MOZ_CO_TAG = FIREBIRD_0_7_RELEASE
-NSPR_CO_TAG = FIREBIRD_0_7_RELEASE
-PSM_CO_TAG = FIREBIRD_0_7_RELEASE
-NSS_CO_TAG = FIREBIRD_0_7_RELEASE
-LDAPCSDK_CO_TAG = FIREBIRD_0_7_RELEASE
-ACCESSIBLE_CO_TAG = FIREBIRD_0_7_RELEASE
-IMGLIB2_CO_TAG = FIREBIRD_0_7_RELEASE
-IPC_CO_TAG = FIREBIRD_0_7_RELEASE
-TOOLKIT_CO_TAG = FIREBIRD_0_7_RELEASE
-BROWSER_CO_TAG = FIREBIRD_0_7_RELEASE
-MAIL_CO_TAG = FIREBIRD_0_7_RELEASE
+MOZ_CO_TAG = FIREBIRD_0_7_1_RELEASE
+NSPR_CO_TAG = FIREBIRD_0_7_1_RELEASE
+PSM_CO_TAG = FIREBIRD_0_7_1_RELEASE
+NSS_CO_TAG = FIREBIRD_0_7_1_RELEASE
+LDAPCSDK_CO_TAG = FIREBIRD_0_7_1_RELEASE
+ACCESSIBLE_CO_TAG = FIREBIRD_0_7_1_RELEASE
+IMGLIB2_CO_TAG = FIREBIRD_0_7_1_RELEASE
+IPC_CO_TAG = FIREBIRD_0_7_1_RELEASE
+TOOLKIT_CO_TAG = FIREBIRD_0_7_1_RELEASE
+BROWSER_CO_TAG = FIREBIRD_0_7_1_RELEASE
+MAIL_CO_TAG = FIREBIRD_0_7_1_RELEASE
 BUILD_MODULES = all
 
 #######################################################################
Index: configure
===================================================================
RCS file: /cvsroot/mozilla/configure,v
retrieving revision 1.1230
retrieving revision 1.1230.2.1
diff -u -r1.1230 -r1.1230.2.1
--- configure	29 Aug 2003 17:19:05 -0000	1.1230
+++ configure	7 Oct 2003 17:32:34 -0000	1.1230.2.1
@@ -4764,7 +4764,7 @@
 #include "confdefs.h"
 
 int main() {
-#if (__IBMCPP__ < 520)
+#if (__IBMCPP__ < 502)
              #error "Bad compiler"
              #endif
 ; return 0; }
Index: configure.in
===================================================================
RCS file: /cvsroot/mozilla/configure.in,v
retrieving revision 1.1274
retrieving revision 1.1274.2.1
diff -u -r1.1274 -r1.1274.2.1
--- configure.in	29 Aug 2003 17:14:58 -0000	1.1274
+++ configure.in	7 Oct 2003 15:20:52 -0000	1.1274.2.1
@@ -814,7 +814,7 @@
         AC_LANG_CPLUSPLUS
         AC_MSG_CHECKING([for VisualAge C++ compiler version >= 5.0.2.0])
         AC_TRY_COMPILE([],
-            [#if (__IBMCPP__ < 520)
+            [#if (__IBMCPP__ < 502)
              #error "Bad compiler"
              #endif],
             _BAD_COMPILER=,_BAD_COMPILER=1)
Index: browser/app/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/browser/app/Makefile.in,v
retrieving revision 1.26
retrieving revision 1.26.2.1
diff -u -r1.26 -r1.26.2.1
--- browser/app/Makefile.in	27 Jul 2003 09:25:48 -0000	1.26
+++ browser/app/Makefile.in	24 Oct 2003 07:37:45 -0000	1.26.2.1
@@ -142,6 +142,12 @@
 include $(topsrcdir)/config/static-rules.mk
 endif
 
+ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
+ifdef BUILD_STATIC_LIBS
+LIBS	+= -framework QuickTime
+endif
+endif
+
 $(PROGRAM): $(DIST)/lib/$(LIB_PREFIX)xulapp_s.$(LIB_SUFFIX)
 
 ifeq ($(OS_ARCH),OS2)
Index: browser/app/profile/all.js
===================================================================
RCS file: /cvsroot/mozilla/browser/app/profile/all.js,v
retrieving revision 1.61.2.1
retrieving revision 1.61.2.2
diff -u -r1.61.2.1 -r1.61.2.2
--- browser/app/profile/all.js	19 Sep 2003 00:39:03 -0000	1.61.2.1
+++ browser/app/profile/all.js	24 Oct 2003 07:37:46 -0000	1.61.2.2
@@ -128,7 +128,8 @@
 pref("browser.download.progressDnldDialog.enable_launch_reveal_buttons", true);
 pref("browser.download.useProgressDialogs", true);
 pref("browser.download.openSidebar", false);
-pref("browser.download.useDownloadDir", false);
+pref("browser.download.useDownloadDir", true);
+pref("browser.download.folderList", 0);
 
 // various default search settings
 pref("browser.search.defaulturl", "chrome://browser-region/locale/region.properties");
Index: browser/base/content/browser.js
===================================================================
RCS file: /cvsroot/mozilla/browser/base/content/browser.js,v
retrieving revision 1.236
retrieving revision 1.236.2.1
diff -u -r1.236 -r1.236.2.1
--- browser/base/content/browser.js	1 Sep 2003 21:31:57 -0000	1.236
+++ browser/base/content/browser.js	24 Oct 2003 07:37:46 -0000	1.236.2.1
@@ -287,7 +287,13 @@
   var sidebarSplitter;
   if (window.opener) {
     var openerSidebarBox = window.opener.document.getElementById("sidebar-box");
-    if (!openerSidebarBox.hidden) {
+#ifdef XP_MACOSX
+	  // The opener can be the hidden window too, if we're coming from the state where
+    // no windows are open, and the hidden window has no sidebar box. 
+    if (openerSidebarBox && !openerSidebarBox.hidden) {
+#else
+	  if (!openSidebarBox.hidden) {
+#endif
       var sidebarBox = document.getElementById("sidebar-box");
       var sidebarTitle = document.getElementById("sidebar-title");
       sidebarTitle.setAttribute("value", window.opener.document.getElementById("sidebar-title").getAttribute("value"));
Index: browser/base/content/contentAreaUtils.js
===================================================================
RCS file: /cvsroot/mozilla/browser/base/content/contentAreaUtils.js,v
retrieving revision 1.18.2.1
retrieving revision 1.18.2.2
diff -u -r1.18.2.1 -r1.18.2.2
--- browser/base/content/contentAreaUtils.js	23 Sep 2003 19:56:17 -0000	1.18.2.1
+++ browser/base/content/contentAreaUtils.js	24 Oct 2003 07:37:46 -0000	1.18.2.2
@@ -298,7 +298,7 @@
         return "DeskV";
 #endif
 #ifdef XP_MACOSX
-        return "Desk";
+        return "UsrDsk";
 #endif
         return "Home";
       }
Index: config/prebind-address-table
===================================================================
RCS file: /cvsroot/mozilla/config/prebind-address-table,v
retrieving revision 3.3
retrieving revision 3.3.10.1
diff -u -r3.3 -r3.3.10.1
--- config/prebind-address-table	9 Jun 2003 22:25:19 -0000	3.3
+++ config/prebind-address-table	26 Oct 2003 22:15:34 -0000	3.3.10.1
@@ -1,4 +1,4 @@
-0x01000000       @executable_path/libnspr4.dylib
+0x01500000       @executable_path/libnspr4.dylib
 0x02000000       @executable_path/libplds4.dylib
 0x03000000       @executable_path/libplc4.dylib
 0x04000000       @executable_path/libmozjs.dylib
Index: nsprpub/pr/src/misc/prtime.c
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/pr/src/misc/prtime.c,v
retrieving revision 3.12.2.4
retrieving revision 3.12.2.4.22.1
diff -u -r3.12.2.4 -r3.12.2.4.22.1
--- nsprpub/pr/src/misc/prtime.c	27 Feb 2003 00:01:02 -0000	3.12.2.4
+++ nsprpub/pr/src/misc/prtime.c	26 Oct 2003 22:18:09 -0000	3.12.2.4.22.1
@@ -1666,7 +1666,7 @@
  */
 
 #if defined(SUNOS4) || (__GLIBC__ >= 2) || defined(XP_BEOS) \
-        || defined(NETBSD)
+        || defined(NETBSD) || defined(DARWIN)
     a.tm_zone = NULL;
     a.tm_gmtoff = tm->tm_params.tp_gmt_offset + tm->tm_params.tp_dst_offset;
 #endif
Index: toolkit/content/widgets/tabbrowser.xml
===================================================================
RCS file: /cvsroot/mozilla/toolkit/content/widgets/tabbrowser.xml,v
retrieving revision 1.31
retrieving revision 1.31.4.1
diff -u -r1.31 -r1.31.4.1
--- toolkit/content/widgets/tabbrowser.xml	28 Aug 2003 02:16:21 -0000	1.31
+++ toolkit/content/widgets/tabbrowser.xml	24 Oct 2003 07:38:37 -0000	1.31.4.1
@@ -85,7 +85,7 @@
                     ondragdrop="nsDragAndDrop.drop(event, this.parentNode.parentNode.parentNode);
                                 event.stopPropagation();"
                     xbl:inherits="onnewtab"
-                    ondblclick="if (event.target.localName == 'tabs') this.parentNode.parentNode.parentNode.selectedTab = this.parentNode.parentNode.parentNode.addTab();"
+                    ondblclick="if (event.originalTarget.localName != 'tab') this.parentNode.parentNode.parentNode.selectedTab = this.parentNode.parentNode.parentNode.addTab();"
                     onclosetab="var node = this.parentNode;
                                 while (node.localName != 'tabbrowser')
                                   node = node.parentNode;
Index: toolkit/mozapps/downloads/content/pref-downloads.js
===================================================================
RCS file: /cvsroot/mozilla/toolkit/mozapps/downloads/content/pref-downloads.js,v
retrieving revision 1.2.4.1
retrieving revision 1.2.4.2
diff -u -r1.2.4.1 -r1.2.4.2
--- toolkit/mozapps/downloads/content/pref-downloads.js	23 Sep 2003 19:56:33 -0000	1.2.4.1
+++ toolkit/mozapps/downloads/content/pref-downloads.js	24 Oct 2003 07:38:37 -0000	1.2.4.2
@@ -287,7 +287,7 @@
   return aFolderType == "Desktop" ? "DeskV" : "Pers";
 #endif
 #ifdef XP_MACOSX
-  return aFolderType == "Desktop" ? "Desk" : "UsrDocs";
+  return aFolderType == "Desktop" ? "UsrDsk" : "UsrDocs";
 #endif
   return "Home";
 }
Index: xpfe/appshell/src/nsCommandLineServiceMac.cpp
===================================================================
RCS file: /cvsroot/mozilla/xpfe/appshell/src/nsCommandLineServiceMac.cpp,v
retrieving revision 1.53
retrieving revision 1.53.8.1
diff -u -r1.53 -r1.53.8.1
--- xpfe/appshell/src/nsCommandLineServiceMac.cpp	2 Aug 2003 20:17:23 -0000	1.53
+++ xpfe/appshell/src/nsCommandLineServiceMac.cpp	26 Oct 2003 22:15:04 -0000	1.53.8.1
@@ -62,6 +62,7 @@
 #include "jsapi.h"
 #include "nsReadableUtils.h"
 #include "nsICloseAllWindows.h"
+#include "nsIPrefService.h"
 
 #include "nsAEEventHandling.h"
 
@@ -303,11 +304,29 @@
   rv = NS_GetURLSpecFromFile(inFile, specBuf);
   if (NS_FAILED(rv))
     return errAEEventNotHandled;
-  nsAutoString urlString;
-  CopyASCIItoUCS2(specBuf, urlString);    
-  rv = OpenWindow( "chrome://navigator/content", urlString.get() );
+  
+  return OpenURL(specBuf.get());
+}
+
+OSErr nsMacCommandLine::OpenURL(const char* aURL)
+{
+  nsresult rv;
+  
+  nsCOMPtr<nsIPrefBranch> prefBranch(do_GetService(NS_PREFSERVICE_CONTRACTID, &rv));
+
+  nsXPIDLCString browserURL;
+  if (NS_SUCCEEDED(rv))
+    rv = prefBranch->GetCharPref("browser.chromeURL", getter_Copies(browserURL));
+  
+  if (NS_FAILED(rv)) {
+    NS_WARNING("browser.chromeURL not supplied! How is the app supposed to know what the main window is?");
+    browserURL.Assign("chrome://navigator/content/navigator.xul");
+  }
+     
+  rv = OpenWindow(browserURL.get(), NS_ConvertASCIItoUCS2(aURL).get());
   if (NS_FAILED(rv))
     return errAEEventNotHandled;
+    
   return noErr;
 }
 
@@ -357,13 +376,7 @@
 {
   OSErr err = errAEEventNotHandled;
   if (mStartedUp)
-  {
-    nsresult rv;
-    rv = OpenWindow("chrome://navigator/content", NS_ConvertASCIItoUCS2(url).get());
-    if (NS_FAILED(rv))
-      return err;
-        err = noErr;  // we handled it
-  }
+    return OpenURL(url);
   else {
     err = AddToCommandLine("-url");
     if (err == noErr)
Index: xpfe/appshell/src/nsCommandLineServiceMac.h
===================================================================
RCS file: /cvsroot/mozilla/xpfe/appshell/src/nsCommandLineServiceMac.h,v
retrieving revision 1.7
retrieving revision 1.7.32.1
diff -u -r1.7 -r1.7.32.1
--- xpfe/appshell/src/nsCommandLineServiceMac.h	9 Dec 2002 15:36:48 -0000	1.7
+++ xpfe/appshell/src/nsCommandLineServiceMac.h	26 Oct 2003 22:15:04 -0000	1.7.32.1
@@ -78,6 +78,8 @@
   
 protected:
 
+  OSErr           OpenURL(const char* aURL);
+
   nsresult        OpenWindow(const char *chrome, const PRUnichar *url);
     
   char**          mArgs;              // array of arg pointers (augmented argv)
