Index: chrony-1.21-ptx/acquire.c
===================================================================
--- chrony-1.21-ptx.orig/acquire.c
+++ chrony-1.21-ptx/acquire.c
@@ -41,7 +41,12 @@
 
   */
 
-#include "sysincl.h"
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <math.h>
+#include <sys/types.h>
+#include <sys/socket.h>
 
 #include "acquire.h"
 #include "memory.h"
Index: chrony-1.21-ptx/addrfilt.c
===================================================================
--- chrony-1.21-ptx.orig/addrfilt.c
+++ chrony-1.21-ptx/addrfilt.c
@@ -31,7 +31,7 @@
 
   */
 
-#include "sysincl.h"
+#include <assert.h>
 
 #include "addrfilt.h"
 #include "memory.h"
Index: chrony-1.21-ptx/broadcast.c
===================================================================
--- chrony-1.21-ptx.orig/broadcast.c
+++ chrony-1.21-ptx/broadcast.c
@@ -28,7 +28,6 @@
   Deal with broadcast server functions.
   */
 
-#include "sysincl.h"
 #include "memory.h"
 
 #include "addressing.h"
Index: chrony-1.21-ptx/candm.h
===================================================================
--- chrony-1.21-ptx.orig/candm.h
+++ chrony-1.21-ptx/candm.h
@@ -33,7 +33,8 @@
 #ifndef GOT_CANDM_H
 #define GOT_CANDM_H
 
-#include "sysincl.h"
+#include <sys/time.h>
+#include <stdint.h>
 
 /* This is the default port to use for CANDM, if no alternative is
    defined */
Index: chrony-1.21-ptx/client.c
===================================================================
--- chrony-1.21-ptx.orig/client.c
+++ chrony-1.21-ptx/client.c
@@ -29,12 +29,19 @@
   from it whilst running.
   */
 
-#include "sysincl.h"
+#include <chrony_config.h>
+
+#include <stdio.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <ctype.h>
+#include <assert.h>
+#include <stddef.h>
+#include <math.h>
 
 #include "candm.h"
 #include "nameserv.h"
 #include "md5.h"
-#include "version.h"
 #include "getdate.h"
 #include "cmdparse.h"
 #include "pktlength.h"
@@ -2521,7 +2528,7 @@ display_gpl(void)
            "This is free software, and you are welcome to redistribute it\n"
            "under certain conditions.\n"
            "See the GNU General Public License version 2 for details.\n\n",
-           PROGRAM_VERSION_STRING);
+           PACKAGE_VERSION);
 }
 
 /* ================================================== */
@@ -2550,7 +2557,7 @@ main(int argc, char **argv)
     } else if (!strcmp(*argv, "-n")) {
       no_dns = 1;
     } else if (!strcmp("-v", *argv) || !strcmp("--version",*argv)) {
-      printf("chronyc (chrony) version %s\n", PROGRAM_VERSION_STRING);
+      printf("chronyc (chrony) version %s\n", PACKAGE_VERSION);
       exit(0);
     } else if (!strncmp(*argv, "-", 1)) {
       fprintf(stderr, "Usage : %s [-h <hostname>] [-p <port-number>] [-n] [command]\n", progname);
Index: chrony-1.21-ptx/clientlog.c
===================================================================
--- chrony-1.21-ptx.orig/clientlog.c
+++ chrony-1.21-ptx/clientlog.c
@@ -34,7 +34,8 @@
 
   */
 
-#include "sysincl.h"
+#include <string.h>
+
 #include "clientlog.h"
 #include "conf.h"
 #include "memory.h"
Index: chrony-1.21-ptx/clientlog.h
===================================================================
--- chrony-1.21-ptx.orig/clientlog.h
+++ chrony-1.21-ptx/clientlog.h
@@ -32,7 +32,8 @@
 #ifndef GOT_CLIENTLOG_H
 #define GOT_CLIENTLOG_H
 
-#include "sysincl.h"
+#include <inttypes.h>
+
 #include "reports.h"
 
 typedef unsigned long CLG_IP_Addr;
Index: chrony-1.21-ptx/cmdmon.c
===================================================================
--- chrony-1.21-ptx.orig/cmdmon.c
+++ chrony-1.21-ptx/cmdmon.c
@@ -28,7 +28,12 @@
   Command and monitoring module in the main program
   */
 
-#include "sysincl.h"
+#include <time.h>
+#include <errno.h>
+#include <unistd.h>
+#include <linux/stddef.h>
+#include <sys/types.h>
+#include <sys/socket.h>
 
 #include "cmdmon.h"
 #include "candm.h"
Index: chrony-1.21-ptx/cmdparse.c
===================================================================
--- chrony-1.21-ptx.orig/cmdparse.c
+++ chrony-1.21-ptx/cmdparse.c
@@ -30,7 +30,8 @@
 
   */
 
-#include "sysincl.h"
+#include <stdio.h>
+#include <string.h>
 
 #include "cmdparse.h"
 #include "nameserv.h"
Index: chrony-1.21-ptx/conf.c
===================================================================
--- chrony-1.21-ptx.orig/conf.c
+++ chrony-1.21-ptx/conf.c
@@ -39,7 +39,11 @@
 
   */
 
-#include "sysincl.h"
+#include <stdio.h>
+#include <values.h>
+#include <string.h>
+#include <ctype.h>
+#include <assert.h>
 
 #include "conf.h"
 #include "ntp_sources.h"
Index: chrony-1.21-ptx/getdate.c
===================================================================
--- chrony-1.21-ptx.orig/getdate.c
+++ chrony-1.21-ptx/getdate.c
@@ -36,7 +36,7 @@
 */
 
 #ifdef HAVE_CONFIG_H
-# include <config.h>
+# include <chrony_config.h>
 # ifdef FORCE_ALLOCA_H
 #  include <alloca.h>
 # endif
Index: chrony-1.21-ptx/local.h
===================================================================
--- chrony-1.21-ptx.orig/local.h
+++ chrony-1.21-ptx/local.h
@@ -33,7 +33,7 @@
 #ifndef GOT_LOCAL_H
 #define GOT_LOCAL_H
 
-#include "sysincl.h"
+#include <sys/time.h>
 
 /* Read the system clock.  This is analogous to gettimeofday(),
    but with the timezone information ignored */
Index: chrony-1.21-ptx/logging.c
===================================================================
--- chrony-1.21-ptx.orig/logging.c
+++ chrony-1.21-ptx/logging.c
@@ -28,11 +28,19 @@
   Module to handle logging of diagnostic information
   */
 
-#include "sysincl.h"
+#include <chrony_config.h>
+
+#include <stdlib.h>
+#include <syslog.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <time.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
 
 #include "main.h"
 #include "logging.h"
-#include "version.h"
 
 /* ================================================== */
 /* Flag indicating we have initialised */
@@ -205,7 +213,7 @@ LOG_GoDaemon(void)
 
       openlog("chronyd", LOG_PID, LOG_DAEMON);
 
-      LOG(LOGS_INFO, LOGF_Logging, "chronyd version %s starting", PROGRAM_VERSION_STRING);
+      LOG(LOGS_INFO, LOGF_Logging, "chronyd version %s starting", PACKAGE_VERSION);
 
     }
   }
Index: chrony-1.21-ptx/main.c
===================================================================
--- chrony-1.21-ptx.orig/main.c
+++ chrony-1.21-ptx/main.c
@@ -28,7 +28,13 @@
   The main program
   */
 
-#include "sysincl.h"
+#include <stdlib.h>
+#include <signal.h>
+#include <string.h>
+
+/* we need getsid() */
+#define __USE_XOPEN_EXTENDED
+#include <unistd.h>
 
 #include "main.h"
 #include "sched.h"
@@ -46,7 +52,6 @@
 #include "keys.h"
 #include "acquire.h"
 #include "manual.h"
-#include "version.h"
 #include "rtc.h"
 #include "clientlog.h"
 #include "broadcast.h"
@@ -224,7 +229,7 @@ int main
       do_init_rtc = 1;
     } else if (!strcmp("-v", *argv) || !strcmp("--version",*argv)) {
       /* This write to the terminal is OK, it comes before we turn into a daemon */
-      printf("chronyd (chrony) version %s\n", PROGRAM_VERSION_STRING);
+      printf("chronyd (chrony) version %s\n", PACKAGE_VERSION);
       exit(0);
     } else if (!strcmp("-d", *argv)) {
       debug = 1;
Index: chrony-1.21-ptx/manual.h
===================================================================
--- chrony-1.21-ptx.orig/manual.h
+++ chrony-1.21-ptx/manual.h
@@ -32,7 +32,6 @@
 #ifndef GOT_MANUAL_H
 #define GOT_MANUAL_H
 
-#include "sysincl.h"
 #include "reports.h"
 
 extern void MNL_Initialise(void);
Index: chrony-1.21-ptx/md5.h
===================================================================
--- chrony-1.21-ptx.orig/md5.h
+++ chrony-1.21-ptx/md5.h
@@ -32,10 +32,12 @@
  ***********************************************************************
  */
 
-#ifdef HAS_STDINT_H
+#include <chrony_config.h>
+
+#ifdef HAVE_STDINT_H
 #include <stdint.h>
-#elif defined(HAS_INTTYPES_H)
-#include <inttypes.h>
+#elif defined(HAVE_INTTYPES_H)
+ #include <inttypes.h>
 #endif
 
 /* typedef a 32-bit type */
Index: chrony-1.21-ptx/mkdirpp.c
===================================================================
--- chrony-1.21-ptx.orig/mkdirpp.c
+++ chrony-1.21-ptx/mkdirpp.c
@@ -30,7 +30,10 @@
 
   */
 
-#include "sysincl.h"
+#include <sys/stat.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
 
 #include "mkdirpp.h"
 
Index: chrony-1.21-ptx/nameserv.c
===================================================================
--- chrony-1.21-ptx.orig/nameserv.c
+++ chrony-1.21-ptx/nameserv.c
@@ -29,7 +29,9 @@
 
   */
 
-#include "sysincl.h"
+#include <netdb.h>
+#include <string.h>
+#include <stdio.h>
 
 #include "nameserv.h"
 
Index: chrony-1.21-ptx/ntp.h
===================================================================
--- chrony-1.21-ptx.orig/ntp.h
+++ chrony-1.21-ptx/ntp.h
@@ -31,9 +31,12 @@
 #ifndef GOT_NTP_H
 #define GOT_NTP_H
 
-#ifdef HAS_STDINT_H
+#include <chrony_config.h>
+#include <netinet/in.h>
+
+#ifdef HAVE_STDINT_H
 #include <stdint.h>
-#elif defined(HAS_INTTYPES_H)
+#elif defined(HAVE_INTTYPES_H)
 #include <inttypes.h>
 #endif
 
Index: chrony-1.21-ptx/ntp_core.c
===================================================================
--- chrony-1.21-ptx.orig/ntp_core.c
+++ chrony-1.21-ptx/ntp_core.c
@@ -28,7 +28,10 @@
   Core NTP protocol engine
   */
 
-#include "sysincl.h"
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+#include <stddef.h>
 
 #include "ntp_core.h"
 #include "ntp_io.h"
Index: chrony-1.21-ptx/ntp_core.h
===================================================================
--- chrony-1.21-ptx.orig/ntp_core.h
+++ chrony-1.21-ptx/ntp_core.h
@@ -31,8 +31,6 @@
 #ifndef GOT_NTP_CORE_H
 #define GOT_NTP_CORE_H
 
-#include "sysincl.h"
-
 #include "addressing.h"
 #include "srcparams.h"
 #include "ntp.h"
Index: chrony-1.21-ptx/ntp_io.c
===================================================================
--- chrony-1.21-ptx.orig/ntp_io.c
+++ chrony-1.21-ptx/ntp_io.c
@@ -28,7 +28,14 @@
   This file deals with the IO aspects of reading and writing NTP packets
   */
 
-#include "sysincl.h"
+#include <assert.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/socket.h>
 
 #include "ntp_io.h"
 #include "ntp_core.h"
Index: chrony-1.21-ptx/ntp_sources.c
===================================================================
--- chrony-1.21-ptx.orig/ntp_sources.c
+++ chrony-1.21-ptx/ntp_sources.c
@@ -30,7 +30,7 @@
 
   */
 
-#include "sysincl.h"
+#include <assert.h>
 
 #include "ntp_sources.h"
 #include "ntp_core.h"
Index: chrony-1.21-ptx/pktlength.c
===================================================================
--- chrony-1.21-ptx.orig/pktlength.c
+++ chrony-1.21-ptx/pktlength.c
@@ -30,7 +30,9 @@
   integer endianness within the structures.
 
   */
-#include "sysincl.h"
+
+#include <stddef.h>
+#include <assert.h>
 
 #include "util.h"
 #include "pktlength.h"
Index: chrony-1.21-ptx/reference.c
===================================================================
--- chrony-1.21-ptx.orig/reference.c
+++ chrony-1.21-ptx/reference.c
@@ -28,7 +28,13 @@
   This module keeps track of the source which we are claiming to be
   our reference, for the purposes of generating outgoing NTP packets */
 
-#include "sysincl.h"
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+#include <assert.h>
 
 #include "memory.h"
 #include "reference.h"
Index: chrony-1.21-ptx/reference.h
===================================================================
--- chrony-1.21-ptx.orig/reference.h
+++ chrony-1.21-ptx/reference.h
@@ -33,8 +33,6 @@
 #ifndef GOT_REFERENCE_H
 #define GOT_REFERENCE_H
 
-#include "sysincl.h"
-
 #include "ntp.h"
 #include "reports.h"
 
Index: chrony-1.21-ptx/reports.h
===================================================================
--- chrony-1.21-ptx.orig/reports.h
+++ chrony-1.21-ptx/reports.h
@@ -31,7 +31,7 @@
 #ifndef GOT_REPORTS_H
 #define GOT_REPORTS_H
 
-#include "sysincl.h"
+#include <sys/time.h>
 
 #define REPORT_INVALID_OFFSET 0x80000000
 
Index: chrony-1.21-ptx/rtc.c
===================================================================
--- chrony-1.21-ptx.orig/rtc.c
+++ chrony-1.21-ptx/rtc.c
@@ -27,15 +27,17 @@
 
   */
 
-#include "sysincl.h"
+#include <chrony_config.h>
+
+#include <stdio.h>
 
 #include "rtc.h"
 #include "logging.h"
 #include "conf.h"
 
-#if defined LINUX
+#if defined HAVE_LINUX
 #include "rtc_linux.h"
-#endif /* defined LINUX */
+#endif
 
 /* ================================================== */
 
@@ -53,7 +55,7 @@ static struct {
   void (*cycle_logfile)(void);
 } driver =
 {
-#if defined LINUX
+#if defined HAVE_LINUX
   RTC_Linux_Initialise,
   RTC_Linux_Finalise,
   RTC_Linux_TimePreInit,
Index: chrony-1.21-ptx/rtc_linux.c
===================================================================
--- chrony-1.21-ptx.orig/rtc_linux.c
+++ chrony-1.21-ptx/rtc_linux.c
@@ -30,7 +30,11 @@
 
   */
 
-#if defined LINUX
+#include <chrony_config.h>
+
+#include <sys/stat.h>
+
+#if defined HAVE_LINUX
 
 #ifdef sparc
 #define __KERNEL__
@@ -1172,4 +1176,4 @@ RTC_Linux_CycleLogFile(void)
 
 /* ================================================== */
 
-#endif /* defined LINUX */
+#endif /* defined HAVE_LINUX */
Index: chrony-1.21-ptx/rtc_linux.h
===================================================================
--- chrony-1.21-ptx.orig/rtc_linux.h
+++ chrony-1.21-ptx/rtc_linux.h
@@ -30,9 +30,11 @@
 #ifndef _GOT_RTC_LINUX_H
 #define _GOT_RTC_LINUX_H
 
+#include <chrony_config.h>
+
 #include "reports.h"
 
-#if defined LINUX
+#if defined HAVE_LINUX
 
 extern int RTC_Linux_Initialise(void);
 extern void RTC_Linux_Finalise(void);
@@ -48,6 +50,6 @@ extern int RTC_Linux_Trim(void);
 
 extern void RTC_Linux_CycleLogFile(void);
 
-#endif /* defined LINUX */
+#endif /* defined HAVE_LINUX */
 
 #endif /* _GOT_RTC_LINUX_H */
Index: chrony-1.21-ptx/sched.c
===================================================================
--- chrony-1.21-ptx.orig/sched.c
+++ chrony-1.21-ptx/sched.c
@@ -29,7 +29,8 @@
 
   */
 
-#include "sysincl.h"
+#include <assert.h>
+#include <string.h>
 
 #include "sched.h"
 #include "memory.h"
Index: chrony-1.21-ptx/sched.h
===================================================================
--- chrony-1.21-ptx.orig/sched.h
+++ chrony-1.21-ptx/sched.h
@@ -31,7 +31,7 @@
 #ifndef GOT_SCHED_H
 #define GOT_SCHED_H
 
-#include "sysincl.h"
+#include <sys/time.h>
 
 typedef unsigned long SCH_TimeoutID;
 
Index: chrony-1.21-ptx/sources.c
===================================================================
--- chrony-1.21-ptx.orig/sources.c
+++ chrony-1.21-ptx/sources.c
@@ -31,7 +31,8 @@
 
   */
 
-#include "sysincl.h"
+#include <math.h>
+#include <string.h>
 
 #include "sources.h"
 #include "sourcestats.h"
Index: chrony-1.21-ptx/sources.h
===================================================================
--- chrony-1.21-ptx.orig/sources.h
+++ chrony-1.21-ptx/sources.h
@@ -33,7 +33,7 @@
 #ifndef GOT_SOURCES_H
 #define GOT_SOURCES_H
 
-#include "sysincl.h"
+#include <string.h>
 
 #include "ntp.h"
 #include "reports.h"
Index: chrony-1.21-ptx/sourcestats.c
===================================================================
--- chrony-1.21-ptx.orig/sourcestats.c
+++ chrony-1.21-ptx/sourcestats.c
@@ -29,7 +29,9 @@
   analysis on the samples obtained from the sources,
   to determined frequencies and error bounds. */
 
-#include "sysincl.h"
+#include <string.h>
+#include <math.h>
+#include <values.h>
 
 #include "sourcestats.h"
 #include "memory.h"
Index: chrony-1.21-ptx/sourcestats.h
===================================================================
--- chrony-1.21-ptx.orig/sourcestats.h
+++ chrony-1.21-ptx/sourcestats.h
@@ -31,7 +31,7 @@
 #ifndef GOT_SOURCESTATS_H
 #define GOT_SOURCESTATS_H
 
-#include "sysincl.h"
+#include <stdio.h>
 
 #include "reports.h"
 
Index: chrony-1.21-ptx/sys.c
===================================================================
--- chrony-1.21-ptx.orig/sys.c
+++ chrony-1.21-ptx/sys.c
@@ -29,9 +29,11 @@
   in the various operating-system specific modules
   */
 
+#include <chrony_config.h>
+
 #include "sys.h"
 
-#if defined (LINUX)
+#if defined (HAVE_LINUX)
 #include "sys_linux.h"
 #endif
 
@@ -53,7 +55,7 @@ void
 SYS_Initialise(void)
 {
 
-#if defined(LINUX)
+#if defined(HAVE_LINUX)
   SYS_Linux_Initialise();
 #endif
 
@@ -77,7 +79,7 @@ void
 SYS_Finalise(void)
 {
   
-#if defined(LINUX)
+#if defined(HAVE_LINUX)
   SYS_Linux_Finalise();
 #endif
 
Index: chrony-1.21-ptx/sys_linux.c
===================================================================
--- chrony-1.21-ptx.orig/sys_linux.c
+++ chrony-1.21-ptx/sys_linux.c
@@ -29,7 +29,9 @@
 
   */
 
-#ifdef LINUX
+#include <chrony_config.h>
+
+#ifdef HAVE_LINUX
 
 #include <sys/time.h>
 #include <stdio.h>
@@ -831,7 +833,7 @@ SYS_Linux_GetKernelVersion(int *major, i
 
 /* ================================================== */
 
-#endif /* LINUX */
+#endif /* HAVE_LINUX */
 
 /* vim:ts=8
  * */
Index: chrony-1.21-ptx/util.c
===================================================================
--- chrony-1.21-ptx.orig/util.c
+++ chrony-1.21-ptx/util.c
@@ -28,10 +28,12 @@
   Various utility functions
   */
 
-#include "sysincl.h"
+#include <time.h>
+#include <stdio.h>
 
 #include "util.h"
 #include "logging.h"
+#include "clientlog.h"
 
 /* ================================================== */
 
Index: chrony-1.21-ptx/util.h
===================================================================
--- chrony-1.21-ptx.orig/util.h
+++ chrony-1.21-ptx/util.h
@@ -31,7 +31,7 @@
 #ifndef GOT_UTIL_H
 #define GOT_UTIL_H
 
-#include "sysincl.h"
+#include <chrony_config.h>
 
 #include "ntp.h"
 
@@ -87,7 +87,7 @@ extern void UTI_Int64ToTimeval(NTP_int64
 
 /* Like assert(0) */
 
-#if defined(LINUX) && defined(__alpha__)
+#if defined(HAVE_LINUX) && defined(__alpha__)
 #define CROAK(message) assert(0) /* Added JGH Feb 24 2001  FIXME */
 #else
 extern int croak(const char *file, int line, const char *msg);
Index: chrony-1.21-ptx/wrap_adjtimex.c
===================================================================
--- chrony-1.21-ptx.orig/wrap_adjtimex.c
+++ chrony-1.21-ptx/wrap_adjtimex.c
@@ -32,7 +32,9 @@
 
   */
 
-#ifdef LINUX
+#include <chrony_config.h>
+
+#ifdef HAVE_LINUX
 
 #define _LOOSE_KERNEL_NAMES
 
