Index: chrony-1.21/Makefile.am
===================================================================
--- /dev/null
+++ chrony-1.21/Makefile.am
@@ -0,0 +1,102 @@
+#
+# chronyd/chronyc - Programs for keeping computer clocks accurate.
+#
+# Copyright (C) Richard P. Curnow  1997-2003
+#
+# Autotoolized by Robert Schwebel <r.schwebel@pengutronix.de>
+#              and Marc Kleine-Budde <mkl@pengutronix.de>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
+#
+
+sbin_PROGRAMS =  chronyd
+bin_PROGRAMS = chronyc
+
+AM_CPPFLAGS = \
+	-I$(top_srcdir)/include \
+	-I$(top_builddir)/include
+
+chronyd_SOURCES = util.c sched.c regress.c local.c \
+	sys.c main.c ntp_io.c ntp_core.c ntp_sources.c \
+	sources.c sourcestats.c reference.c \
+	logging.c conf.c cmdmon.c md5.c keys.c \
+	nameserv.c acquire.c manual.c addrfilt.c \
+	cmdparse.c mkdirpp.c rtc.c pktlength.c clientlog.c \
+	broadcast.c
+
+chronyd_SOURCES += $(CHRONYD_EXTRA_SOURCES)
+
+# if CONFIG_SUN4
+# chronyd_SOURCES += \
+# 	sys_sunos.c \
+# 	strerror.c
+# endif
+# if CONFIG_SUN5
+# chronyd_SOURCES += \
+# 	sys_solaris.c
+# endif
+if CONFIG_LINUX
+chronyd_SOURCES += \
+	sys_linux.c \
+	wrap_adjtimex.c \
+	rtc_linux.c
+endif
+# if CONFIG_BSD386
+# chronyd_SOURCES += \
+# 	sys_sunos.o \
+# 	strerror.o
+# endif
+# if CONFIG_NETBSD
+# chronyd_SOURCES += \
+# 	sys_netbsd.c
+# endif
+# if CONFIG_SUNOS_I386
+# chronyd_SOURCES += \
+# 	sys_netbsd.c
+# endif
+# if CONFIG_CYGWIN32
+# chronyd_SOURCES += \
+# 	sys_winnt.c
+# endif
+
+chronyc_SOURCES = client.c md5.c nameserv.c getdate.c cmdparse.c \
+	pktlength.c
+
+noinst_HEADERS = \
+	acquire.h chrony_config.h conf.h localp.h memory.h ntp_io.h \
+	reports.h sourcestats.h sys_solaris.h addressing.h chrony_timex.h \
+	getdate.h logging.h mkdirpp.h ntp_sources.h rtc.h srcparams.h \
+	sys_sunos.h addrfilt.h clientlog.h io_linux.h main.h nameserv.h \
+	pktlength.h rtc_linux.h sys.h util.h broadcast.h cmdmon.h keys.h \
+	manual.h ntp.h reference.h sched.h sys_linux.h wrap_adjtimex.h \
+	candm.h cmdparse.h local.h md5.h ntp_core.h regress.h sources.h \
+	sys_netbsd.h
+
+MAINTAINERCLEANFILES = \
+	configure \
+	GNUmakefile.in \
+	aclocal.m4 \
+	compile \
+	config.guess \
+	config.sub \
+	depcomp \
+	install-sh \
+	ltmain.sh \
+	mdate-sh \
+	missing \
+	texinfo.tex \
+	libtool.m4 \
+	ltoptions.m4 \
+	ltsugar.m4 \
+	ltversion.m4
Index: chrony-1.21/configure.ac
===================================================================
--- /dev/null
+++ chrony-1.21/configure.ac
@@ -0,0 +1,147 @@
+# -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+AC_PREREQ(2.59)
+
+AC_INIT([chrony], [1.21], [rc@rc0.org.uk])
+AC_CONFIG_HEADERS([chrony_config.h])
+AC_CONFIG_SRCDIR([main.c])
+AC_CONFIG_MACRO_DIR([.])
+AC_CONFIG_AUX_DIR([.])
+AC_CANONICAL_BUILD
+AC_CANONICAL_HOST
+
+CFLAGS="${CFLAGS} -Wall"
+
+
+
+#
+# Checks for programs.
+#
+AC_PROG_CC
+AM_MISSING_PROG(PERL, perl, $missing_dir)
+# libtool, old:
+AC_LIBTOOL_WIN32_DLL
+#AC_LIBTOOL_TAGS([])
+AC_PROG_LIBTOOL
+# libtool, new:
+# LT_INIT(win32-dll)
+
+AM_INIT_AUTOMAKE([foreign no-exeext dist-bzip2])
+
+
+#
+#
+#
+case "$host" in
+     "*-linux-*")
+	AC_DEFINE([HAVE_LINUX], [], [We compile for a Linux host])
+	;;
+	*)
+	;;
+esac
+AM_CONDITIONAL(CONFIG_LINUX, [test "$host" = "*linux*"])
+
+
+#
+# Checks for libraries.
+#
+
+
+
+#
+# Checks for header files.
+#
+AC_FUNC_ALLOCA
+AC_HEADER_STDC
+AC_CHECK_HEADERS([\
+	fcntl.h \
+	inttypes.h \
+	malloc.h \
+	netdb.h \
+	nlist.h \
+	stddef.h \
+	stdint.h \
+	stdlib.h \
+	string.h \
+	syslog.h \
+	unistd.h \
+	values.h \
+	\
+	netinet/in.h \
+	\
+	sys/ioctl.h \
+	sys/socket.h \
+	sys/time.h \
+	])
+
+
+#
+# Checks for typedefs, structures, and compiler characteristics.
+#
+AC_C_CONST
+AC_C_INLINE
+AC_HEADER_TIME
+AC_STRUCT_TM
+AC_C_VOLATILE
+
+
+#
+# Checks for library functions.
+#
+AC_FUNC_CHOWN
+AC_FUNC_FORK
+AC_PROG_GCC_TRADITIONAL
+AC_FUNC_MEMCMP
+AC_FUNC_MKTIME
+AC_FUNC_SELECT_ARGTYPES
+AC_TYPE_SIGNAL
+AC_FUNC_STAT
+AC_FUNC_STRFTIME
+AC_CHECK_FUNCS([\
+	bzero \
+	gethostbyaddr \
+	gethostbyname \
+	gethostname \
+	getpass \
+	gettimeofday \
+	memmove \
+	memset \
+	mkdir \
+	select \
+	socket \
+	sqrt \
+	strchr \
+	strerror \
+	strncasecmp \
+	uname\
+	])
+
+AC_SEARCH_LIBS(sqrt, m, [],
+        [AC_MSG_ERROR([unable do identify library containing sqrt])])
+
+
+#
+# Debugging
+#
+AC_MSG_CHECKING([whether to enable debugging])
+AC_ARG_ENABLE(debug,
+    AS_HELP_STRING([--enable-debug], [enable debugging @<:@default=no@:>@]),
+	[case "$enableval" in
+	y | yes) CONFIG_DEBUG=yes ;;
+        *) CONFIG_DEBUG=no ;;
+    esac],
+    [CONFIG_DEBUG=no])
+AC_MSG_RESULT([${CONFIG_DEBUG}])
+if test "${CONFIG_DEBUG}" = "yes"; then
+    CFLAGS="${CFLAGS} -g -O1"
+    AC_DEFINE(DEBUG, 1, [debugging])
+else
+    CFLAGS="${CFLAGS} -O2"
+fi
+
+
+AC_CONFIG_FILES([
+	Makefile
+	])
+AC_OUTPUT
+
Index: chrony-1.21/autogen.sh
===================================================================
--- /dev/null
+++ chrony-1.21/autogen.sh
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+#
+# usage:
+#
+# banner <target name>
+#
+banner() {
+
+	echo
+	TG=`echo $1 | sed -e "s,/.*/,,g"`
+	LINE=`echo $TG |sed -e "s/./-/g"`
+	echo $LINE
+	echo $TG
+	echo $LINE
+	echo
+}
+
+
+ACLOCAL=${ACLOCAL:=aclocal}
+AUTOHEADER=${AUTOHEADER:=autoheader}
+AUTOMAKE=${AUTOMAKE:=automake}
+AUTOCONF=${AUTOCONF:=autoconf}
+
+$ACLOCAL --version | \
+   awk -vPROG="aclocal" -vVERS=1.7\
+   '{if ($1 == PROG) {gsub ("-.*","",$4); if ($4 < VERS) print PROG" < version "VERS"\nThis may result in errors\n"}}'
+
+$AUTOMAKE --version | \
+   awk -vPROG="automake" -vVERS=1.7\
+   '{if ($1 == PROG) {gsub ("-.*","",$4); if ($4 < VERS) print PROG" < version "VERS"\nThis may result in errors\n"}}'
+
+
+banner "running libtoolize"
+libtoolize --force || exit
+
+banner "running aclocal"
+$ACLOCAL -I . || exit
+
+banner "running autoheader"
+$AUTOHEADER || exit
+
+banner "running automake"
+$AUTOMAKE --gnu --add-missing -Wall || exit
+
+banner "running autoconf"
+$AUTOCONF -Wall || exit
+
+banner "Finished"
Index: chrony-1.21/chrony_config.h.in
===================================================================
--- /dev/null
+++ chrony-1.21/chrony_config.h.in
@@ -0,0 +1,243 @@
+/* chrony_config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* Define to one of `_getb67', `GETB67', `getb67' for Cray-2 and Cray-YMP
+   systems. This function is required for `alloca.c' support on those systems.
+   */
+#undef CRAY_STACKSEG_END
+
+/* Define to 1 if using `alloca.c'. */
+#undef C_ALLOCA
+
+/* debugging */
+#undef DEBUG
+
+/* Define to 1 if you have the `alarm' function. */
+#undef HAVE_ALARM
+
+/* Define to 1 if you have `alloca', as a function or macro. */
+#undef HAVE_ALLOCA
+
+/* Define to 1 if you have <alloca.h> and it should be used (not on Ultrix).
+   */
+#undef HAVE_ALLOCA_H
+
+/* Define to 1 if you have the `bzero' function. */
+#undef HAVE_BZERO
+
+/* Define to 1 if your system has a working `chown' function. */
+#undef HAVE_CHOWN
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#undef HAVE_DLFCN_H
+
+/* Define to 1 if you have the <fcntl.h> header file. */
+#undef HAVE_FCNTL_H
+
+/* Define to 1 if you have the `fork' function. */
+#undef HAVE_FORK
+
+/* Define to 1 if you have the `gethostbyaddr' function. */
+#undef HAVE_GETHOSTBYADDR
+
+/* Define to 1 if you have the `gethostbyname' function. */
+#undef HAVE_GETHOSTBYNAME
+
+/* Define to 1 if you have the `gethostname' function. */
+#undef HAVE_GETHOSTNAME
+
+/* Define to 1 if you have the `getpass' function. */
+#undef HAVE_GETPASS
+
+/* Define to 1 if you have the `gettimeofday' function. */
+#undef HAVE_GETTIMEOFDAY
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#undef HAVE_INTTYPES_H
+
+/* We compile for a Linux host */
+#undef HAVE_LINUX
+
+/* Define to 1 if you have the <malloc.h> header file. */
+#undef HAVE_MALLOC_H
+
+/* Define to 1 if you have the `memmove' function. */
+#undef HAVE_MEMMOVE
+
+/* Define to 1 if you have the <memory.h> header file. */
+#undef HAVE_MEMORY_H
+
+/* Define to 1 if you have the `memset' function. */
+#undef HAVE_MEMSET
+
+/* Define to 1 if you have the `mkdir' function. */
+#undef HAVE_MKDIR
+
+/* Define to 1 if you have the <netdb.h> header file. */
+#undef HAVE_NETDB_H
+
+/* Define to 1 if you have the <netinet/in.h> header file. */
+#undef HAVE_NETINET_IN_H
+
+/* Define to 1 if you have the <nlist.h> header file. */
+#undef HAVE_NLIST_H
+
+/* Define to 1 if you have the `select' function. */
+#undef HAVE_SELECT
+
+/* Define to 1 if you have the `socket' function. */
+#undef HAVE_SOCKET
+
+/* Define to 1 if you have the `sqrt' function. */
+#undef HAVE_SQRT
+
+/* Define to 1 if `stat' has the bug that it succeeds when given the
+   zero-length file name argument. */
+#undef HAVE_STAT_EMPTY_STRING_BUG
+
+/* Define to 1 if you have the <stddef.h> header file. */
+#undef HAVE_STDDEF_H
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#undef HAVE_STDINT_H
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#undef HAVE_STDLIB_H
+
+/* Define to 1 if you have the `strchr' function. */
+#undef HAVE_STRCHR
+
+/* Define to 1 if you have the `strerror' function. */
+#undef HAVE_STRERROR
+
+/* Define to 1 if you have the `strftime' function. */
+#undef HAVE_STRFTIME
+
+/* Define to 1 if you have the <strings.h> header file. */
+#undef HAVE_STRINGS_H
+
+/* Define to 1 if you have the <string.h> header file. */
+#undef HAVE_STRING_H
+
+/* Define to 1 if you have the `strncasecmp' function. */
+#undef HAVE_STRNCASECMP
+
+/* Define to 1 if you have the <syslog.h> header file. */
+#undef HAVE_SYSLOG_H
+
+/* Define to 1 if you have the <sys/ioctl.h> header file. */
+#undef HAVE_SYS_IOCTL_H
+
+/* Define to 1 if you have the <sys/select.h> header file. */
+#undef HAVE_SYS_SELECT_H
+
+/* Define to 1 if you have the <sys/socket.h> header file. */
+#undef HAVE_SYS_SOCKET_H
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#undef HAVE_SYS_STAT_H
+
+/* Define to 1 if you have the <sys/time.h> header file. */
+#undef HAVE_SYS_TIME_H
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#undef HAVE_SYS_TYPES_H
+
+/* Define to 1 if you have the `uname' function. */
+#undef HAVE_UNAME
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#undef HAVE_UNISTD_H
+
+/* Define to 1 if you have the <values.h> header file. */
+#undef HAVE_VALUES_H
+
+/* Define to 1 if you have the `vfork' function. */
+#undef HAVE_VFORK
+
+/* Define to 1 if you have the <vfork.h> header file. */
+#undef HAVE_VFORK_H
+
+/* Define to 1 if `fork' works. */
+#undef HAVE_WORKING_FORK
+
+/* Define to 1 if `vfork' works. */
+#undef HAVE_WORKING_VFORK
+
+/* Define to 1 if `lstat' dereferences a symlink specified with a trailing
+   slash. */
+#undef LSTAT_FOLLOWS_SLASHED_SYMLINK
+
+/* Name of package */
+#undef PACKAGE
+
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* Define as the return type of signal handlers (`int' or `void'). */
+#undef RETSIGTYPE
+
+/* Define to the type of arg 1 for `select'. */
+#undef SELECT_TYPE_ARG1
+
+/* Define to the type of args 2, 3 and 4 for `select'. */
+#undef SELECT_TYPE_ARG234
+
+/* Define to the type of arg 5 for `select'. */
+#undef SELECT_TYPE_ARG5
+
+/* If using the C implementation of alloca, define if you know the
+   direction of stack growth for your system; otherwise it will be
+   automatically deduced at run-time.
+	STACK_DIRECTION > 0 => grows toward higher addresses
+	STACK_DIRECTION < 0 => grows toward lower addresses
+	STACK_DIRECTION = 0 => direction of growth unknown */
+#undef STACK_DIRECTION
+
+/* Define to 1 if you have the ANSI C header files. */
+#undef STDC_HEADERS
+
+/* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
+#undef TIME_WITH_SYS_TIME
+
+/* Define to 1 if your <sys/time.h> declares `struct tm'. */
+#undef TM_IN_SYS_TIME
+
+/* Version number of package */
+#undef VERSION
+
+/* Define to empty if `const' does not conform to ANSI C. */
+#undef const
+
+/* Define to `int' if <sys/types.h> doesn't define. */
+#undef gid_t
+
+/* Define to `__inline__' or `__inline' if that's what the C compiler
+   calls it, or to nothing if 'inline' is not supported under any name.  */
+#ifndef __cplusplus
+#undef inline
+#endif
+
+/* Define to `int' if <sys/types.h> does not define. */
+#undef pid_t
+
+/* Define to `int' if <sys/types.h> doesn't define. */
+#undef uid_t
+
+/* Define as `fork' if `vfork' does not work. */
+#undef vfork
+
+/* Define to empty if the keyword `volatile' does not work. Warning: valid
+   code using `volatile' can become incorrect without. Disable with care. */
+#undef volatile
