diff -urN ACE-5.5.2+TAO-1.5.2/apps/Makefile.in ACE-5.5.2+TAO-1.5.2-autogen/apps/Makefile.in
--- ACE-5.5.2+TAO-1.5.2/apps/Makefile.in	2006-07-09 14:39:35.000000000 +0200
+++ ACE-5.5.2+TAO-1.5.2-autogen/apps/Makefile.in	2006-07-30 22:15:10.000000000 +0200
@@ -63,7 +63,7 @@
 	uninstall-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = Gateway JAWS JAWS2 JAWS3 drwho gperf mkcsregdb soreduce
+DIST_SUBDIRS = Gateway JAWS JAWS2 JAWS3 drwho mkcsregdb soreduce gperf
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACE_BETA = @ACE_BETA@
 ACE_CROSS_COMPILED_FALSE = @ACE_CROSS_COMPILED_FALSE@
@@ -243,6 +243,7 @@
 FFLAGS = @FFLAGS@
 FLTKCONFIG = @FLTKCONFIG@
 GPERF = @GPERF@
+GREP = @GREP@
 INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
@@ -282,22 +283,17 @@
 TCLCONFIG = @TCLCONFIG@
 TKCONFIG = @TKCONFIG@
 VERSION = @VERSION@
+XMKMF = @XMKMF@
 XTREACTOR_TEST_XLIBS = @XTREACTOR_TEST_XLIBS@
 X_CFLAGS = @X_CFLAGS@
 X_EXTRA_LIBS = @X_EXTRA_LIBS@
 X_LIBS = @X_LIBS@
 X_PRE_LIBS = @X_PRE_LIBS@
 YACC = @YACC@
-ac_ct_AR = @ac_ct_AR@
-ac_ct_AS = @ac_ct_AS@
+YFLAGS = @YFLAGS@
 ac_ct_CC = @ac_ct_CC@
 ac_ct_CXX = @ac_ct_CXX@
-ac_ct_DLLTOOL = @ac_ct_DLLTOOL@
 ac_ct_F77 = @ac_ct_F77@
-ac_ct_OBJDUMP = @ac_ct_OBJDUMP@
-ac_ct_RANLIB = @ac_ct_RANLIB@
-ac_ct_STRIP = @ac_ct_STRIP@
-ac_pt_PKG_CONFIG = @ac_pt_PKG_CONFIG@
 am__fastdepCC_FALSE = @am__fastdepCC_FALSE@
 am__fastdepCC_TRUE = @am__fastdepCC_TRUE@
 am__fastdepCXX_FALSE = @am__fastdepCXX_FALSE@
@@ -314,23 +310,30 @@
 build_os = @build_os@
 build_vendor = @build_vendor@
 datadir = @datadir@
+datarootdir = @datarootdir@
+docdir = @docdir@
+dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
 host_alias = @host_alias@
 host_cpu = @host_cpu@
 host_os = @host_os@
 host_vendor = @host_vendor@
+htmldir = @htmldir@
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
 libdir = @libdir@
 libexecdir = @libexecdir@
+localedir = @localedir@
 localstatedir = @localstatedir@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
 oldincludedir = @oldincludedir@
+pdfdir = @pdfdir@
 prefix = @prefix@
 program_transform_name = @program_transform_name@
+psdir = @psdir@
 sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 subdirs = @subdirs@
@@ -347,7 +350,6 @@
         JAWS2 \
         JAWS3 \
         drwho \
-        gperf \
         mkcsregdb \
         soreduce \
         $(OPTDIRS)
