diff -urN /home/robert/embedded/ptxdist-0.3.20pre/build/tcp_wrappers_7.6/Makefile tcp_wrappers_7.6/Makefile
--- /home/robert/embedded/ptxdist-0.3.20pre/build/tcp_wrappers_7.6/Makefile	1997-03-21 19:27:21.000000000 +0100
+++ tcp_wrappers_7.6/Makefile	2003-02-04 16:18:35.000000000 +0100
@@ -44,7 +44,7 @@
 #REAL_DAEMON_DIR=/usr/etc
 #
 # SysV.4 Solaris 2.x OSF AIX
-#REAL_DAEMON_DIR=/usr/sbin
+REAL_DAEMON_DIR=/usr/sbin
 #
 # BSD 4.4
 #REAL_DAEMON_DIR=/usr/libexec
diff -urN /home/robert/embedded/ptxdist-0.3.20pre/build/tcp_wrappers_7.6/percent_m.c tcp_wrappers_7.6/percent_m.c
--- /home/robert/embedded/ptxdist-0.3.20pre/build/tcp_wrappers_7.6/percent_m.c	1994-12-28 17:42:37.000000000 +0100
+++ tcp_wrappers_7.6/percent_m.c	2003-02-04 16:20:11.000000000 +0100
@@ -14,7 +14,7 @@
 
 extern int errno;
 #ifndef SYS_ERRLIST_DEFINED
-extern char *sys_errlist[];
+extern __const char *__const sys_errlist[];
 extern int sys_nerr;
 #endif
 
