---
 Makefile    |    2 +-
 percent_m.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: tcp_wrappers_7.6/Makefile
===================================================================
--- tcp_wrappers_7.6.orig/Makefile
+++ tcp_wrappers_7.6/Makefile
@@ -44,7 +44,7 @@ what:
 #REAL_DAEMON_DIR=/usr/etc
 #
 # SysV.4 Solaris 2.x OSF AIX
-#REAL_DAEMON_DIR=/usr/sbin
+REAL_DAEMON_DIR=/usr/sbin
 #
 # BSD 4.4
 #REAL_DAEMON_DIR=/usr/libexec
Index: tcp_wrappers_7.6/percent_m.c
===================================================================
--- tcp_wrappers_7.6.orig/percent_m.c
+++ tcp_wrappers_7.6/percent_m.c
@@ -14,7 +14,7 @@ static char sccsid[] = "@(#) percent_m.c
 
 extern int errno;
 #ifndef SYS_ERRLIST_DEFINED
-extern char *sys_errlist[];
+extern __const char *__const sys_errlist[];
 extern int sys_nerr;
 #endif
 
