From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: Fix CONFIG_HEADERS

Fix a typo.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---

Index: configure.ac
===================================================================
--- configure.ac.orig
+++ configure.ac
@@ -4,7 +4,7 @@ AC_PREREQ(2.59)
 AC_INIT([libdbus-c++], r9139, [shackan@gmail.com])
 
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
-AM_CONFIG_HEADER([include/dbus-c++/config.h])
+AC_CONFIG_HEADERS([include/dbus-c++/config.h])
 
 AC_CANONICAL_HOST
 
