From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Wed, 4 Aug 2010 10:57:24 +0200
Subject: [PATCH] make constant unsigned to avoid integer overflow

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 include/unarchive.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/unarchive.h b/include/unarchive.h
index b55af6d..4ec4b08 100644
--- a/include/unarchive.h
+++ b/include/unarchive.h
@@ -14,7 +14,7 @@ enum {
 	XZ_MAGIC1   = 0xfd * 256 + '7',
 	XZ_MAGIC2   = (('z' * 256 + 'X') * 256 + 'Z') * 256 + 0,
 	/* Different form: 32 bits, then 16 bits: */
-	XZ_MAGIC1a  = ((0xfd * 256 + '7') * 256 + 'z') * 256 + 'X',
+	XZ_MAGIC1a  = ((0xfdu * 256 + '7') * 256 + 'z') * 256 + 'X',
 	XZ_MAGIC2a  = 'Z' * 256 + 0,
 #else
 	COMPRESS_MAGIC = 0x9d1f,
-- 
1.7.5.1

