diff -ur libdbus-c++-r9139-orig/aclocal.m4 libdbus-c++-r9139-autogen/aclocal.m4
--- libdbus-c++-r9139-orig/aclocal.m4	2007-01-06 16:32:13.000000000 +0100
+++ libdbus-c++-r9139-autogen/aclocal.m4	2007-01-06 17:02:17.000000000 +0100
@@ -6867,18 +6867,6 @@
      [AMDEP_TRUE="$AMDEP_TRUE" ac_aux_dir="$ac_aux_dir"])
 ])
 
-# Copyright (C) 1996, 1997, 2000, 2001, 2003, 2005
-# Free Software Foundation, Inc.
-#
-# This file is free software; the Free Software Foundation
-# gives unlimited permission to copy and/or distribute it,
-# with or without modifications, as long as this notice is preserved.
-
-# serial 8
-
-# AM_CONFIG_HEADER is obsolete.  It has been replaced by AC_CONFIG_HEADERS.
-AU_DEFUN([AM_CONFIG_HEADER], [AC_CONFIG_HEADERS($@)])
-
 # Do all the work for Automake.                             -*- Autoconf -*-
 
 # Copyright (C) 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
Only in libdbus-c++-r9139-autogen: autom4te.cache
diff -ur libdbus-c++-r9139-orig/examples/echo/Makefile.in libdbus-c++-r9139-autogen/examples/echo/Makefile.in
--- libdbus-c++-r9139-orig/examples/echo/Makefile.in	2007-01-06 16:32:15.000000000 +0100
+++ libdbus-c++-r9139-autogen/examples/echo/Makefile.in	2007-01-06 17:02:20.000000000 +0100
@@ -48,7 +48,9 @@
 CONFIG_CLEAN_FILES =
 PROGRAMS = $(noinst_PROGRAMS)
 am_echo_server_OBJECTS = echo-server.$(OBJEXT)
-echo_server_OBJECTS = $(am_echo_server_OBJECTS)
+nodist_echo_server_OBJECTS =
+echo_server_OBJECTS = $(am_echo_server_OBJECTS) \
+	$(nodist_echo_server_OBJECTS)
 echo_server_DEPENDENCIES = $(top_builddir)/src/libdbus-c++-1.la
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)/include/dbus-c++
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -69,7 +71,7 @@
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
-SOURCES = $(echo_server_SOURCES)
+SOURCES = $(echo_server_SOURCES) $(nodist_echo_server_SOURCES)
 DIST_SOURCES = $(echo_server_SOURCES)
 ETAGS = etags
 CTAGS = ctags
@@ -189,7 +191,8 @@
 xml_LIBS = @xml_LIBS@
 EXTRA_DIST = README echo-introspect.xml
 AM_CPPFLAGS = -I$(top_srcdir)/include
-echo_server_SOURCES = echo-glue.h echo-server.h echo-server.cpp
+echo_server_SOURCES = echo-server.h echo-server.cpp
+nodist_echo_server_SOURCES = echo-glue.h
 echo_server_LDADD = $(top_builddir)/src/libdbus-c++-1.la
 BUILT_SOURCES = echo-glue.h
 CLEANFILES = $(BUILT_SOURCES)
@@ -355,9 +358,6 @@
 	    || exit 1; \
 	  fi; \
 	done
-	$(MAKE) $(AM_MAKEFLAGS) \
-	  top_distdir="$(top_distdir)" distdir="$(distdir)" \
-	  dist-hook
 check-am: all-am
 check: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) check-am
@@ -443,7 +443,7 @@
 uninstall-am: uninstall-info-am
 
 .PHONY: CTAGS GTAGS all all-am check check-am clean clean-generic \
-	clean-libtool clean-noinstPROGRAMS ctags dist-hook distclean \
+	clean-libtool clean-noinstPROGRAMS ctags distclean \
 	distclean-compile distclean-generic distclean-libtool \
 	distclean-tags distdir dvi dvi-am html html-am info info-am \
 	install install-am install-data install-data-am install-exec \
@@ -457,9 +457,6 @@
 
 echo-glue.h: echo-introspect.xml
 	$(top_builddir)/tools/dbusxx-xml2cpp $^ --adaptor=$@
-
-dist-hook:
-	cd $(distdir); rm -f $(BUILT_SOURCES)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -ur libdbus-c++-r9139-orig/examples/properties/Makefile.in libdbus-c++-r9139-autogen/examples/properties/Makefile.in
--- libdbus-c++-r9139-orig/examples/properties/Makefile.in	2007-01-06 16:32:15.000000000 +0100
+++ libdbus-c++-r9139-autogen/examples/properties/Makefile.in	2007-01-06 17:02:20.000000000 +0100
@@ -48,7 +48,9 @@
 CONFIG_CLEAN_FILES =
 PROGRAMS = $(noinst_PROGRAMS)
 am_props_server_OBJECTS = props-server.$(OBJEXT)
-props_server_OBJECTS = $(am_props_server_OBJECTS)
+nodist_props_server_OBJECTS =
+props_server_OBJECTS = $(am_props_server_OBJECTS) \
+	$(nodist_props_server_OBJECTS)
 props_server_DEPENDENCIES = $(top_builddir)/src/libdbus-c++-1.la
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)/include/dbus-c++
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -69,7 +71,7 @@
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
-SOURCES = $(props_server_SOURCES)
+SOURCES = $(props_server_SOURCES) $(nodist_props_server_SOURCES)
 DIST_SOURCES = $(props_server_SOURCES)
 ETAGS = etags
 CTAGS = ctags
@@ -189,7 +191,8 @@
 xml_LIBS = @xml_LIBS@
 EXTRA_DIST = README props-introspect.xml
 AM_CPPFLAGS = -I$(top_srcdir)/include
-props_server_SOURCES = props-glue.h props-server.h props-server.cpp
+props_server_SOURCES = props-server.h props-server.cpp
+nodist_props_server_SOURCES = props-glue.h
 props_server_LDADD = $(top_builddir)/src/libdbus-c++-1.la
 BUILT_SOURCES = props-glue.h
 CLEANFILES = $(BUILT_SOURCES)
@@ -355,9 +358,6 @@
 	    || exit 1; \
 	  fi; \
 	done
-	$(MAKE) $(AM_MAKEFLAGS) \
-	  top_distdir="$(top_distdir)" distdir="$(distdir)" \
-	  dist-hook
 check-am: all-am
 check: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) check-am
@@ -443,7 +443,7 @@
 uninstall-am: uninstall-info-am
 
 .PHONY: CTAGS GTAGS all all-am check check-am clean clean-generic \
-	clean-libtool clean-noinstPROGRAMS ctags dist-hook distclean \
+	clean-libtool clean-noinstPROGRAMS ctags distclean \
 	distclean-compile distclean-generic distclean-libtool \
 	distclean-tags distdir dvi dvi-am html html-am info info-am \
 	install install-am install-data install-data-am install-exec \
@@ -457,9 +457,6 @@
 
 props-glue.h: props-introspect.xml
 	$(top_builddir)/tools/dbusxx-xml2cpp $^ --adaptor=$@
-
-dist-hook:
-	cd $(distdir); rm -f $(BUILT_SOURCES)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
