From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 28 Feb 2015 11:07:19 +0100
Subject: [PATCH] lib/Makefile: don't rebuild lib during install

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 lib/Makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/lib/Makefile b/lib/Makefile
index 02ddd3b5f285..feaaa13525d8 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -60,19 +60,19 @@ else
 	SHARED_EXT_VER = $(SHARED_EXT).$(LIBVER)
 endif
 
-default: liblz4
+default: liblz4.$(SHARED_EXT)
 
-all: liblz4
+all: liblz4.$(SHARED_EXT)
 
-liblz4: lz4.c lz4hc.c lz4frame.c xxhash.c
+liblz4.$(SHARED_EXT): lz4.c lz4hc.c lz4frame.c xxhash.c
 	@echo compiling static library
 	@$(CC) $(CPPFLAGS) $(CFLAGS) -c $^
 	@$(AR) rcs liblz4.a lz4.o lz4hc.o lz4frame.o xxhash.o
 	@echo compiling dynamic library $(LIBVER)
-	@$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -shared $^ -fPIC $(SONAME_FLAGS) -o $@.$(SHARED_EXT_VER)
+	@$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -shared $^ -fPIC $(SONAME_FLAGS) -o $(basename $@).$(SHARED_EXT_VER)
 	@echo creating versioned links
-	@ln -sf $@.$(SHARED_EXT_VER) $@.$(SHARED_EXT_MAJOR)
-	@ln -sf $@.$(SHARED_EXT_VER) $@.$(SHARED_EXT)
+	@ln -sf $(basename $@).$(SHARED_EXT_VER) $(basename $@).$(SHARED_EXT_MAJOR)
+	@ln -sf $(basename $@).$(SHARED_EXT_VER) $(basename $@).$(SHARED_EXT)
 
 clean:
 	@rm -f core *.o *.a *.$(SHARED_EXT) *.$(SHARED_EXT).* liblz4.pc
@@ -91,7 +91,7 @@ liblz4.pc: liblz4.pc.in Makefile
             -e 's|@VERSION@|$(VERSION)|' \
              $< >$@
 
-install: liblz4 liblz4.pc
+install: liblz4.$(SHARED_EXT) liblz4.pc
 	@install -d -m 755 $(DESTDIR)$(LIBDIR)/pkgconfig/ $(DESTDIR)$(INCLUDEDIR)/
 	@install -m 755 liblz4.$(SHARED_EXT_VER) $(DESTDIR)$(LIBDIR)/liblz4.$(SHARED_EXT_VER)
 	@cp -a liblz4.$(SHARED_EXT_MAJOR) $(DESTDIR)$(LIBDIR)
