diff -ruN dnsmasq-1.18.orig/Makefile dnsmasq-1.18/Makefile
--- dnsmasq-1.18.orig/Makefile	2003-11-06 22:05:58.000000000 +0100
+++ dnsmasq-1.18/Makefile	2003-12-21 10:53:42.000000000 +0100
@@ -2,11 +2,6 @@
 #LIBS = -lsocket -lnsl
 
 OBJS = cache.o rfc1035.o dhcp.o util.o option.o forward.o network.o dnsmasq.o
-BINDIR = /usr/local/sbin
-MANDIR = /usr/local/man
-
-CC = gcc
-CFLAGS = -O2
 
 all : dnsmasq
 
@@ -14,7 +9,7 @@
 	rm -f *~ */*~ $(OBJS) dnsmasq core 
 
 .c.o: dnsmasq.h config.h
-	$(CC) $(CFLAGS) $(RPM_OPT_FLAGS) -Wall -W -c $*.c 
+	$(CC) $(CFLAGS) -O2 -c $*.c 
 
 dnsmasq : $(OBJS) dnsmasq.h config.h
 	$(CC) -o $@  $(OBJS) $(LIBS)
