---
#20081124: wsa: Sent upstream.

 configure.in |    1 -
 1 file changed, 1 deletion(-)

Index: usbutils-0.72/configure.in
===================================================================
--- usbutils-0.72.orig/configure.in
+++ usbutils-0.72/configure.in
@@ -28,7 +28,6 @@ AC_TYPE_OFF_T
 AC_TYPE_SIZE_T
 
 # Checks for library functions.
-AC_FUNC_MALLOC
 AC_FUNC_VPRINTF
 AC_CHECK_FUNCS([getcwd memset getopt_long strchr strerror strstr strtoul uname])
 AC_CHECK_LIB(usb, usb_get_string_simple, ,
