--- ffmpeg-r5452/configure.orig	2006-06-05 15:09:17.000000000 +0200
+++ ffmpeg-r5452/configure	2006-06-05 15:09:32.000000000 +0200
@@ -129,12 +129,12 @@
 
 check_cc(){
     cat >$TMPC
-    ${cross_prefix}$cc $CFLAGS "$@" -c -o $TMPO $TMPC >/dev/null 2>&1
+    $cc $CFLAGS "$@" -c -o $TMPO $TMPC >/dev/null 2>&1
 }
 
 check_ld(){
     cat >$TMPC
-    ${cross_prefix}$cc $CFLAGS $LDFLAGS "$@" -o $TMPE $TMPC $extralibs >/dev/null 2>&1
+    $cc $CFLAGS $LDFLAGS "$@" -o $TMPE $TMPC $extralibs >/dev/null 2>&1
 }
 
 check_cflags(){
