diff -urN Randr-20041103-1/autogen-rs.sh Randr-ptx/autogen-rs.sh
--- Randr-20041103-1/autogen-rs.sh	1970-01-01 01:00:00.000000000 +0100
+++ Randr-ptx/autogen-rs.sh	2004-11-03 07:41:35.000000000 +0100
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+aclocal-1.7 -I /usr/share/aclocal/
+#autoheader
+autoconf
+automake-1.7 --add-missing --copy
+
diff -urN Randr-20041103-1/configure.ac Randr-ptx/configure.ac
--- Randr-20041103-1/configure.ac	2004-11-03 09:50:40.000000000 +0100
+++ Randr-ptx/configure.ac	2004-11-03 08:11:40.000000000 +0100
@@ -25,7 +25,7 @@
 
 AC_PREREQ([2.57])
 AC_INIT([randrext], [1.0], [keithp@keithp.com], randrext)
-AM_INIT_AUTOMAKE([dist-bzip2])
+AM_INIT_AUTOMAKE(Randr, 20041103-1, [dist-bzip2])
 AM_MAINTAINER_MODE
 
 AC_OUTPUT([Makefile
diff -urN Randr-20041103-1/Makefile.am Randr-ptx/Makefile.am
--- Randr-20041103-1/Makefile.am	2004-11-03 09:50:40.000000000 +0100
+++ Randr-ptx/Makefile.am	2004-11-02 17:15:15.000000000 +0100
@@ -30,5 +30,5 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = randr.pc
 
-EXTRA_DIST = autogen.sh randr.pc.in $(randrdoc_DATA)
+EXTRA_DIST = autogen.sh autogen-rs.sh randr.pc.in $(randrdoc_DATA)
 
