From 2a3cc3bb6f1bf8b60fd896de8a91d2c7f50b12d6 Mon Sep 17 00:00:00 2001
From: Luotao Fu <l.fu@pengutronix.de>
Date: Sun, 10 Jan 2010 00:50:58 +0100
Subject: [PATCH] pcre-config: add SYSROOT support

Signed-off-by: Luotao Fu <l.fu@pengutronix.de>
Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 pcre-config.in |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/pcre-config.in b/pcre-config.in
index a555744..847023f 100644
--- a/pcre-config.in
+++ b/pcre-config.in
@@ -49,16 +49,16 @@ while test $# -gt 0; do
       echo @PACKAGE_VERSION@
       ;;
     --cflags | --cflags-posix)
-      if test @includedir@ != /usr/include ; then
-        includes=-I@includedir@
+      if test ${SYSROOT}@includedir@ != /usr/include ; then
+        includes=-I${SYSROOT}@includedir@
       fi
       echo $includes @PCRE_STATIC_CFLAG@
       ;;
     --libs-posix)
-      echo -L@libdir@$libR -lpcreposix -lpcre
+      echo -L${SYSROOT}@libdir@$libR -lpcreposix -lpcre
       ;;
     --libs)
-      echo -L@libdir@$libR -lpcre
+      echo -L${SYSROOT}@libdir@$libR -lpcre
       ;;
     *)
       echo "${usage}" 1>&2
-- 
1.6.6

