From 10494d2724cac05d6a8a352205da18b86787eaba Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Petr=20Jedin=C3=BD?= <petr.jediny@gmail.com>
Date: Thu, 7 Aug 2008 16:45:33 +0200
Subject: [PATCH 02/23] Changing commandline help text to be more uniform...

---
 ip/ip.c |    4 ++--
 tc/tc.c |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ip/ip.c b/ip/ip.c
index c4c773f..2bd54b2 100644
--- a/ip/ip.c
+++ b/ip/ip.c
@@ -45,12 +45,12 @@ static void usage(void)
 {
 	fprintf(stderr,
 "Usage: ip [ OPTIONS ] OBJECT { COMMAND | help }\n"
-"       ip [ -force ] [-batch filename\n"
+"       ip [ -force ] -batch filename\n"
 "where  OBJECT := { link | addr | addrlabel | route | rule | neigh | ntable |\n"
 "                   tunnel | maddr | mroute | monitor | xfrm }\n"
 "       OPTIONS := { -V[ersion] | -s[tatistics] | -d[etails] | -r[esolve] |\n"
 "                    -f[amily] { inet | inet6 | ipx | dnet | link } |\n"
-"                    -o[neline] | -t[imestamp] }\n");
+"                    -o[neline] | -t[imestamp] | -b[atch] [filename] }\n");
 	exit(-1);
 }
 
diff --git a/tc/tc.c b/tc/tc.c
index aabeac8..8e362d2 100644
--- a/tc/tc.c
+++ b/tc/tc.c
@@ -182,9 +182,9 @@ noexist:
 static void usage(void)
 {
 	fprintf(stderr, "Usage: tc [ OPTIONS ] OBJECT { COMMAND | help }\n"
-			"       tc [-force] -batch file\n"
+			"       tc [-force] -batch filename\n"
 	                "where  OBJECT := { qdisc | class | filter | action | monitor }\n"
-	                "       OPTIONS := { -s[tatistics] | -d[etails] | -r[aw] | -p[retty] | -b[atch] [file] }\n");
+	                "       OPTIONS := { -s[tatistics] | -d[etails] | -r[aw] | -p[retty] | -b[atch] [filename] }\n");
 }
 
 static int do_cmd(int argc, char **argv)
-- 
1.6.3.3

