From ed1af7e868829367f70079d0ad2b81c23e3cfc0f Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@grandegger.com>
Date: Tue, 12 May 2009 11:42:49 +0200
Subject: [PATCH 21/23] iproute2: Fixes an issue with cross-compilation

Signed-off-by: Wolfgang Grandegger <wg@grandegger.com>
---
 Makefile |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index ce8dd06..74e9d62 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,5 @@
 DESTDIR=/usr/
+ROOTDIR=$(DESTDIR)
 LIBDIR=/usr/lib/
 SBINDIR=/sbin
 CONFDIR=/etc/iproute2
@@ -7,7 +8,7 @@ MANDIR=/share/man
 ARPDDIR=/var/lib/arpd
 
 # Path to db_185.h include
-DBM_INCLUDE:=/usr/include
+DBM_INCLUDE:=$(ROOTDIR)/usr/include
 
 DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
 
-- 
1.6.3.3

