From 1558971d43bbcd5db7f9be966654482a96179bf1 Mon Sep 17 00:00:00 2001
From: Stephen Hemminger <stephen.hemminger@vyatta.com>
Date: Tue, 26 May 2009 15:58:05 -0700
Subject: [PATCH 23/23] fix handling of GRED DPs args

---
 tc/q_gred.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tc/q_gred.c b/tc/q_gred.c
index e26840b..ecef42e 100644
--- a/tc/q_gred.c
+++ b/tc/q_gred.c
@@ -73,7 +73,7 @@ static int init_gred(struct qdisc_util *qu, int argc, char **argv,
 		} else if (strcmp(*argv, "default") == 0) {
 			NEXT_ARG();
 			def_dp = strtol(*argv, (char **)NULL, 10);
-			if (dps) {
+			if (dps == 0) {
 				fprintf(stderr, "\"default DP\" must be "
 					"defined after DPs\n");
 				return -1;
-- 
1.6.3.3

