--- mknbi-1.4.4/Makefile	2004-08-16 19:47:29.000000000 -0700
+++ mknbi-1.4.4/Makefile.foo	2005-08-30 15:45:20.000000000 -0700
@@ -38,20 +38,20 @@
 # Must pick ANSIMODE=1 if ANSIESC is chosen
 BOOTMENU_FLAGS=	-DCONSOLE_CRT -DANSIESC -DANSIMODE=1 -DMOTD -DUSRPARMS -DPASSWD -DPOWERSAVE -DCOMCONSOLE=0x3f8 -DCOMPRESERVE
 
-PREFIX=		/usr/local
-INSTPREFIX=	$(BUILD_ROOT)$(PREFIX)
-LIBDIR=		$(PREFIX)/lib/mknbi
-INSTLIBDIR=	$(INSTPREFIX)/lib/mknbi
-BINDIR=		$(INSTPREFIX)/bin
-MANDIR=		$(INSTPREFIX)/share/man/man1
-DOCDIR=		$(INSTPREFIX)/share/doc/packages/mknbi-$(RPMVERSION)
+PREFIX?=		/usr/local
+INSTPREFIX?=	$(BUILD_ROOT)$(PREFIX)
+LIBDIR?=		$(PREFIX)/lib/mknbi
+INSTLIBDIR?=	$(INSTPREFIX)/lib/mknbi
+BINDIR?=		$(INSTPREFIX)/bin
+MANDIR?=		$(INSTPREFIX)/share/man/man1
+DOCDIR?=		$(INSTPREFIX)/share/doc/packages/mknbi-$(RPMVERSION)
 
 INSTALL=	install
 
-PROG=	mknbi disnbi dismbr disdosbb nbitoelf
+PROG=	mknbi disnbi dismbr nbitoelf
 MODULES=Nbi.pm Elf.pm TruncFD.pm
 FIRSTS=	$(foreach i,$(FIRSTRELOCS),first32@$(i).linux first32elf@$(i).linux) \
-	first.dos first.fdos menu nfl
+	menu nfl
 ALTBOOT=altboot.bin
 RMRD=	rmrd.com
 MANS=	mknbi.1 disnbi.1 menuc.1
