--- netcat-1.10/generic.h.old	1996-02-17 17:36:43.000000000 -0800
+++ netcat-1.10/generic.h	2003-09-29 10:14:26.000000000 -0700
@@ -86,6 +86,9 @@
 /* stdarg is a weird one */
 #define HAVE_STDARG_H
 
+/* resolv for res_init */
+#define HAVE_RESOLV_H
+
 /* dir.h or maybe ndir.h otherwise. */
 #define HAVE_DIRENT_H
 
--- netcat-1.10/netcat.c.old	1996-03-20 16:38:04.000000000 -0800
+++ netcat-1.10/netcat.c	2003-09-29 10:14:26.000000000 -0700
@@ -43,6 +43,9 @@
 #ifdef HAVE_SELECT_H		/* random SV variants need this */
 #include <sys/select.h>
 #endif
+#ifdef HAVE_RESOLV_H		/* Linux seems to need this for res_init */
+#include <resolv.h>
+#endif
 
 /* have to do this *before* including types.h. xxx: Linux still has it wrong */
 #ifdef FD_SETSIZE		/* should be in types.h, butcha never know. */
