Subject: busybox 1.9.1 fixes
From: http://busybox.net/downloads/fixes-1.9.1

---
 networking/udhcp/files.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: busybox-1.9.1/networking/udhcp/files.c
===================================================================
--- busybox-1.9.1.orig/networking/udhcp/files.c
+++ busybox-1.9.1/networking/udhcp/files.c
@@ -167,7 +167,7 @@ static int read_opt(const char *const_li
 	if (!opt)
 		return 0;
 
-	idx = index_in_strings(opt, dhcp_option_strings); /* NB: was strcasecmp! */
+	idx = index_in_strings(dhcp_option_strings, opt); /* NB: was strcasecmp! */
 	if (idx < 0)
 		return 0;
 	option = &dhcp_options[idx];
