From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 5 Jul 2011 11:14:02 +0200
Subject: [PATCH] add variable to allow external 'wart'

this fixes cross-compiling

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 makefile |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/makefile b/makefile
index 176bc4e..ba5e3ba 100644
--- a/makefile
+++ b/makefile
@@ -920,6 +920,7 @@ CC= cc
 CC2= cc
 MAKE= make
 SHELL=/bin/sh
+WART=./wart
 
 ###########################################################################
 # (Ancient) UNIX V7-specific variables.
@@ -1414,9 +1415,9 @@ ckclib.$(EXT): ckclib.c ckclib.h ckcdeb.h ckcasc.h ckcsym.h
 
 ckcpro.$(EXT): ckcpro.c ckcker.h ckcdeb.h ckcsym.h ckcasc.h ckclib.h
 
-ckcpro.c: ckcpro.w wart ckcdeb.h ckcsym.h ckcasc.h ckcker.h ckcnet.h ckctel.h \
+ckcpro.c: ckcpro.w $(WART) ckcdeb.h ckcsym.h ckcasc.h ckcker.h ckcnet.h ckctel.h \
 	 ckclib.h
-	./wart ckcpro.w ckcpro.c
+	$(WART) ckcpro.w ckcpro.c
 
 ckcfns.$(EXT): ckcfns.c ckcker.h ckcdeb.h ckcsym.h ckcasc.h ckcxla.h ckcuni.h \
 		ckuxla.h ckclib.h ckcnet.h
