From: Markus Pargmann <mpa@pengutronix.de>
Date: Thu, 17 Apr 2014 15:34:35 +0200
Subject: [PATCH] Makefile: Fix libreg build

Signed-off-by: Markus Pargmann <mpa@pengutronix.de>
---
 Makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index a3ead30371c9..60a3182474e6 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,8 @@ REG_GIT?=git://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.g
 
 PREFIX ?= /usr/
 MANDIR ?= $(PREFIX)/share/man/
-INCLUDE_DIR ?= $(PREFIX)/include/reglib/
-LIBDIR ?= $(PREFIX)/lib
+INCLUDE_DIR ?= /include/reglib/
+LIBDIR ?= /lib/
 
 SBINDIR ?= /sbin/
 
@@ -120,14 +120,14 @@ $(LIBREG): regdb.h reglib.h reglib.c
 
 install-libreg-headers:
 	$(NQ) '  INSTALL  libreg-headers'
-	$(Q)mkdir -p $(DESTDIR)/$(INCLUDE_DIR)
-	$(Q)cp *.h $(DESTDIR)/$(INCLUDE_DIR)/
+	$(Q)$(MKDIR) $(DESTDIR)/$(INCLUDE_DIR)
+	$(Q)$(INSTALL) -m 644 -t $(DESTDIR)/$(INCLUDE_DIR) *.h
 
 install-libreg:
 	$(NQ) '  INSTALL  libreg'
-	$(Q)mkdir -p $(DESTDIR)/$(LIBDIR)
-	$(Q)cp $(LIBREG) $(DESTDIR)/$(LIBDIR)/
-	$(Q)ldconfig
+	$(Q)$(MKDIR) $(LIBDIR)
+	$(Q)$(INSTALL) -m 644 -t $(DESTDIR)/$(LIBDIR) $(LIBREG)
+#	$(Q)ldconfig
 
 %.o: %.c regdb.h $(LIBREG)
 	$(NQ) '  CC  ' $@
