From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 13 Oct 2012 10:39:47 +0200
Subject: [PATCH] mconf: fix comment on exit

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 scripts/kconfig/mconf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/kconfig/mconf.c b/scripts/kconfig/mconf.c
index 26a330881282..360db50fcd30 100644
--- a/scripts/kconfig/mconf.c
+++ b/scripts/kconfig/mconf.c
@@ -975,7 +975,7 @@ static int handle_exit(void)
 		if (!silent)
 			printf(_("\n\n"
 				 "*** End of the configuration.\n"
-				 "*** Execute 'make' to start the build or try 'make help'."
+				 "*** Execute 'ptxdist go' to build the project or try 'ptxdist help'."
 				 "\n\n"));
 		res = 0;
 		break;
