--- util-linux-2.11z/MCONFIG.orig	2003-02-19 17:51:51.000000000 -0800
+++ util-linux-2.11z/MCONFIG	2003-02-19 17:53:19.000000000 -0800
@@ -10,8 +10,8 @@
 # Select for ARCH one of intel, alpha, sparc, arm, m68k, mips
 # Select for CPU i386 if the binaries must be able to run on an intel 386
 # (by default i486 code is generated, see below)
-CPU=$(shell uname -m)
-ARCH=$(shell echo $(CPU) | sed 's/i.86/intel/;s/arm.*/arm/')
+#CPU=$(shell uname -m)
+#ARCH=$(shell echo $(CPU) | sed 's/i.86/intel/;s/arm.*/arm/')

 # If HAVE_PAM is set to "yes", then login, chfn, chsh, and newgrp
 # will use PAM for authentication. Additionally, passwd will not be
