From: Wolfram Sang <w.sang@pengutronix.de>
Subject: Add Pengutronix logo

Signed-off-by: (not for upstream)

---
 src/df_andi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/df_andi.c
===================================================================
--- src/df_andi.c.orig
+++ src/df_andi.c
@@ -513,7 +513,7 @@ static void init_resources( int argc, ch
      tuximage->SetSrcColorKey( tuximage, 0x00, 0xFF, 0x00 );
 
      /* load the background image */
-     DFBCHECK(dfb->CreateImageProvider( dfb, DATADIR"/wood_andi.jpg",
+     DFBCHECK(dfb->CreateImageProvider( dfb, DATADIR"/ptx_logo_640_480.png",
                                         &provider ));
 
      DFBCHECK (provider->GetSurfaceDescription (provider, &dsc));
