Index: util-linux-2.13-pre7/configure.ac
===================================================================
--- util-linux-2.13-pre7.orig/configure.ac
+++ util-linux-2.13-pre7/configure.ac
@@ -45,6 +45,7 @@ AC_CHECK_FUNCS(updwtmp)
 AC_FUNC_FSEEKO
 AC_CHECK_FUNCS(lchown)
 AC_CHECK_FUNCS(rpmatch)
+AC_CHECK_FUNCS(umount2)
 
 
 AC_CHECK_LIB(uuid, uuid_is_null)
Index: util-linux-2.13-pre7/mount/umount.c
===================================================================
--- util-linux-2.13-pre7.orig/mount/umount.c
+++ util-linux-2.13-pre7/mount/umount.c
@@ -29,28 +29,21 @@
 #include "nfsmount.h"
 #include <arpa/inet.h>
 
+#include <config.h>
+
 #if defined(MNT_FORCE) && !defined(__sparc__) && !defined(__arm__)
 /* Interesting ... it seems libc knows about MNT_FORCE and presumably
    about umount2 as well -- need not do anything */
 #else /* MNT_FORCE */
 
-/* Does the present kernel source know about umount2? */
-#include <linux/unistd.h>
-#ifdef __NR_umount2
-
-static int umount2(const char *path, int flags);
-
-_syscall2(int, umount2, const char *, path, int, flags);
-
-#else /* __NR_umount2 */
-
+#if !defined(HAVE_UMOUNT2)
 static int
 umount2(const char *path, int flags) {
 	fprintf(stderr, _("umount: compiled without support for -f\n"));
 	errno = ENOSYS;
 	return -1;
 }
-#endif /* __NR_umount2 */
+#endif
 
 #if !defined(MNT_FORCE)
 /* dare not try to include <linux/mount.h> -- lots of errors */
Index: util-linux-2.13-pre7/mount/Makefile.am
===================================================================
--- util-linux-2.13-pre7.orig/mount/Makefile.am
+++ util-linux-2.13-pre7/mount/Makefile.am
@@ -1,6 +1,6 @@
 include $(top_srcdir)/config/include-Makefile.am
 
-AM_CPPFLAGS += -DHAVE_NFS
+AM_CPPFLAGS += -DHAVE_NFS -I$(top_srcdir)
 
 bin_PROGRAMS = mount umount
 sbin_PROGRAMS = losetup swapon
