Index: pure-ftpd-1.0.20/configure
===================================================================
--- pure-ftpd-1.0.20.orig/configure
+++ pure-ftpd-1.0.20/configure
@@ -18909,11 +18909,7 @@ rm -f conftest.err conftest.$ac_objext c
 echo "$as_me:$LINENO: checking whether snprintf is C99 conformant" >&5
 echo $ECHO_N "checking whether snprintf is C99 conformant... $ECHO_C" >&6
 if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+  CONF_SNPRINTF_TYPE=8
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -18983,11 +18979,10 @@ fi
 echo "$as_me:$LINENO: checking whether getgroups 0 is sane" >&5
 echo $ECHO_N "checking whether getgroups 0 is sane... $ECHO_C" >&6
 if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+
+echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6
+
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -19051,11 +19046,13 @@ fi
 echo "$as_me:$LINENO: checking whether realpath likes unreadable directories" >&5
 echo $ECHO_N "checking whether realpath likes unreadable directories... $ECHO_C" >&6
 if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+  echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6
+cat >>confdefs.h <<\_ACEOF
+#define USE_BUILTIN_REALPATH 1
+_ACEOF
+
+
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -19157,11 +19154,9 @@ fi
 echo "$as_me:$LINENO: checking whether you already have a standard MD5 implementation" >&5
 echo $ECHO_N "checking whether you already have a standard MD5 implementation... $ECHO_C" >&6
 if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+  echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6
+
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -19226,11 +19221,9 @@ fi
 echo "$as_me:$LINENO: checking whether you already have a standard SHA1 implementation" >&5
 echo $ECHO_N "checking whether you already have a standard SHA1 implementation... $ECHO_C" >&6
 if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+  echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6
+
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
