Index: pure-ftpd-1.0.20/configure.ac
===================================================================
--- pure-ftpd-1.0.20.orig/configure.ac
+++ pure-ftpd-1.0.20/configure.ac
@@ -884,7 +884,7 @@ int main(void)
     return 0;
 }
 ]])],[CONF_SNPRINTF_TYPE=`cat conftestval`
-],[],[])
+],[],[CONF_SNPRINTF_TYPE=8])
 AC_MSG_RESULT(done)
 if test "x$CONF_SNPRINTF_TYPE" = "x" ; then
   AC_MSG_WARN(your operating system doesn't implement snprintf)
@@ -916,7 +916,9 @@ int main(void)
 AC_MSG_RESULT(yes)
 AC_DEFINE(SAFE_GETGROUPS_0,,[Define is getgroups(0, NULL) works on your system])
 ],[AC_MSG_RESULT(no)
-],[])
+],[
+AC_MSG_RESULT(no)
+])
 
 AC_MSG_CHECKING(whether realpath likes unreadable directories)
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
@@ -977,7 +979,9 @@ AC_DEFINE(REALPATH_IGNORES_UNREADABLE_DI
 [Define if realpath() works on unreadable directories])
 ],[AC_MSG_RESULT(no)
 AC_DEFINE(USE_BUILTIN_REALPATH)
-],[])
+],[AC_MSG_RESULT(no)
+AC_DEFINE(USE_BUILTIN_REALPATH)
+])
 
 AC_MSG_CHECKING(whether you already have a standard MD5 implementation)
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
@@ -1004,7 +1008,8 @@ AC_MSG_RESULT(yes)
 AC_DEFINE(USE_SYSTEM_CRYPT_MD5,,[Define if you already have standard
 MD5 functions])
 ],[AC_MSG_RESULT(no)
-],[])
+],[AC_MSG_RESULT(no)
+])
 
 AC_MSG_CHECKING(whether you already have a standard SHA1 implementation)
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
@@ -1030,7 +1035,8 @@ AC_MSG_RESULT(yes)
 AC_DEFINE(USE_SYSTEM_CRYPT_SHA1,,[Define if you already have standard
 SHA1 functions])
 ],[AC_MSG_RESULT(no)
-],[])
+],[AC_MSG_RESULT(no)
+])
 
 AC_MSG_CHECKING([whether we are inside a Virtuozzo virtual host])
 if test -d /proc/vz; then
