From e3cbc39bf1a94056f4a4bbae0067109d0317e68b Mon Sep 17 00:00:00 2001
From: Jon Ringle <jon@ringle.org>
Date: Thu, 29 Oct 2009 16:27:14 +0100
Subject: [PATCH 20/21] fix warning include/signal.h:47:5: warning: "SIGRTMAX" is not defined

Signed-off-by: Jon Ringle <jon@ringle.org>
Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 usr/include/signal.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/usr/include/signal.h b/usr/include/signal.h
index a513282..4ed65bc 100644
--- a/usr/include/signal.h
+++ b/usr/include/signal.h
@@ -44,10 +44,12 @@ typedef int sig_atomic_t;
 /* The kernel header files are inconsistent whether or not
    SIGRTMAX is inclusive or exclusive.  POSIX seems to state that
    it's inclusive, however. */
+#ifdef SIGRTMAX
 #if SIGRTMAX >= _NSIG
 # undef  SIGRTMAX
 # define SIGRTMAX (_NSIG-1)
 #endif
+#endif
 
 __extern const char *const sys_siglist[_NSIG];
 __extern const char *const sys_sigabbrev[_NSIG];
-- 
1.6.5

