diff -urN confuse-2.6-orig/src/Makefile.in confuse-2.6/src/Makefile.in
--- confuse-2.6-orig/src/Makefile.in	2007-12-29 11:56:12.000000000 +0100
+++ confuse-2.6/src/Makefile.in	2009-01-01 23:27:27.000000000 +0100
@@ -196,7 +196,7 @@
 libconfuse_la_SOURCES = confuse.c lexer.l
 libconfuse_la_LIBADD = $(LTLIBINTL)
 include_HEADERS = confuse.h
-@cc_is_gcc_TRUE@AM_CFLAGS = -Wall -Werror
+@cc_is_gcc_TRUE@AM_CFLAGS = -Wall
 CLEANFILES = *~ \#*\#
 AM_LFLAGS = -Pcfg_yy -olex.yy.c
 all: all-am
