Index: configure
===================================================================
--- a/configure	2015-04-28 14:28:30.000000000 +0200
+++b/configure	2015-04-28 14:32:05.782896922 +0200
@@ -14513,7 +14513,7 @@
 else
 
 if test "$cross_compiling" = yes; then :
-  ac_cv_working_tzset=no
+  ac_cv_working_tzset=yes
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
Index: configure.ac
===================================================================
--- a/configure.ac	2015-04-28 14:31:45.000000000 +0200
+++ b/configure.ac	2015-04-28 14:31:58.614896931 +0200
@@ -4360,7 +4360,7 @@
 ]])],
 [ac_cv_working_tzset=yes],
 [ac_cv_working_tzset=no],
-[ac_cv_working_tzset=no])])
+[ac_cv_working_tzset=yes])])
 AC_MSG_RESULT($ac_cv_working_tzset)
 if test "$ac_cv_working_tzset" = yes
 then
