From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 15 Jul 2010 14:50:27 +0200
Subject: [PATCH] don't link the shared lib against the static lib

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/configure b/configure
index bd9edd2..e6c2ead 100755
--- a/configure
+++ b/configure
@@ -19,7 +19,6 @@ if [ -n "${CHOST}" ]; then
 fi
 
 STATICLIB=libz.a
-LDFLAGS="${LDFLAGS} -L. ${STATICLIB}"
 VER=`sed -n -e '/VERSION "/s/.*"\(.*\)".*/\1/p' < zlib.h`
 VER3=`sed -n -e '/VERSION "/s/.*"\([0-9]*\\.[0-9]*\\.[0-9]*\).*/\1/p' < zlib.h`
 VER2=`sed -n -e '/VERSION "/s/.*"\([0-9]*\\.[0-9]*\)\\..*/\1/p' < zlib.h`
-- 
1.7.1

