Index: xorg-server-1.4/hw/xfree86/dri/dri.c
===================================================================
--- xorg-server-1.4.orig/hw/xfree86/dri/dri.c	2009-07-28 12:00:40.000000000 +0200
+++ xorg-server-1.4/hw/xfree86/dri/dri.c	2009-07-28 12:02:21.000000000 +0200
@@ -74,10 +74,6 @@
 
 #define PCI_BUS_NO_DOMAIN(bus) ((bus) & 0xffu)
 
-#if !defined(PANORAMIX)
-extern Bool noPanoramiXExtension;
-#endif
-
 static int DRIEntPrivIndex = -1;
 static int DRIScreenPrivIndex = -1;
 static int DRIWindowPrivIndex = -1;
@@ -308,7 +304,6 @@
     drm_context_t *       reserved;
     int                 reserved_count;
     int                 i;
-    Bool                xineramaInCore = FALSE;
     DRIEntPrivPtr       pDRIEntPriv;
     ScrnInfoPtr         pScrn = xf86Screens[pScreen->myNum];
     DRIContextFlags	flags    = 0;
@@ -320,22 +315,18 @@
 		  "Direct rendering has been disabled.\n");
 	return FALSE;
     }
-
+#ifdef PANORAMIX
     /*
      * If Xinerama is on, don't allow DRI to initialise.  It won't be usable
      * anyway.
      */
-    if (xf86LoaderCheckSymbol("noPanoramiXExtension"))
-	xineramaInCore = TRUE;
 
-    if (xineramaInCore) {
 	if (!noPanoramiXExtension) {
 	    DRIDrvMsg(pScreen->myNum, X_WARNING,
 		"Direct rendering is not supported when Xinerama is enabled\n");
 	    return FALSE;
 	}
-    }
-
+#endif
     if (!DRIOpenDRMMaster(pScrn, pDRIInfo->SAREASize,
 			  pDRIInfo->busIdString,
 			  pDRIInfo->drmDriverName))
