diff -urN bing-1.0.5-orig/bing.c bing-1.0.5/bing.c
--- bing-1.0.5-orig/bing.c	2004-06-29 20:20:57.000000000 +0200
+++ bing-1.0.5/bing.c	2004-06-30 08:29:53.000000000 +0200
@@ -169,6 +169,7 @@
 #endif /* linux */
 #include <netdb.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -1228,7 +1229,7 @@
 	int i;
 	int ch, hold, recv_packlen, preload;
 	u_char *datap, *recv_packet;
-	char *target1, *target2, *malloc();
+	char *target1, *target2;
 	u_char ttl, loop;
 #ifdef IP_OPTIONS
 	char rspace[3 + 4 * NROUTES + 1];	/* record route space */
