From 6e30f01c5dc810e943123f0b272a2299f194c94a Mon Sep 17 00:00:00 2001
From: unknown author <unknown.author@example.com>
Date: Thu, 16 Jun 2011 21:13:59 +0200
Subject: [PATCH 01/10] enable-sdl.diff

---
 Src/configure.in |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Src/configure.in b/Src/configure.in
index cee913c..b439941 100644
--- a/Src/configure.in
+++ b/Src/configure.in
@@ -80,15 +80,15 @@ elif [[ x"$no_x" = "xyes" ]]; then
     TARGET=svgalib
     CFLAGS="$CFLAGS -D__svgalib__"
     LIBS="$LIBS -lvga"
+  elif [[ x"$HAVE_SDL" = "xyes" ]]; then
+    AC_MSG_RESULT([SDL])
+    TARGET=sdl
+    CFLAGS="$CFLAGS $SDL_CFLAGS -DHAVE_SDL"
+    LIBS="$LIBS $SDL_LIBS"
   else
     AC_MSG_RESULT([Ummm...])
     AC_MSG_ERROR([Neither X nor SVGAlib found, don't know what target to use.])
   fi
-elif [[ x"$HAVE_SDL" = "xyes" ]]; then
-  AC_MSG_RESULT([SDL])
-  TARGET=sdl
-  CFLAGS="$CFLAGS $SDL_CFLAGS -DHAVE_SDL"
-  LIBS="$LIBS $SDL_LIBS"
 else
   AC_MSG_RESULT([X Window System])
   TARGET=x11
-- 
1.7.5.3

