From e9f6e3b31c6864f8defc0576a18e48485911dc56 Mon Sep 17 00:00:00 2001
From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Thu, 16 Jun 2011 21:29:06 +0200
Subject: [PATCH 02/10] frodo: add cppflags support

---
 Src/Makefile.in |   21 +++++++++++----------
 1 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/Src/Makefile.in b/Src/Makefile.in
index e848b47..f5b70f8 100644
--- a/Src/Makefile.in
+++ b/Src/Makefile.in
@@ -9,6 +9,7 @@ REVISION  = 1
 CXX    = @CXX@
 LIBS   = @LIBS@
 CFLAGS = @CFLAGS@ -I./ -DFRODO_HPUX_REV=@HPUX_REV@ -DKBD_LANG=@KBD_LANG@
+CPPFLAGS = @CPPFLAGS@
 
 INSTALL         = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
@@ -62,34 +63,34 @@ depend dep:
 	makedepend $(CPPFLAGS) -Y. $(SRCS) 2>/dev/null
 
 .cpp.o:
-	$(CC) $(INCLUDES) $(CFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) -o $@ -c $*.cpp
        
 .cpp.s:
-	$(CC) $(INCLUDES) $(CFLAGS) $(EXTRAFLAGS) -o $@ -S $*.cpp -g0
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(EXTRAFLAGS) -o $@ -S $*.cpp -g0
 
 C64_PC.o: C64.cpp C64.h C64_x.i CmdPipe.h CPUC64.h CPU1541.h VIC.h SID.h CIA.h REU.h IEC.h 1541job.h Display.h Prefs.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(PCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(PCFLAGS) -o $@ -c $*.cpp
 
 CPUC64_PC.o: CPUC64.cpp CPUC64.h CPU_emulline.i C64.h VIC.h SID.h CIA.h REU.h IEC.h Display.h Version.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(PCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(PCFLAGS) -o $@ -c $*.cpp
 
 CPU1541_PC.o: CPU1541.cpp CPU1541.h CPU_emulline.i 1541job.h C64.h CIA.h Display.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(PCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(PCFLAGS) -o $@ -c $*.cpp
 
 C64_SC.o: C64_SC.cpp C64.h C64_x.i CmdPipe.h CPUC64.h CPU1541.h VIC.h SID.h CIA.h REU.h IEC.h 1541job.h Display.h Prefs.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
 
 CPUC64_SC.o: CPUC64_SC.cpp CPUC64.h CPU_emulcycle.i CPU_common.h C64.h VIC.h SID.h CIA.h REU.h IEC.h Display.h Version.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
 
 CPU1541_SC.o: CPU1541_SC.cpp CPU1541.h CPU_emulcycle.i CPU_common.h 1541job.h C64.h CIA.h Display.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
 
 VIC_SC.o: VIC_SC.cpp VIC.h C64.h CPUC64.h Display.h Prefs.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
 
 CIA_SC.o: CIA_SC.cpp CIA.h CPUC64.h CPU1541.h VIC.h Prefs.h
-	$(CC) $(INCLUDES) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
+	$(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(SCFLAGS) -o $@ -c $*.cpp
 
 #-------------------------------------------------------------------------
 # DO NOT DELETE THIS LINE -- make depend depends on it.
-- 
1.7.5.3

