From: Bernhard Walle <bernhard@bwalle.de>
Date: Sun, 11 Mar 2012 20:03:50 +0100
Subject: [PATCH] make ubifs optional

Signed-off-by: Bernhard Walle <bernhard@bwalle.de>
---
 Makefile |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 17a1216..5e1692f 100644
--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,9 @@ UBI_BINS = \
 	ubidetach ubinize ubiformat ubirename mtdinfo ubirsvol
 
 BINS = $(MTD_BINS)
+ifneq ($(WITHOUT_MKUBIFS), 1)
 BINS += mkfs.ubifs/mkfs.ubifs
+endif
 BINS += $(addprefix ubi-utils/,$(UBI_BINS))
 SCRIPTS = flash_eraseall
 
