From: Ladislav Michl <ladis@linux-mips.org>
Date: Tue, 3 Aug 2010 13:00:00 +0200
Subject: [PATCH] make mkubifs optional

Signed-off-by: Ladislav Michl <ladis@linux-mips.org>
---
 Makefile |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: mtd-utils-1.4.6/Makefile
===================================================================
--- mtd-utils-1.4.6.orig/Makefile
+++ mtd-utils-1.4.6/Makefile
@@ -13,7 +13,11 @@ else
   JFFS2_COMPR_LZO_O := compr_lzo.o
 endif
 
-SUBDIRS = lib ubi-utils mkfs.ubifs
+SUBDIRS = lib ubi-utils
+ifneq ($(WITHOUT_MKUBIFS), 1)
+  SUBDIRS += mkfs.ubifs
+endif
+
 TESTS = tests
 
 TARGETS = ftl_format flash_erase nanddump doc_loadbios \
