Index: DirectFB-0.9.24/configure.in
===================================================================
--- DirectFB-0.9.24.orig/configure.in
+++ DirectFB-0.9.24/configure.in
@@ -300,12 +300,12 @@ if test "$have_kos" = "no"; then
     AC_CHECK_LIB(c_r, pthread_attr_init, ,
       AC_MSG_ERROR([
 *** DirectFB requires phtreads in libc_r.]))
-    THREADLIB="-lc_r"
+    THREADLIB="-Wl,-lc_r"
   else
     AC_CHECK_LIB(pthread, pthread_attr_init, ,
       AC_MSG_ERROR([
 *** DirectFB requires libpthread.]))
-    THREADLIB="-lpthread"
+    THREADLIB="-Wl,-lpthread"
   fi
 fi
 
@@ -572,7 +572,7 @@ if test "$enable_sysfs" = "yes"; then
 	AC_CHECK_HEADER(sysfs/libsysfs.h,
 	use_sysfs=yes
         AC_DEFINE(USE_SYSFS,1,[Define to 1 to build with sysfs support.])
-	SYSFS_LIBS='-lsysfs',
+	SYSFS_LIBS='-Wl,-lsysfs',
 	AC_MSG_WARN([
 *** libsysfs header files not found -- Building without sysfs support.]))
     ],[
@@ -601,7 +601,7 @@ if test "$enable_jpeg" = "yes"; then
         jpeg_ok=no)
       if test "$jpeg_ok" = yes; then
 	JPEG=yes
-	LIBJPEG='-ljpeg'
+	LIBJPEG='-Wl,-ljpeg'
       else
 	JPEG=no
 	AC_MSG_WARN([
@@ -637,7 +637,7 @@ if test "$enable_zlib" = "yes"; then
 	AC_CHECK_HEADER(zlib.h,
 	use_zlib=yes
         AC_DEFINE(USE_ZLIB,1,[Define to 1 to build with zlib compression.])
-	ZLIB_LIBS='-lz',
+	ZLIB_LIBS='-Wl,-lz',
 	AC_MSG_ERROR([
 *** libz header files not found.]))
     ],[
@@ -661,7 +661,7 @@ if test "$enable_png" = "yes"; then
     AC_CHECK_LIB(z, gzsetparams,
       [
 	AC_CHECK_HEADER(zlib.h,
-	  ZLIB_LIBS='-lz',
+	  ZLIB_LIBS='-Wl,-lz',
 	  AC_MSG_WARN([
 *** libz header files not found. PNG image provider will not be built.]))
       ],[
@@ -681,7 +681,7 @@ if test "$enable_png" = "yes"; then
         AC_MSG_WARN([
 *** PNG library not found. PNG image provider will not be built.])
 	],
-      $ZLIB_LIBS -lm)
+      $ZLIB_LIBS -Wl,-lm)
     if test "$png_ok" = yes; then
       AC_MSG_CHECKING([for png_structp in png.h])
       AC_TRY_COMPILE([#include <png.h>],
@@ -693,7 +693,7 @@ if test "$enable_png" = "yes"; then
       AC_MSG_RESULT($png_ok)
       if test "$png_ok" = yes; then
 	PNG=yes
-        LIBPNG="-lpng $ZLIB_LIBS -lm"
+        LIBPNG="-Wl,-lpng $ZLIB_LIBS -Wl,-lm"
       else
 	PNG=no
         AC_MSG_WARN([
