Index: DirectFB-0.9.24/configure
===================================================================
--- DirectFB-0.9.24.orig/configure
+++ DirectFB-0.9.24/configure
@@ -21618,7 +21618,7 @@ if test $ac_cv_lib_c_r_pthread_attr_init
 #define HAVE_LIBC_R 1
 _ACEOF
 
-  LIBS="-lc_r $LIBS"
+  LIBS="-Wl,-lc_r $LIBS"
 
 else
   { { echo "$as_me:$LINENO: error:
@@ -21628,7 +21628,7 @@ echo "$as_me: error:
    { (exit 1); exit 1; }; }
 fi
 
-    THREADLIB="-lc_r"
+    THREADLIB="-Wl,-lc_r"
   else
 
 echo "$as_me:$LINENO: checking for pthread_attr_init in -lpthread" >&5
@@ -21709,7 +21709,7 @@ echo "$as_me: error:
    { (exit 1); exit 1; }; }
 fi
 
-    THREADLIB="-lpthread"
+    THREADLIB="-Wl,-lpthread"
   fi
 fi
 
@@ -22683,7 +22683,7 @@ cat >>confdefs.h <<\_ACEOF
 #define USE_SYSFS 1
 _ACEOF
 
-	SYSFS_LIBS='-lsysfs'
+	SYSFS_LIBS='-Wl,-lsysfs'
 else
   { echo "$as_me:$LINENO: WARNING:
 *** libsysfs header files not found -- Building without sysfs support." >&5
@@ -22932,7 +22932,7 @@ fi
 
       if test "$jpeg_ok" = yes; then
 	JPEG=yes
-	LIBJPEG='-ljpeg'
+	LIBJPEG='-Wl,-ljpeg'
       else
 	JPEG=no
 	{ echo "$as_me:$LINENO: WARNING:
@@ -23186,7 +23186,7 @@ cat >>confdefs.h <<\_ACEOF
 #define USE_ZLIB 1
 _ACEOF
 
-	ZLIB_LIBS='-lz'
+	ZLIB_LIBS='-Wl,-lz'
 else
   { { echo "$as_me:$LINENO: error:
 *** libz header files not found." >&5
@@ -23424,7 +23424,7 @@ echo "${ECHO_T}$ac_cv_header_zlib_h" >&6
 
 fi
 if test $ac_cv_header_zlib_h = yes; then
-  ZLIB_LIBS='-lz'
+  ZLIB_LIBS='-Wl,-lz'
 else
   { echo "$as_me:$LINENO: WARNING:
 *** libz header files not found. PNG image provider will not be built." >&5
@@ -23450,7 +23450,7 @@ if test "${ac_cv_lib_png_png_read_info+s
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng $ZLIB_LIBS -lm $LIBS"
+LIBS="-Wl,-lpng $ZLIB_LIBS -lm $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -23719,7 +23719,7 @@ rm -f conftest.err conftest.$ac_objext c
 echo "${ECHO_T}$png_ok" >&6
       if test "$png_ok" = yes; then
 	PNG=yes
-        LIBPNG="-lpng $ZLIB_LIBS -lm"
+        LIBPNG="-Wl,-lpng $ZLIB_LIBS -Wl,-lm"
       else
 	PNG=no
         { echo "$as_me:$LINENO: WARNING:
