---
 misc/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: misc/Makefile
===================================================================
--- misc/Makefile.orig
+++ misc/Makefile
@@ -19,7 +19,7 @@ rtacct: rtacct.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -o rtacct rtacct.c $(LIBNETLINK) -lm
 
 arpd: arpd.c
-	$(CC) $(CFLAGS) -I$(DBM_INCLUDE) $(LDFLAGS) -o arpd arpd.c $(LIBNETLINK) -ldb -lpthread
+	$(CC) $(CFLAGS) -I$(DBM_INCLUDE) $(LDFLAGS) -o arpd arpd.c $(LIBNETLINK) -ldb44 -lpthread
 
 ssfilter.c: ssfilter.y
 	bison ssfilter.y -o ssfilter.c
