From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Tue, 24 Aug 2010 21:33:28 +0200
Subject: [PATCH] Makefile.in: disable target which will fail while cross compiling

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 Makefile.in |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 24d4bd6..3ce8512 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -88,7 +88,7 @@ PUBLIBOBJS	=	mxml-attr.o mxml-entity.o mxml-file.o mxml-index.o \
 			mxml-node.o mxml-search.o mxml-set.o
 LIBOBJS		=	$(PUBLIBOBJS) mxml-private.o mxml-string.o
 OBJS		=	mxmldoc.o testmxml.o $(LIBOBJS)
-TARGETS		=	$(LIBMXML) mxmldoc testmxml mxml.xml doc/mxml.man
+TARGETS		=	$(LIBMXML) mxmldoc #testmxml mxml.xml doc/mxml.man
 
 
 #
@@ -144,10 +144,10 @@ install:	$(TARGETS) install-$(LIBMXML) install-libmxml.a
 	$(INSTALL_DATA) mxml.h $(BUILDROOT)$(includedir)
 	$(INSTALL_DIR) $(BUILDROOT)$(libdir)/pkgconfig
 	$(INSTALL_DATA) mxml.pc $(BUILDROOT)$(libdir)/pkgconfig
-	$(INSTALL_DIR) $(BUILDROOT)$(mandir)/man1
-	$(INSTALL_MAN) doc/mxmldoc.man $(BUILDROOT)$(mandir)/man1/mxmldoc.1
-	$(INSTALL_DIR) $(BUILDROOT)$(mandir)/man3
-	$(INSTALL_MAN) doc/mxml.man $(BUILDROOT)$(mandir)/man3/mxml.3
+#	$(INSTALL_DIR) $(BUILDROOT)$(mandir)/man1
+#	$(INSTALL_MAN) doc/mxmldoc.man $(BUILDROOT)$(mandir)/man1/mxmldoc.1
+#	$(INSTALL_DIR) $(BUILDROOT)$(mandir)/man3
+#	$(INSTALL_MAN) doc/mxml.man $(BUILDROOT)$(mandir)/man3/mxml.3
 
 install-libmxml.a:
 	$(INSTALL_DIR) $(BUILDROOT)$(libdir)
-- 
1.7.1

