Index: rules/crosstool.make
===================================================================
--- rules/crosstool.make	(revision 2507)
+++ rules/crosstool.make	(working copy)
@@ -19,7 +19,7 @@
 #
 # Paths and names
 #
-CROSSTOOL_VERSION	= 0.28-rc37
+CROSSTOOL_VERSION	= 0.31
 CROSSTOOL		= crosstool-$(CROSSTOOL_VERSION)
 CROSSTOOL_SUFFIX	= tar.gz
 CROSSTOOL_URL		= http://www.kegel.com/crosstool/$(CROSSTOOL).$(CROSSTOOL_SUFFIX)
@@ -88,7 +88,7 @@
 
 ifdef PTXCONF_OPT_PPC405
 CROSSTOOL_GCC_EXTRA_CONFIG	= "--with-cpu=405 --enable-cxx-flags=-mcpu=405"
-CROSSTOOL_TARGET_CFLAGS		= "-O -mcpu=405"
+CROSSTOOL_TARGET_CFLAGS		= "-O2 -mcpu=405"
 endif
 
 CROSSTOOL_GCCLANG		=  c
@@ -105,13 +105,39 @@
 ifdef PTXCONF_GLIBC
 CROSSTOOL_LIBC_DIR		= $(GLIBC)
 CROSSTOOL_LIBC			= glibc
+
+ifdef PTXCONF_GLIBC_2_3_4
+CROSSTOOL_LIBC_VERSION = GLIBC_DIR=glibc-2.3.4
+else 
+ifdef PTXCONF_GLIBC_2_3_3
+CROSSTOOL_LIBC_VERSION = GLIBC_DIR=glibc-2.3.3
+else 
+ifdef PTXCONF_GLIBC_2_3_2
+CROSSTOOL_LIBC_VERSION = GLIBC_DIR=glibc-2.3.2
+else 
+ifdef PTXCONF_GLIBC_2_2_5
+CROSSTOOL_LIBC_VERSION = GLIBC_DIR=glibc-2.2.5
+else 
+ifdef PTXCONF_GLIBC_2_2_4
+CROSSTOOL_LIBC_VERSION = GLIBC_DIR=glibc-2.2.4
+else 
+ifdef PTXCONF_GLIBC_2_2_3
+CROSSTOOL_LIBC_VERSION = GLIBC_DIR=glibc-2.2.3
+endif 
+endif 
+endif 
+endif 
+endif 
+endif 
 else
 ifdef PTXCONF_UCLIBC
 CROSSTOOL_LIBC_DIR		= $(UCLIBC)
 CROSSTOOL_LIBC			= uclibc
+CROSSTOOL_LIBC_VERSION 		=
 endif
 endif
 
+
 $(STATEDIR)/crosstool.prepare: $(crosstool_prepare_deps)
 	@$(call targetinfo, $@)
 	touch $@
@@ -155,6 +181,7 @@
 		\
 		export KERNELCONFIG=$(subst $(quote),,$(CROSSTOOL_DIR)/$(PTXCONF_CROSSTOOL_KERNELCONFIG)); \
 		\
+		$(CROSSTOOL_LIBC_VERSION) \
 		TARGET=$(subst $(quote),,$(PTXCONF_GNU_TARGET)) \
 		TARGET_CFLAGS="$(call remove_quotes,$(CROSSTOOL_TARGET_CFLAGS))" \
 		GCC_EXTRA_CONFIG=$(CROSSTOOL_GCC_EXTRA_CONFIG) \
