From: Marc Kleine-Budde <mkl@pengutronix.de>
Subject: Don't mess with /etc/ld.so.conf

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>

---
 Makefile.am |   18 ------------------
 1 file changed, 18 deletions(-)

Index: commoncpp2-1.3.25/Makefile.am
===================================================================
--- commoncpp2-1.3.25.orig/Makefile.am
+++ commoncpp2-1.3.25/Makefile.am
@@ -34,23 +34,5 @@ macosx:
 darwin:
 	(cd src ; make macosx)
 
-install-data-local:
-	@if test -f /etc/ld.so.conf \
-	  -a `echo $(libdir) | grep ^/usr`XXX != "XXX" ; then \
-		if ! grep -q "$(libdir)" /etc/ld.so.conf; then \
-			echo "------------------------------------------------------------------------------" ;\
-			echo "Hmm, looks like you forgot to add $(libdir) to /etc/ld.so.conf." ;\
-			echo "I'll try to fix it for you (but pay attention to libtool's output" ;\
-			echo "next time, ok?)" ;\
-			echo "$(libdir)" >> /etc/ld.so.conf && \
-				echo "Ok, all fixed." \
-				|| \
-				(echo "" ;\
-				echo "Well, that didn't work.  Remember to add $(libdir) to" ;\
-				echo "/etc/ld.so.conf if you want to use this library.") ;\
-			echo "------------------------------------------------------------------------------" ;\
-			ldconfig ;\
-		fi ;\
-	fi
 
 
