From e6a499abadedab27c98b17c0d7c0cbe2dbabed51 Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Mon, 27 Jul 2009 22:24:28 +0200
Subject: [PATCH 3/3] [Makefile.in] don't strip during install

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 5a61be8..8beb56c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -42,6 +42,6 @@ distclean: clean
 
 install: $(PRODUCT)
 	$(INSTALL) -d -m 755 $(DESTDIR)$(bindir)
-	$(INSTALL) -s -m 755 $(PRODUCT) $(DESTDIR)$(bindir)
+	$(INSTALL)    -m 755 $(PRODUCT) $(DESTDIR)$(bindir)
 	$(INSTALL) -d -m 755 $(DESTDIR)$(mandir)/man1
 	$(INSTALL) -m 644 $(PRODUCT).1 $(DESTDIR)$(mandir)/man1
-- 
1.6.3.3

