---
 Makefile.in |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -1,6 +1,6 @@
-INSTALL = @INSTALL@
-INSTALL_PROGRAM = @INSTALL_PROGRAM@
-INSTALL_DATA = @INSTALL_DATA@
+INSTALL = @INSTALL@ -D
+INSTALL_PROGRAM = @INSTALL_PROGRAM@ -D
+INSTALL_DATA = @INSTALL_DATA@ -D
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
