From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Fri, 1 Jan 2010 16:08:33 +0100
Subject: [PATCH] inotify-syscalls: don't define inotofy syscalls if already defined

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 plugins/fmon/inotify-syscalls.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/plugins/fmon/inotify-syscalls.h b/plugins/fmon/inotify-syscalls.h
index 8cfa879..495d592 100644
--- a/plugins/fmon/inotify-syscalls.h
+++ b/plugins/fmon/inotify-syscalls.h
@@ -3,6 +3,7 @@
 
 #include <sys/syscall.h>
 
+#ifndef __NR_inotify_init
 #if defined(__i386__)
 # define __NR_inotify_init	291
 # define __NR_inotify_add_watch	292
@@ -46,6 +47,7 @@
 #else
 # error "Unsupported architecture!"
 #endif
+#endif
 
 static inline int inotify_init(void)
 {
-- 
1.7.2.3

