From 6be918d0ae15ca5f8b3b3879c496205bc8a11c2b Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Mon, 14 Dec 2009 01:32:11 +0100
Subject: [PATCH] init: make FEATURE_EXTRA_QUIET more consistent. +1 byte

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
 init/init.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/init/init.c b/init/init.c
index 1413685..3748a15 100644
--- a/init/init.c
+++ b/init/init.c
@@ -86,7 +86,6 @@ static const char *log_console = VC_5;
 enum {
 	L_LOG = 0x1,
 	L_CONSOLE = 0x2,
-	MAYBE_CONSOLE = L_CONSOLE * !ENABLE_FEATURE_EXTRA_QUIET,
 #ifndef RB_HALT_SYSTEM
 	RB_HALT_SYSTEM = 0xcdef0123, /* FIXME: this overflows enum */
 	RB_ENABLE_CAD = 0x89abcdef,
@@ -856,8 +855,10 @@ int init_main(int argc UNUSED_PARAM, char **argv)
 	if (argv[1])
 		xsetenv("RUNLEVEL", argv[1]);
 
+#if ENABLE_FEATURE_EXTRA_QUIET
 	/* Hello world */
-	message(MAYBE_CONSOLE | L_LOG, "init started: %s", bb_banner);
+	message(L_CONSOLE | L_LOG, "init started: %s", bb_banner);
+#endif
 
 	/* Make sure there is enough memory to do something useful. */
 	if (ENABLE_SWAPONOFF) {
-- 
1.5.6.5

