Index: qt-embedded-free-3.3.4/src/kernel/qfont_qws.cpp
===================================================================
--- qt-embedded-free-3.3.4.orig/src/kernel/qfont_qws.cpp
+++ qt-embedded-free-3.3.4/src/kernel/qfont_qws.cpp
@@ -133,7 +133,7 @@ void QFontPrivate::load( QFont::Script )
 	req.pointSize = req.pixelSize*10;
 
     if ( ! engineData ) {
-	QFontCache::Key key( req, QFont::NoScript, (int)paintdevice );
+	QFontCache::Key key( req, QFont::NoScript, (long)paintdevice );
 
 	// look for the requested font in the engine data cache
 	engineData = QFontCache::instance->findEngineData( key );
