diff -ur /work/jb/Pengutronix/ptxdist-projects/Winterm/build-target/portmap_5beta/daemon.c ./daemon.c
--- /work/jb/Pengutronix/ptxdist-projects/Winterm/build-target/portmap_5beta/daemon.c	2006-08-30 21:50:00.196845755 +0200
+++ ./daemon.c	2006-08-24 21:06:29.832427000 +0200
@@ -37,6 +37,7 @@
 
 #include <fcntl.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 
 /* From paths.h */
diff -ur /work/jb/Pengutronix/ptxdist-projects/Winterm/build-target/portmap_5beta/Makefile ./Makefile
--- /work/jb/Pengutronix/ptxdist-projects/Winterm/build-target/portmap_5beta/Makefile	2006-08-30 21:50:00.186845476 +0200
+++ ./Makefile	2006-08-24 21:08:55.122427000 +0200
@@ -15,9 +15,9 @@
 # no access control tables. The local system, since it runs the portmap
 # daemon, is always treated as an authorized host.
 
-HOSTS_ACCESS= -DHOSTS_ACCESS
+#HOSTS_ACCESS= -DHOSTS_ACCESS
 #WRAP_LIB = $(WRAP_DIR)/libwrap.a
-WRAP_LIB = -lwrap
+#WRAP_LIB = -lwrap
 
 # Comment out if your RPC library does not allocate privileged ports for
 # requests from processes with root privilege, or the new portmap will
@@ -86,7 +86,8 @@
 
 # Auxiliary object files that may be missing from your C library.
 #
-AUX	= daemon.o strerror.o
+AUX	= 
+#daemon.o strerror.o
 
 # NEXTSTEP is a little different. The following seems to work with NS 3.2
 #
Nur in .: my_mod.patch.
diff -ur /work/jb/Pengutronix/ptxdist-projects/Winterm/build-target/portmap_5beta/strerror.c ./strerror.c
--- /work/jb/Pengutronix/ptxdist-projects/Winterm/build-target/portmap_5beta/strerror.c	1992-06-11 22:53:13.000000000 +0200
+++ ./strerror.c	2006-08-24 21:07:51.502427000 +0200
@@ -36,7 +36,7 @@
 #endif /* LIBC_SCCS and not lint */
 
 #include <string.h>
-
+#if 0
 char *
 strerror(num)
 	int num;
@@ -65,3 +65,4 @@
 	}
 	return(ebuf);
 }
+#endif
