From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 16 Jun 2014 15:29:58 +0200
Subject: [PATCH] configure: includes are needed for host tools when
 cross-compiling

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 6592e70..18363f3 100644
--- a/configure.in
+++ b/configure.in
@@ -1328,7 +1328,7 @@ if test $cross_compiling = no; then
    BUILD_CFLAGS="$CFLAGS $CPPFLAGS $INCLUDES -DHAVE_CONFIG_H"
    BUILD_LDFLAGS="$LDFLAGS"
 else
-   BUILD_CFLAGS=
+   BUILD_CFLAGS='-I. -I$(top_builddir)/lib -I$(top_srcdir)/lib'
    BUILD_LDFLAGS=
 fi
 AC_SUBST(BUILD_CFLAGS)
