Index: libXrender-0.9.4/src/Makefile.am
===================================================================
--- libXrender-0.9.4.orig/src/Makefile.am	2007-08-31 22:44:10.000000000 +0200
+++ libXrender-0.9.4/src/Makefile.am	2007-10-18 19:16:26.000000000 +0200
@@ -17,7 +17,7 @@
                         Xrender.c \
                         Xrenderint.h
 
-libXrender_la_LIBADD = @X_LIBS@ -lX11
+libXrender_la_LIBADD = @X_LIBS@
 
 # -version-number requires libtool >= 1.5
 libXrender_la_LDFLAGS = -version-number 1:3:0 -no-undefined 
Index: libXrender-0.9.4/src/Makefile.in
===================================================================
--- libXrender-0.9.4.orig/src/Makefile.in	2007-09-05 17:20:48.000000000 +0200
+++ libXrender-0.9.4/src/Makefile.in	2007-10-18 19:16:26.000000000 +0200
@@ -215,7 +215,7 @@
                         Xrender.c \
                         Xrenderint.h
 
-libXrender_la_LIBADD = @X_LIBS@ -lX11
+libXrender_la_LIBADD = @X_LIBS@
 
 # -version-number requires libtool >= 1.5
 libXrender_la_LDFLAGS = -version-number 1:3:0 -no-undefined 
