From: Christos Zoulas <christos@zoulas.com>
Date: Tue, 26 Jun 2018 20:29:29 +0000
Subject: [PATCH] Fix pasto (Clemens Gruber)

---
 src/seccomp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/seccomp.c b/src/seccomp.c
index 6da7d658deb9..a5abb4a159f9 100644
--- a/src/seccomp.c
+++ b/src/seccomp.c
@@ -27,7 +27,7 @@
 #include "file.h"
 
 #ifndef	lint
-FILE_RCSID("@(#)$File: seccomp.c,v 1.5 2018/06/23 16:19:02 christos Exp $")
+FILE_RCSID("@(#)$File: seccomp.c,v 1.6 2018/06/26 20:29:29 christos Exp $")
 #endif	/* lint */
 
 #if HAVE_LIBSECCOMP
@@ -166,7 +166,7 @@ enable_sandbox_full(void)
 	ALLOW_RULE(fcntl);  
  	ALLOW_RULE(fcntl64);  
 	ALLOW_RULE(fstat);
- 	ALLOW_RULE(fcntl64);  
+ 	ALLOW_RULE(fstat64);  
 	ALLOW_RULE(getdents);
 #ifdef __NR_getdents64
 	ALLOW_RULE(getdents64);
