From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 29 Mar 2012 16:43:04 +0200
Subject: [PATCH] opkg.py: cast lines from controlfile as string

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 opkg.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opkg.py b/opkg.py
index 84583be..6c39616 100644
--- a/opkg.py
+++ b/opkg.py
@@ -206,7 +206,7 @@ class Package:
         while 1:
             if not line: break
             line = line.rstrip()
-            lineparts = re.match(r'([\w-]*?):\s*(.*)', line)
+            lineparts = re.match(r'([\w-]*?):\s*(.*)', str(line))
             if lineparts:
                 name = lineparts.group(1).lower()
                 value = lineparts.group(2)
