From: Marc Olzheim <marc@iphion.nl>
Date: Thu, 4 Oct 2012 11:01:10 +0200
Subject: [PATCH] opkg-make-index: fix mis-indented else:

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 opkg-make-index | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opkg-make-index b/opkg-make-index
index 1c3a8e1..44fa64d 100755
--- a/opkg-make-index
+++ b/opkg-make-index
@@ -213,7 +213,7 @@ if filelist_filename:
             (h,t) = os.path.split(fn)
             if not t: continue
             if t not in files: files[t] = name+':'+fn
-        else: files[t] = files[t] + ',' + name+':'+fn
+            else: files[t] = files[t] + ',' + name+':'+fn
 
     tmp_filelist_filename = ("%s.%d" % (filelist_filename, os.getpid()))
     f = open(tmp_filelist_filename, "w")
