From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 3 Feb 2014 11:27:53 +0100
Subject: [PATCH] all programs that are linked with -lrttest must depend on
 librttest.a

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9a55fc5..4b4c5aa 100644
--- a/Makefile
+++ b/Makefile
@@ -68,14 +68,14 @@ cyclictest: cyclictest.o librttest.a
 signaltest: signaltest.o librttest.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
-pi_stress: pi_stress.o
+pi_stress: pi_stress.o librttest.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 hwlatdetect:  src/hwlatdetect/hwlatdetect.py
 	chmod +x src/hwlatdetect/hwlatdetect.py
 	ln -s src/hwlatdetect/hwlatdetect.py hwlatdetect
 
-rt-migrate-test: rt-migrate-test.o
+rt-migrate-test: rt-migrate-test.o librttest.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 ptsematest: ptsematest.o librttest.a
@@ -96,7 +96,7 @@ sendme: sendme.o librttest.a
 pip_stress: pip_stress.o librttest.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
-hackbench: hackbench.o
+hackbench: hackbench.o librttest.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 librttest.a: rt-utils.o error.o rt-get_cpu.o
