diff -urN modutils-2.4.27/config.sub modutils-2.4.27-ptx/config.sub
--- modutils-2.4.27/config.sub	2003-10-27 03:54:45.000000000 +0100
+++ modutils-2.4.27-ptx/config.sub	2005-05-08 14:16:21.000000000 +0200
@@ -283,10 +283,10 @@
 	  basic_machine=$basic_machine-pc
 	  ;;
 	# Object if more than one company name word.
-	*-*-*)
-		echo Invalid configuration \`$1\': machine \`$basic_machine\' not recognized 1>&2
-		exit 1
-		;;
+#	*-*-*)
+#		echo Invalid configuration \`$1\': machine \`$basic_machine\' not recognized 1>&2
+#		exit 1
+#		;;
 	# Recognize the basic CPU types with company name.
 	580-* \
 	| a29k-* \
@@ -1118,7 +1118,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* | -uclibc* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
