--- strace-4.4.98/syscall.c	Wed Jun  4 01:29:04 2003
+++ strace-4.4.98/syscall.c	Thu Jul 10 09:45:22 2003
@@ -1657,8 +1657,8 @@
 		return -1;
 #else /* !M68K */
 #ifdef ARM
-	r0 = error ? -error : rval;
-	if (ptrace(PTRACE_POKEUSER, tcp->pid, (char*)(4*0), r0) < 0)
+	regs.ARM_r0 = error ? -error : rval;
+	if (ptrace(PTRACE_POKEUSER, tcp->pid, (char*)(4*0), regs.ARM_r0) < 0)
 		return -1;
 #else /* !ARM */
 #ifdef ALPHA
