From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Wed, 6 Jul 2011 10:39:24 +0200
Subject: [PATCH] improve multi-lib toolchain handling

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 makefile |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/makefile b/makefile
index ba5e3ba..954c65c 100644
--- a/makefile
+++ b/makefile
@@ -6107,12 +6107,17 @@ linux:
 	then echo -lresolv; fi` \
 	`if test -f /usr/lib64/libcrypt.a || test -f /usr/lib64/libcrypt.so \
 	|| test -f /usr/lib/libcrypt.a || test -f /usr/lib/libcrypt.so \
+	|| test -f /usr/lib/i386-linux-gnu/libcrypt.a \
+	|| test -f /usr/lib/i386-linux-gnu/libcrypt.so \
 	|| ls /lib/x86_64-linux-gnu/libcrypt.* > /dev/null 2> /dev/null; \
 	then echo -lcrypt; fi` \
 	`if test -f /usr/lib64/liblockdev.a || \
 	test -f /usr/lib64/liblockdev.so || \
 	test -f /usr/lib/liblockdev.a || \
-	test -f /usr/lib/liblockdev.so; \
+	test -f /usr/lib/liblockdev.so \
+	|| test -f /usr/lib/i386-linux-gnu/liblockdev.a \
+	|| test -f /usr/lib/i386-linux-gnu/liblockdev.so \
+	|| ls /lib/x86_64-linux-gnu/liblockdev.* > /dev/null 2> /dev/null; \
 	then echo -llockdev; fi`" \
 	linuxa
 
