From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 8 Jul 2010 14:21:32 +0200
Subject: [PATCH 1/2] make dconf editor optional

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 Makefile.am  |    6 +++++-
 configure.ac |   11 ++++++++---
 2 files changed, 13 insertions(+), 4 deletions(-)

Index: dconf-0.5/Makefile.am
===================================================================
--- dconf-0.5.orig/Makefile.am
+++ dconf-0.5/Makefile.am
@@ -1,5 +1,9 @@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = gvdb service gsettings tests client bin engine common docs editor
+SUBDIRS = gvdb service gsettings tests client bin engine common docs
+
+if ENABLE_EDITOR
+SUBDIRS += editor
+endif
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
Index: dconf-0.5/configure.ac
===================================================================
--- dconf-0.5.orig/configure.ac
+++ dconf-0.5/configure.ac
@@ -22,9 +22,14 @@ GTK_DOC_CHECK([1.15])
 
 # Dependencies
 PKG_CHECK_MODULES(gio, gio-2.0 >= 2.25.12)
-PKG_CHECK_MODULES(gtk, gtk+-2.0)
-PKG_CHECK_MODULES(gee, gee-1.0)
-PKG_CHECK_MODULES(libxml, libxml-2.0)
+
+AC_ARG_ENABLE(editor, [  --disable-editor Disable the dconf editor, [default=yes]],,[enable_editor=yes])
+if test "x$enable_editor" = "xyes"; then
+  PKG_CHECK_MODULES(gtk, gtk+-2.0)
+  PKG_CHECK_MODULES(gee, gee-1.0)
+  PKG_CHECK_MODULES(libxml, libxml-2.0)
+fi
+AM_CONDITIONAL(ENABLE_EDITOR, test "x$enable_editor" = "xyes")
 
 AC_ARG_WITH(gio_modules_dir, [  --with-gio-modules-dir=PATH choose directory for the GIO module, [default=LIBDIR/gio/modules]], giomodulesdir="$withval", giomodulesdir=${libdir}/gio/modules)
 AC_SUBST(giomodulesdir)
