Index: fontconfig-2.3.2/configure.in
===================================================================
--- fontconfig-2.3.2.orig/configure.in
+++ fontconfig-2.3.2/configure.in
@@ -222,17 +222,17 @@ no)
 	yes)
 		case "$expat" in
 		yes)
-			EXPAT_LIBS="-lexpat"
+			EXPAT_LIBS="-Wl,-lexpat"
 			;;
 		*)
-			EXPAT_LIBS="-L$expat/lib -lexpat"
+			EXPAT_LIBS="-L$expat/lib -Wl,-lexpat"
 			;;
 		esac
 		;;
 	no)
 		;;
 	*)
-		EXPAT_LIBS="-L$expat_lib -lexpat"
+		EXPAT_LIBS="-L$expat_lib -Wl,-lexpat"
 		;;
 	esac
 
