From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Tue, 28 Apr 2009 14:05:30 +0200
Subject: [PATCH] use PGEN_FOR_BUILD

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 Makefile.pre.in |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index f8a8d36..01d4c98 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -205,6 +205,7 @@ GRAMMAR_INPUT=	$(srcdir)/Grammar/Grammar
 ##########################################################################
 # Parser
 PGEN=		Parser/pgen$(EXE)
+PGEN_FOR_BUILD=	$(PGEN)
 
 POBJS=		\
 		Parser/acceler.o \
@@ -515,9 +516,9 @@ Modules/python.o: $(srcdir)/Modules/python.c
 	$(MAINCC) -c $(PY_CFLAGS) -o $@ $(srcdir)/Modules/python.c
 
 
-$(GRAMMAR_H) $(GRAMMAR_C): $(PGEN) $(GRAMMAR_INPUT)
+$(GRAMMAR_H) $(GRAMMAR_C): $(PGEN_FOR_BUILD) $(GRAMMAR_INPUT)
 		-@$(INSTALL) -d Include
-		-$(PGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
+		-$(PGEN_FOR_BUILD) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
 
 $(PGEN):	$(PGENOBJS)
 		$(CC) $(OPT) $(LDFLAGS) $(PGENOBJS) $(LIBS) -o $(PGEN)
-- 
1.7.2.3

