From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Thu Jul 21 12:29:58 CEST 2011
Subject: [PATCH] python: don't call cross python

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
 Makefile.pre.in |    2 +-
 configure.in    |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

Index: Python-2.6.6/Makefile.pre.in
===================================================================
--- Python-2.6.6.orig/Makefile.pre.in
+++ Python-2.6.6/Makefile.pre.in
@@ -918,7 +918,7 @@ libinstall:	build_all $(srcdir)/Lib/$(PL
 $(srcdir)/Lib/$(PLATDIR):
 	mkdir $(srcdir)/Lib/$(PLATDIR)
 	cp $(srcdir)/Lib/plat-generic/regen $(srcdir)/Lib/$(PLATDIR)/regen
-	export PATH; PATH="`pwd`:$$PATH"; \
+	export PATH; \
 	export PYTHONPATH; PYTHONPATH="`pwd`/Lib"; \
 	export DYLD_FRAMEWORK_PATH; DYLD_FRAMEWORK_PATH="`pwd`"; \
 	export EXE; EXE="$(BUILDEXE)"; \
