From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 27 Aug 2012 12:29:15 +0200
Subject: [PATCH] also build _sha256/_sha512 if openssl is not found

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 setup.py |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 8c86ed8..fc6cd8a 100644
--- a/setup.py
+++ b/setup.py
@@ -779,7 +779,9 @@ class PyBuildExt(build_ext):
                             depends = ['md5.h']) )
             missing.append('_hashlib')
 
-        if (openssl_ver < 0x00908000):
+        if (ssl_incs is None or
+            ssl_libs is None or
+            openssl_ver < 0x00908000):
             # OpenSSL doesn't do these until 0.9.8 so we'll bring our own hash
             exts.append( Extension('_sha256', ['sha256module.c']) )
             exts.append( Extension('_sha512', ['sha512module.c']) )
