diff -urN pcmciautils-006-orig/src/pccardctl.c pcmciautils-006/src/pccardctl.c
--- pcmciautils-006-orig/src/pccardctl.c	2005-06-17 22:14:26.000000000 +0200
+++ pcmciautils-006/src/pccardctl.c	2006-04-04 08:02:34.000000000 +0200
@@ -63,41 +63,58 @@
 static char * pccardctl_get_string(unsigned long socket_no, const char *in_file)
 {
 	int ret, len;
-	char value[SYSFS_PATH_MAX];
-	char file[SYSFS_PATH_MAX];
+	char path[SYSFS_PATH_MAX];
 	char *result;
-
-	snprintf(file, SYSFS_PATH_MAX, "/sys/bus/pcmcia/devices/%lu.0/%s",
+	struct sysfs_attribute *sysfs_attr;
+	
+	snprintf(path, SYSFS_PATH_MAX, "/sys/bus/pcmcia/devices/%lu.0/%s",
 		 socket_no, in_file);
-	ret = sysfs_read_attribute_value(file, value, SYSFS_PATH_MAX);
+	sysfs_attr = sysfs_open_attribute(path);
+	if (!sysfs_attr)
+		return NULL;
+
+	ret = sysfs_read_attribute(sysfs_attr);
 	if (ret)
 		return NULL;
 
-	len = strlen(value);
-	if (len < 2)
+	len = strlen(sysfs_attr->value);
+	if (len < 2) {
+		sysfs_close_attribute(sysfs_attr);
 		return NULL;
+	}
 
 	result = malloc(sizeof(char) * len);
-	if (!result)
+	if (!result) {
+		sysfs_close_attribute(sysfs_attr);
 		return NULL;
+	}
 
-        strncpy(result, value, len);
+        strncpy(result, sysfs_attr->value, len);
         result[len - 1] = '\0';
 
+	sysfs_close_attribute(sysfs_attr);
+
 	return (result);
 }
 
 static int pccardctl_get_one(unsigned long socket_no, const char *in_file, unsigned int *result)
 {
 	int ret;
-	char value[SYSFS_PATH_MAX];
-	char file[SYSFS_PATH_MAX];
+	char path[SYSFS_PATH_MAX];
+	struct sysfs_attribute *sysfs_attr;
 
-	snprintf(file, SYSFS_PATH_MAX, "/sys/bus/pcmcia/devices/%lu.0/%s",
+	snprintf(path, SYSFS_PATH_MAX, "/sys/bus/pcmcia/devices/%lu.0/%s",
 		 socket_no, in_file);
-	ret = sysfs_read_attribute_value(file, value, SYSFS_PATH_MAX);
-	if (!ret)
-		sscanf(value, "0x%X", result);
+	sysfs_attr = sysfs_open_attribute(path);
+	if (!sysfs_attr)
+		return 0;
+
+	ret = sysfs_read_attribute(sysfs_attr);
+	if (ret)
+		return 0;
+	
+	sscanf(sysfs_attr->value, "0x%X", result);
+	sysfs_close_attribute(sysfs_attr);
 
 	return (ret);
 }
