From: the openembedded project <http://openembedded.net/>
Date: Fri, 22 Jan 2010 11:05:05 +0100
Subject: [PATCH] ipkg-utils-fix

---
 ipkg.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ipkg.py b/ipkg.py
index 02aab22..e175c8d 100644
--- a/ipkg.py
+++ b/ipkg.py
@@ -152,9 +152,9 @@ class Package:
             self.filename = os.path.basename(fn)
 	    ## sys.stderr.write("  extracting control.tar.gz from %s\n"% (fn,)) 
             if self.isdeb:
-                control = os.popen("ar p "+fn+" control.tar.gz | tar xfzO - '*control'","r")
+                control = os.popen("ar p "+fn+" control.tar.gz | tar xfzO - './control'","r")
             else:
-                control = os.popen("tar xfzO "+fn+" '*control.tar.gz' | tar xfzO - '*control'","r")
+                control = os.popen("tar xfzO "+fn+" 'control.tar.gz' | tar xfzO - './control'","r")
             line = control.readline()
             while 1:
                 if not line: break
-- 
1.7.2.3

