From f6efdc7de100f6177f3bbb70416e722ce1fda660 Mon Sep 17 00:00:00 2001
From: the openembedded project <http://openembedded.net/>
Date: Fri, 22 Jan 2010 11:05:12 +0100
Subject: [PATCH 4/9] arfile_even_alignment

---
 arfile.py |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arfile.py b/arfile.py
index 0bcd8d0..597cb3d 100644
--- a/arfile.py
+++ b/arfile.py
@@ -86,6 +86,8 @@ class ArFile:
                 return FileSection(self.f, self.f.tell(), size)
 
             # Skip data and loop
+            if size % 2:
+                size = size + 1
             data = self.f.seek(size, 1)
 #            print hex(f.tell())
 
-- 
1.6.6

