From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Fri, 22 Jan 2010 11:22:41 +0100
Subject: [PATCH] ipkg.py: don't use wildcards

it's not portable

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 ipkg.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ipkg.py b/ipkg.py
index d6f7e69..c951371 100644
--- a/ipkg.py
+++ b/ipkg.py
@@ -166,7 +166,7 @@ class Package:
         	except KeyError:
         	    control = tarf.extractfile("./control")
 	    else:
-		control = os.popen("tar --wildcards -xzO -f " + fn + " '*control.tar.gz' | tar xfzO - './control'", "r")
+		control = os.popen("tar -xzO -f " + fn + " './control.tar.gz' | tar xfzO - './control'", "r")
 
             self.read_control(control)
             control.close()
-- 
1.7.2.3

