From 58ae8050e076f2cd4a073b3090c71606160ea4eb Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Fri, 22 Jan 2010 11:27:37 +0100
Subject: [PATCH 9/9] Makefile: add prefix to python install

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 7093a0d..249c687 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ ipkg-compare-versions: ipkg-compare-versions.c
 
 install: ${UTILS}
 	cp ${UTILS} $(PREFIX)/bin
-	python setup.py install
+	python setup.py install --prefix=$(PREFIX)
 	chmod agu+rx ipkg-make-index
 	cp -f ipkg-make-index $(PREFIX)/bin
 	chmod agu+rx ipkg-list-fields
-- 
1.6.6

