From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: add upstream patch

From: http://ftp.de.debian.org/debian/pool/main/libm/libmad/libmad_0.15.1b-3.diff.gz

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
 configure.ac |    1 +
 1 file changed, 1 insertion(+)

Index: libmad-0.15.1b/configure.ac
===================================================================
--- libmad-0.15.1b.orig/configure.ac
+++ libmad-0.15.1b/configure.ac
@@ -233,6 +233,7 @@ if test -z "$FPM" && test "$GCC" = yes
 then
     case "$host" in
 	i?86-*)     FPM="INTEL"  ;;
+	x86_64*)    FPM="64BIT"  ;;
 	arm*-*)     FPM="ARM"    ;;
 	mips*-*)    FPM="MIPS"   ;;
 	sparc*-*)   FPM="SPARC"  ;;
