From: Gavin Schenk <g.schenk@eckelmann.de>
Date: Wed, 19 Sep 2012 12:21:01 +0200
Subject: [PATCH] fix unitfile for newer systemd implementation

---
 rsyslog.service.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/rsyslog.service.in b/rsyslog.service.in
index aa9e5db..9f91923 100644
--- a/rsyslog.service.in
+++ b/rsyslog.service.in
@@ -1,8 +1,8 @@
 [Unit]
 Description=System Logging Service
+Requires=syslog.socket
 
 [Service]
-ExecStartPre=/bin/systemctl stop systemd-kmsg-syslogd.service
 ExecStart=@sbindir@/rsyslogd -n -c5
 Sockets=syslog.socket
 StandardOutput=null
-- 
1.7.0.4

