From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 5 Feb 2015 17:06:35 +0100
Subject: [PATCH] fix building with newer glibc

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/intel_device.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/intel_device.c b/src/intel_device.c
index d9ff8bccfd6d..94f8cc69d80b 100644
--- a/src/intel_device.c
+++ b/src/intel_device.c
@@ -34,6 +34,8 @@
 #include <sys/ioctl.h>
 
 #include <pciaccess.h>
+
+#include <xorg-server.h>
 #include <xf86.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
