diff -urN apache_1.3.34/src/main/Makefile.tmpl apache_1.3.34-ptx/src/main/Makefile.tmpl
--- apache_1.3.34/src/main/Makefile.tmpl	2004-11-24 20:10:19.000000000 +0100
+++ apache_1.3.34-ptx/src/main/Makefile.tmpl	2006-01-03 11:25:23.000000000 +0100
@@ -32,14 +32,14 @@
 uri_delims.h: gen_uri_delims
 	./gen_uri_delims >uri_delims.h
 
-gen_uri_delims: gen_uri_delims.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o gen_uri_delims gen_uri_delims.o $(LIBS)
+gen_uri_delims: gen_uri_delims.c
+	$(CC_FOR_BUILD) -I../os/unix -I../include -o gen_uri_delims gen_uri_delims.c $(LIBS)
 
 test_char.h: gen_test_char
 	./gen_test_char >test_char.h
 
-gen_test_char: gen_test_char.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o gen_test_char gen_test_char.o $(LIBS)
+gen_test_char: gen_test_char.c
+	$(CC_FOR_BUILD)  -I../os/unix -I../include -o gen_test_char gen_test_char.c $(LIBS)
 
 # We really don't expect end users to use this rule.  It works only with
 # gcc, and rebuilds Makefile.tmpl.  You have to re-run Configure after
diff -urN apache_1.3.34/src/support/apxs.pl apache_1.3.34-ptx/src/support/apxs.pl
--- apache_1.3.34/src/support/apxs.pl	2004-11-24 20:10:19.000000000 +0100
+++ apache_1.3.34-ptx/src/support/apxs.pl	2006-01-03 11:25:23.000000000 +0100
@@ -190,13 +190,13 @@
     print STDERR "apxs:Error: $CFG_SBINDIR/$CFG_TARGET not found or not executable\n";
     exit(1);
 }
-if (not grep(/mod_so/, `$CFG_SBINDIR/$CFG_TARGET -l`)) {
-    print STDERR "apxs:Error: Sorry, no DSO support for Apache available\n";
-    print STDERR "apxs:Error: under your platform. Make sure the Apache\n";
-    print STDERR "apxs:Error: module mod_so is compiled into your server\n";
-    print STDERR "apxs:Error: binary `$CFG_SBINDIR/$CFG_TARGET'.\n";
-    exit(1);
-}
+#if (not grep(/mod_so/, `$CFG_SBINDIR/$CFG_TARGET -l`)) {
+#    print STDERR "apxs:Error: Sorry, no DSO support for Apache available\n";
+#    print STDERR "apxs:Error: under your platform. Make sure the Apache\n";
+#    print STDERR "apxs:Error: module mod_so is compiled into your server\n";
+#    print STDERR "apxs:Error: binary `$CFG_SBINDIR/$CFG_TARGET'.\n";
+#    exit(1);
+#}
 }
 
 ##
