From 6f48413916d3575a1f12bc14e0990f40606e894a Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Tue, 7 Jul 2009 18:57:16 +0200
Subject: [PATCH 5/8] [configure.in] fix time_t detection

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 source/configure.in |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/source/configure.in b/source/configure.in
index 85b0bd7..164dbcb 100644
--- a/source/configure.in
+++ b/source/configure.in
@@ -1951,10 +1951,13 @@ if test x"$samba_cv_compiler_supports_ll" = x"yes"; then
 fi
 
 
+AC_CHECK_SIZEOF(time_t)
 AC_CACHE_CHECK([for 64 bit time_t],samba_cv_SIZEOF_TIME_T,[
-AC_TRY_RUN([#include <time.h>
-main() { exit((sizeof(time_t) == 8) ? 0 : 1); }],
-samba_cv_SIZEOF_TIME_T=yes,samba_cv_SIZEOF_TIME_T=no,samba_cv_SIZEOF_TIME_T=cross)])
+if test $ac_cv_sizeof_time_t -eq 8; then
+    samba_cv_SIZEOF_TIME_T=yes
+else
+    samba_cv_SIZEOF_TIME_T=no
+fi])
 if test x"$samba_cv_SIZEOF_TIME_T" = x"yes"; then
     AC_DEFINE(SIZEOF_TIME_T,8,[The size of the 'time_t' type])
 fi
-- 
1.6.3.3

