Closes: #330783

Allow the use of $1 in scripts.


Index: logrotate-3.7.1/logrotate.c
===================================================================
--- logrotate-3.7.1.orig/logrotate.c	2006-04-08 21:47:41.000000000 +0100
+++ logrotate-3.7.1/logrotate.c	2006-04-08 22:01:23.722498089 +0100
@@ -101,7 +101,7 @@
     }
 
     if (!fork()) {
-	execl("/bin/sh", "sh", "-c", script, NULL);
+	execl("/bin/sh", "sh", "-c", script, "sh", logfn, NULL);
 	exit(1);
     }
 
