---
 Makefile.in |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

Index: b/Makefile.in
===================================================================
--- a/Makefile.in
+++ b/Makefile.in
@@ -72,7 +72,6 @@ am__libusb_la_SOURCES_DIST = usb.c usbi.
 @LINUX_API_TRUE@am__objects_1 = linux.lo
 am_libusb_la_OBJECTS = usb.lo error.lo descriptors.lo $(am__objects_1)
 libusb_la_OBJECTS = $(am_libusb_la_OBJECTS)
-libusbpp_la_LIBADD =
 am_libusbpp_la_OBJECTS = usbpp.lo
 libusbpp_la_OBJECTS = $(am_libusbpp_la_OBJECTS)
 binSCRIPT_INSTALL = $(INSTALL_SCRIPT)
@@ -295,7 +294,10 @@ libusbpp_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-release $(LT_RELEASE) \
 	-export-dynamic \
-	$(LDADDS) -lusb $(PREBIND_FLAGSPP)
+	$(LDADDS) $(PREBIND_FLAGSPP)
+
+libusbpp_la_LIBADD = \
+	$(top_builddir)/libusb.la
 
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
