diff -urN figlet221/Makefile figlet221-orig/Makefile
+++ figlet221/Makefile	2002-07-13 12:24:16.000000000 +0200
--- figlet221-orig/Makefile	2004-06-28 17:31:39.000000000 +0200
@@ -11,8 +11,8 @@
 # Don't change this even if your shell is different. The only reason
 # for changing this is if sh is not in the same place.
 SHELL = /bin/sh
-CC = gcc
-CFLAGS = -g
+CC ?= gcc
+CFLAGS ?= -g
 
 # Where the executables should be put
 DESTDIR = /usr/local/bin
