---
 gst/parse/grammar.y |   16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

Index: gstreamer-0.10.20/gst/parse/grammar.y
===================================================================
--- gstreamer-0.10.20.orig/gst/parse/grammar.y
+++ gstreamer-0.10.20/gst/parse/grammar.y
@@ -718,7 +718,9 @@ chain:   	element			      { $$ = gst_par
 						$2->front = $1;
 						$$ = $2;
 					      }
-	|	PARSE_URL chain		      { $$ = $2;
+	|	PARSE_URL chain		      {
+#ifndef GST_DISABLE_URI
+						$$ = $2;
 						if ($$->front) {
 						  GstElement *element = 
 							  gst_element_make_from_uri (GST_URI_SRC, $1, NULL);
@@ -736,9 +738,15 @@ chain:   	element			      { $$ = gst_par
 						  SET_ERROR (graph->error, GST_PARSE_ERROR_LINK, 
 							  _("no element to link URI \"%s\" to"), $1);
 						}
+#else
+						SET_ERROR (graph->error, GST_PARSE_ERROR_LINK,
+						          _("no URI support"));
+#endif
 						g_free ($1);
 					      }
-	|	link PARSE_URL		      { GstElement *element =
+	|	link PARSE_URL		      {
+#ifndef GST_DISABLE_URI
+						GstElement *element =
 							  gst_element_make_from_uri (GST_URI_SINK, $2, NULL);
 						if (!element) {
 						  SET_ERROR (graph->error, GST_PARSE_ERROR_NO_SUCH_ELEMENT, 
@@ -760,6 +768,10 @@ chain:   	element			      { $$ = gst_par
 						  $$->front->sink = element;
 						  $$->elements = g_slist_prepend (NULL, element);
 						}
+#else
+						SET_ERROR (graph->error, GST_PARSE_ERROR_LINK,
+						          _("no URI support"));
+#endif
 						g_free ($2);
 					      }
 	;
