From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Wed, 3 Aug 2011 16:09:42 +0200
Subject: [PATCH] disable designer plugins

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 qwtconfig.pri |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/qwtconfig.pri b/qwtconfig.pri
index 21cf28b..b54abeb 100644
--- a/qwtconfig.pri
+++ b/qwtconfig.pri
@@ -98,7 +98,7 @@ QWT_CONFIG     += QwtMathML
 # Otherwise you have to build it from the designer directory.
 ######################################################################
 
-QWT_CONFIG     += QwtDesigner
+# QWT_CONFIG     += QwtDesigner
 
 ######################################################################
 # If you want to auto build the examples, enable the line below
