From: Christos Zoulas <christos@zoulas.com>
Date: Fri, 21 Feb 2014 14:32:48 +0000
Subject: [PATCH] Fix memory leak (Anatol Belski)

Conflicts:
	src/softmagic.c
---
 src/softmagic.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/softmagic.c b/src/softmagic.c
index 5ed347e..7ee3f6f 100644
--- a/src/softmagic.c
+++ b/src/softmagic.c
@@ -1755,12 +1755,16 @@ mget(struct magic_set *ms, const unsigned char *s, struct magic *m,
 		ms->offset = soffset;
 		if (rv == 1) {
 			if ((ms->flags & (MAGIC_MIME|MAGIC_APPLE)) == 0 &&
-			    file_printf(ms, F(m->desc, "%u"), offset) == -1)
+			    file_printf(ms, F(m->desc, "%u"), offset) == -1) {
+				free(rbuf);
 				return -1;
-			if (file_printf(ms, "%s", rbuf) == -1)
+			}
+			if (file_printf(ms, "%s", rbuf) == -1) {
+				free(rbuf);
 				return -1;
-			free(rbuf);
+			}
 		}
+		free(rbuf);
 		return rv;
 
 	case FILE_USE:
