---
 Makefile.in |   47 +++++++++++++++++++++++++++--------------------
 1 file changed, 27 insertions(+), 20 deletions(-)

Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -5,22 +5,27 @@ INSTALL_DATA = @INSTALL_DATA@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 
-SHAREDOPT = -shared
+SHAREDOPT = -shared -fPIC,-Wl,-soname,libcgi.so.0
 LIBDIR = $(prefix)/lib
-INCDIR = $(prefix)/include
+INCDIR = $(prefix)/include/libcgi/
 MANDIR	= $(prefix)/man/man3
 SHELL = /bin/sh
 EXTRA_LIBS = 
 
 INCS =  -Isrc
-FLAGS = -Wall -fpic
+FLAGS = -Wall -D_REENTRANT
 
 OBJS = src/error.o src/cgi.o src/session.o src/base64.o src/md5.o \
 	src/string.o src/general.o src/list.o src/cookie.o
+SHOBJS=$(OBJS:.o=.sh.o)
 
-.c.o: $(CC) $(FLAGS) -c $<
+%.o: %.c
+	$(CC) $(FLAGS) -c $*.c -o $@
 
-all: $(OBJS) src/libcgi.so
+%.sh.o: %.c
+	$(CC) $(FLAGS) -fPIC -c $*.c -o $@
+
+all: src/libcgi.so src/libcgi.a
 
 	@echo ""
 	@echo ""
@@ -43,19 +48,20 @@ all: $(OBJS) src/libcgi.so
 
 
 shared: src/libcgi.so
-	cp src/libcgi.so $(LIBDIR)
+	cp src/libcgi.so $(DESTDIR)$(LIBDIR)
 
 src/libcgi.a: $(OBJS)
 	$(AR) rc src/libcgi.a $(OBJS)
 
-src/libcgi.so: src/libcgi.a
-	$(CC) $(SHAREDOPT) -o src/libcgi.so $(OBJS) $(EXTRA_LIBS)
+src/libcgi.so: $(SHOBJS)
+	$(CC) $(SHAREDOPT) -o src/libcgi.so $(SHOBJS) $(EXTRA_LIBS)
 
 install:
-	cp src/libcgi.a $(LIBDIR)
-	cp src/libcgi.so $(LIBDIR)
-	cp src/cgi.h $(INCDIR)
-	cp src/session.h $(INCDIR)
+	cp src/libcgi.a $(DESTDIR)/$(LIBDIR)
+	cp src/libcgi.so $(DESTDIR)/$(LIBDIR)
+	[ -d $(DESTDIR)/$(INCDIR) ] || mkdir $(DESTDIR)/$(INCDIR)
+	cp src/cgi.h $(DESTDIR)/$(INCDIR)
+	cp src/session.h $(DESTDIR)/$(INCDIR)
 
 
 src/error.o: src/error.c src/error.h
@@ -69,8 +75,9 @@ src/general.o: src/general.c
 src/list.o: src/list.c
 
 clean:
-	find src/ -name *.*o -exec rm -f {} \;
+	find src/ -name *.o -exec rm -f {} \;
 	find src/ -name *.a -exec rm -f {} \;
+	find src/ -name *.so -exec rm -f {} \;
 
 uninstall: clean
 	rm -f $(LIBDIR)/libcgi.*
@@ -78,11 +85,11 @@ uninstall: clean
 	rm -f $(INCDIR)/session.h
 	rm -f $(MANDIR)/libcgi*
 
-install_man:
-	cp doc/man/man3/libcgi_base64.3 $(MANDIR)
-	cp doc/man/man3/libcgi_cgi.3 $(MANDIR)
-	cp doc/man/man3/libcgi_general.3 $(MANDIR)
-	cp doc/man/man3/libcgi_string.3 $(MANDIR)
-	cp doc/man/man3/libcgi_session.3 $(MANDIR)
-	cp doc/man/man3/libcgi_cookie.3 $(MANDIR)
+#install_man:
+#	cp doc/man/man3/libcgi_base64.3 $(MANDIR)
+#	cp doc/man/man3/libcgi_cgi.3 $(MANDIR)
+#	cp doc/man/man3/libcgi_general.3 $(MANDIR)
+#	cp doc/man/man3/libcgi_string.3 $(MANDIR)
+#	cp doc/man/man3/libcgi_session.3 $(MANDIR)
+#	cp doc/man/man3/libcgi_cookie.3 $(MANDIR)
 
