---
 Makefile.in |   11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -48,7 +48,7 @@ all: src/libcgi.so src/libcgi.a
 
 
 shared: src/libcgi.so
-	cp src/libcgi.so $(DESTDIR)$(LIBDIR)
+	$(INSTALL) -m 0644 src/libcgi.so $(DESTDIR)$(LIBDIR)
 
 src/libcgi.a: $(OBJS)
 	$(AR) rc src/libcgi.a $(OBJS)
@@ -57,11 +57,10 @@ src/libcgi.so: $(SHOBJS)
 	$(CC) $(SHAREDOPT) -o src/libcgi.so $(SHOBJS) $(EXTRA_LIBS)
 
 install:
-	cp src/libcgi.a $(DESTDIR)/$(LIBDIR)
-	cp src/libcgi.so $(DESTDIR)/$(LIBDIR)
-	[ -d $(DESTDIR)/$(INCDIR) ] || mkdir $(DESTDIR)/$(INCDIR)
-	cp src/cgi.h $(DESTDIR)/$(INCDIR)
-	cp src/session.h $(DESTDIR)/$(INCDIR)
+	$(INSTALL) -m 0644 src/libcgi.a $(DESTDIR)/$(LIBDIR)/libcgi.a
+	$(INSTALL) -m 0644 src/libcgi.so $(DESTDIR)/$(LIBDIR)/libcgi.so
+	$(INSTALL) -m 0644 src/cgi.h $(DESTDIR)/$(INCDIR)/cgi.h
+	$(INSTALL) -m 0644 src/session.h $(DESTDIR)/$(INCDIR)/session.h
 
 
 src/error.o: src/error.c src/error.h
