---
 tools/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: tools/Makefile.am
===================================================================
--- tools/Makefile.am.orig
+++ tools/Makefile.am
@@ -1,5 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = getiversion getkversion locktest rpcdebug rpcgen nlmtest
+SUBDIRS = getiversion locktest rpcdebug rpcgen nlmtest
 
 MAINTAINERCLEANFILES = Makefile.in
