From 5b997526032903b38105db27550ef22a05ee1f5f Mon Sep 17 00:00:00 2001
From: H. Peter Anvin <hpa@zytor.com>
Date: Sun, 4 Jan 2009 19:29:33 +0000
Subject: [PATCH 10/21] [klibc] 1.5.15 released, next version is 1.5.16

---
 usr/klibc/version |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/usr/klibc/version b/usr/klibc/version
index f6ed435..0ed0610 100644
--- a/usr/klibc/version
+++ b/usr/klibc/version
@@ -1 +1 @@
-1.5.15
+1.5.16
-- 
1.6.5

