---
 qwtconfig.pri |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/qwtconfig.pri
===================================================================
--- a/qwtconfig.pri
+++ b/qwtconfig.pri
@@ -98,7 +98,7 @@
 # Otherwise you have to build it from the designer directory.
 ######################################################################
 
-QWT_CONFIG     += QwtDesigner
+# QWT_CONFIG     += QwtDesigner
 
 ######################################################################
 # If you want to auto build the examples, enable the line below
