---
 scripts/libpng-config.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: scripts/libpng-config.in
===================================================================
--- scripts/libpng-config.in.orig
+++ scripts/libpng-config.in
@@ -9,7 +9,7 @@
 # Modeled after libxml-config.
 
 version="@PNGLIB_VERSION@"
-prefix="@prefix@"
+prefix="$SYSROOT@prefix@"
 exec_prefix="@exec_prefix@"
 libdir="@libdir@"
 includedir="@includedir@/libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@"
