diff -x .deps -urN chrony-1.20/wrap_adjtimex.c chrony-1.20-right/wrap_adjtimex.c
--- chrony-1.20/wrap_adjtimex.c	2005-07-14 10:37:04.000000000 +0200
+++ chrony-1.20-right/wrap_adjtimex.c	2005-07-14 10:04:44.000000000 +0200
@@ -38,16 +38,12 @@
 
 #define _LOOSE_KERNEL_NAMES
 
-/* Use sys/time.h directly to remove a warning */
-#include <sys/time.h>
-#include <linux/timex.h>
-#include <asm/param.h>
+#include <sys/timex.h>
 
-#include "wrap_adjtimex.h"
-
-/* This doesn't seem to be in any include files !! */
+/* FIXME: this was formerly taken from linux/timex.h */
+#define SHIFT_USEC 16
 
-extern int adjtimex(struct timex *);
+#include "wrap_adjtimex.h"
 
 int
 TMX_SetTick(long tick)
