diff -ruN gdb-5.3-orig/readline/Makefile.in gdb-5.3/readline/Makefile.in
--- gdb-5.3-orig/readline/Makefile.in	Sun Jul  9 19:19:55 2000
+++ gdb-5.3/readline/Makefile.in	Fri Sep 26 17:30:45 2003
@@ -63,7 +63,7 @@
 TERMCAP_LIB = @TERMCAP_LIB@
 
 # For libraries which include headers from other libraries.
-INCLUDES = -I. -I$(srcdir) -I$(includedir)
+INCLUDES = -I. -I$(srcdir)
 
 CCFLAGS = $(DEFS) $(LOCAL_DEFS) $(CPPFLAGS) $(INCLUDES) $(LOCAL_CFLAGS) $(CFLAGS)
 
