From: Juergen Beisert <jbe@pengutronix.de>
Date: Tue, 12 Jun 2012 16:14:22 +0200
Subject: [PATCH] Check for LIBUDEV only if udev support is enabled

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>
---
 configure.ac |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index d7c59ed..92fdb70 100644
--- a/configure.ac
+++ b/configure.ac
@@ -818,8 +818,8 @@ if test "x$CONFIG_UDEV" = xyes &&
 	AC_MSG_ERROR([Hotplugging through both libudev and dbus/hal not allowed])
 fi
 
-PKG_CHECK_MODULES(UDEV, $LIBUDEV, [HAVE_LIBUDEV=yes], [HAVE_LIBUDEV=no])
-if test "x$CONFIG_UDEV" = xauto; then
+if test "x$CONFIG_UDEV" = xauto -o "x$CONFIG_UDEV" = xyes; then
+	PKG_CHECK_MODULES(UDEV, $LIBUDEV, [HAVE_LIBUDEV=yes], [HAVE_LIBUDEV=no])
 	CONFIG_UDEV="$HAVE_LIBUDEV"
 fi
 AM_CONDITIONAL(CONFIG_UDEV, [test "x$CONFIG_UDEV" = xyes])
