Only in util-linux-ng-2.14.1-rc2: autom4te.cache
diff -ur util-linux-ng-2.14.1-rc2-orig/configure util-linux-ng-2.14.1-rc2/configure
--- util-linux-ng-2.14.1-rc2-orig/configure	2008-08-22 13:08:17.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/configure	2008-09-01 21:00:15.000000000 +0200
@@ -724,6 +724,7 @@
 HAVE_STATIC_SFDISK_FALSE
 HAVE_UUID_TRUE
 HAVE_UUID_FALSE
+UUID_LIBS
 HAVE_UTIL_TRUE
 HAVE_UTIL_FALSE
 HAVE_TERMCAP_TRUE
@@ -1434,6 +1435,7 @@
   --enable-static-programs=LIST
                           link static the programs in LIST (comma-separated,
                           supported for losetup, mount, umount, fdisk, sfdisk)
+  --enable-uuid           enable uuid support for mkswap(8) [default=auto]
   --disable-nls           do not use Native Language Support
   --disable-rpath         do not hardcode runtime library paths
   --enable-arch           do build arch
@@ -5608,11 +5610,13 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <stdio.h>
+#include <sys/types.h> /* for off_t */
+     #include <stdio.h>
 int
 main ()
 {
-return fseeko (stdin, 0, 0) && (fseeko) (stdin, 0, 0);
+int (*fp) (FILE *, off_t, int) = fseeko;
+     return fseeko (stdin, 0, 0) && fp (stdin, 0, 0);
   ;
   return 0;
 }
@@ -5652,11 +5656,13 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #define _LARGEFILE_SOURCE 1
-#include <stdio.h>
+#include <sys/types.h> /* for off_t */
+     #include <stdio.h>
 int
 main ()
 {
-return fseeko (stdin, 0, 0) && (fseeko) (stdin, 0, 0);
+int (*fp) (FILE *, off_t, int) = fseeko;
+     return fseeko (stdin, 0, 0) && fp (stdin, 0, 0);
   ;
   return 0;
 }
@@ -5961,6 +5967,33 @@
 
 
 
+{ echo "$as_me:$LINENO: checking whether to enable uuid support for mkswap(8)" >&5
+echo $ECHO_N "checking whether to enable uuid support for mkswap(8)... $ECHO_C" >&6; }
+# Check whether --enable-uuid was given.
+if test "${enable_uuid+set}" = set; then
+  enableval=$enable_uuid; case "$enableval" in
+	y | yes) CONFIG_UUID=yes;;
+	n | no)  CONFIG_UUID=no;;
+        auto)    CONFIG_UUID=auto;;
+	*)
+		{ { echo "$as_me:$LINENO: error: --enable-uuid=\"$enableval\" is not valid" >&5
+echo "$as_me: error: --enable-uuid=\"$enableval\" is not valid" >&2;}
+   { (exit 1); exit 1; }; }
+		;;
+        esac
+else
+  CONFIG_UUID=auto
+
+fi
+
+{ echo "$as_me:$LINENO: result: $CONFIG_UUID" >&5
+echo "${ECHO_T}$CONFIG_UUID" >&6; }
+
+if test $CONFIG_UUID = no; then
+	HAVE_UUID=false
+	UUID_LIBS=
+else
+
 
   have_uuid=yes
 
@@ -6045,13 +6078,35 @@
 fi
 
 
-if test "x$have_uuid" = xno; then
-  { echo "$as_me:$LINENO: WARNING: uuid library is not found; mkswap(8) will not generate UUIDs" >&5
+	if test $CONFIG_UUID = yes && test x$have_uuid = xno; then
+		{ { echo "$as_me:$LINENO: error: uuid library is not found but --enable-uuid given; please install libuuid" >&5
+echo "$as_me: error: uuid library is not found but --enable-uuid given; please install libuuid" >&2;}
+   { (exit 1); exit 1; }; }
+	fi
+	if test $CONFIG_UUID = auto && test x$have_uuid = xno; then
+  		{ echo "$as_me:$LINENO: WARNING: uuid library is not found; mkswap(8) will not generate UUIDs" >&5
 echo "$as_me: WARNING: uuid library is not found; mkswap(8) will not generate UUIDs" >&2;}
+		HAVE_UUID=no
+		UUID_LIBS=
+	fi
+	if test x$have_uuid = xyes; then
+		HAVE_UUID=true
+		UUID_LIBS=-luuid
+	fi
+fi
+ if $HAVE_UUID; then
+  HAVE_UUID_TRUE=
+  HAVE_UUID_FALSE='#'
+else
+  HAVE_UUID_TRUE='#'
+  HAVE_UUID_FALSE=
 fi
 
 
 
+
+
+
   have_util=yes
 
 { echo "$as_me:$LINENO: checking for openpty in -lutil" >&5
@@ -11681,6 +11736,13 @@
 Usually this means the macro was only invoked conditionally." >&2;}
    { (exit 1); exit 1; }; }
 fi
+if test -z "${HAVE_UUID_TRUE}" && test -z "${HAVE_UUID_FALSE}"; then
+  { { echo "$as_me:$LINENO: error: conditional \"HAVE_UUID\" was never defined.
+Usually this means the macro was only invoked conditionally." >&5
+echo "$as_me: error: conditional \"HAVE_UUID\" was never defined.
+Usually this means the macro was only invoked conditionally." >&2;}
+   { (exit 1); exit 1; }; }
+fi
 if test -z "${HAVE_UTIL_TRUE}" && test -z "${HAVE_UTIL_FALSE}"; then
   { { echo "$as_me:$LINENO: error: conditional \"HAVE_UTIL\" was never defined.
 Usually this means the macro was only invoked conditionally." >&5
@@ -12738,6 +12800,7 @@
 HAVE_STATIC_SFDISK_FALSE!$HAVE_STATIC_SFDISK_FALSE$ac_delim
 HAVE_UUID_TRUE!$HAVE_UUID_TRUE$ac_delim
 HAVE_UUID_FALSE!$HAVE_UUID_FALSE$ac_delim
+UUID_LIBS!$UUID_LIBS$ac_delim
 HAVE_UTIL_TRUE!$HAVE_UTIL_TRUE$ac_delim
 HAVE_UTIL_FALSE!$HAVE_UTIL_FALSE$ac_delim
 HAVE_TERMCAP_TRUE!$HAVE_TERMCAP_TRUE$ac_delim
@@ -12823,7 +12886,6 @@
 BUILD_RENAME_FALSE!$BUILD_RENAME_FALSE$ac_delim
 BUILD_RESET_TRUE!$BUILD_RESET_TRUE$ac_delim
 BUILD_RESET_FALSE!$BUILD_RESET_FALSE$ac_delim
-BUILD_LOGIN_UTILS_TRUE!$BUILD_LOGIN_UTILS_TRUE$ac_delim
 _ACEOF
 
   if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 97; then
@@ -12865,6 +12927,7 @@
 ac_delim='%!_!# '
 for ac_last_try in false false false false false :; do
   cat >conf$$subs.sed <<_ACEOF
+BUILD_LOGIN_UTILS_TRUE!$BUILD_LOGIN_UTILS_TRUE$ac_delim
 BUILD_LOGIN_UTILS_FALSE!$BUILD_LOGIN_UTILS_FALSE$ac_delim
 HAVE_PAM_TRUE!$HAVE_PAM_TRUE$ac_delim
 HAVE_PAM_FALSE!$HAVE_PAM_FALSE$ac_delim
@@ -12890,7 +12953,7 @@
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 23; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 24; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
diff -ur util-linux-ng-2.14.1-rc2-orig/disk-utils/Makefile.in util-linux-ng-2.14.1-rc2/disk-utils/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/disk-utils/Makefile.in	2008-08-22 13:08:18.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/disk-utils/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -206,6 +206,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/fdisk/Makefile.in util-linux-ng-2.14.1-rc2/fdisk/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/fdisk/Makefile.in	2008-08-22 13:08:18.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/fdisk/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -211,6 +211,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/getopt/Makefile.in util-linux-ng-2.14.1-rc2/getopt/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/getopt/Makefile.in	2008-08-22 13:08:18.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/getopt/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -139,6 +139,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/hwclock/Makefile.in util-linux-ng-2.14.1-rc2/hwclock/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/hwclock/Makefile.in	2008-08-22 13:08:18.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/hwclock/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -137,6 +137,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/include/Makefile.in util-linux-ng-2.14.1-rc2/include/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/include/Makefile.in	2008-08-22 13:08:18.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/include/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -118,6 +118,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/lib/Makefile.in util-linux-ng-2.14.1-rc2/lib/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/lib/Makefile.in	2008-08-22 13:08:18.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/lib/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -131,6 +131,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/login-utils/Makefile.in util-linux-ng-2.14.1-rc2/login-utils/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/login-utils/Makefile.in	2008-08-22 13:08:19.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/login-utils/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -276,6 +276,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/Makefile.in util-linux-ng-2.14.1-rc2/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/Makefile.in	2008-08-22 13:08:20.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -149,6 +149,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/misc-utils/Makefile.in util-linux-ng-2.14.1-rc2/misc-utils/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/misc-utils/Makefile.in	2008-08-22 13:08:19.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/misc-utils/Makefile.in	2008-09-01 21:00:17.000000000 +0200
@@ -207,6 +207,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/mount/Makefile.in util-linux-ng-2.14.1-rc2/mount/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/mount/Makefile.in	2008-08-22 13:08:19.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/mount/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -125,7 +125,8 @@
 @HAVE_VOLUME_ID_TRUE@am__DEPENDENCIES_3 = $(am__DEPENDENCIES_1)
 am__DEPENDENCIES_4 = $(am__DEPENDENCIES_2) $(am__DEPENDENCIES_3)
 @HAVE_SELINUX_TRUE@am__DEPENDENCIES_5 = $(am__DEPENDENCIES_1)
-mount_DEPENDENCIES = $(am__DEPENDENCIES_4) $(am__DEPENDENCIES_5)
+mount_DEPENDENCIES = $(am__DEPENDENCIES_4) $(am__DEPENDENCIES_1) \
+	$(am__DEPENDENCIES_5)
 mount_LINK = $(CCLD) $(mount_CFLAGS) $(CFLAGS) $(mount_LDFLAGS) \
 	$(LDFLAGS) -o $@
 am__mount_static_SOURCES_DIST = mount.c fstab.c mount_mntent.c \
@@ -169,7 +170,7 @@
 am_swapon_OBJECTS = swapon.$(OBJEXT) $(am__objects_12) \
 	$(am__objects_16)
 swapon_OBJECTS = $(am_swapon_OBJECTS)
-swapon_DEPENDENCIES = $(am__DEPENDENCIES_4)
+swapon_DEPENDENCIES = $(am__DEPENDENCIES_4) $(am__DEPENDENCIES_1)
 am__umount_SOURCES_DIST = umount.c fstab.c mount_mntent.c \
 	getusername.c lomount.c sundries.c xmalloc.c realpath.c \
 	fsprobe.c fsprobe_blkid.c fsprobe_volumeid.c fstab.h \
@@ -191,7 +192,7 @@
 	$(am__objects_20)
 am_umount_OBJECTS = umount-umount.$(OBJEXT) $(am__objects_21)
 umount_OBJECTS = $(am_umount_OBJECTS)
-umount_DEPENDENCIES = $(am__DEPENDENCIES_4)
+umount_DEPENDENCIES = $(am__DEPENDENCIES_4) $(am__DEPENDENCIES_1)
 umount_LINK = $(CCLD) $(umount_CFLAGS) $(CFLAGS) $(umount_LDFLAGS) \
 	$(LDFLAGS) -o $@
 am__umount_static_SOURCES_DIST = umount.c fstab.c mount_mntent.c \
@@ -297,6 +298,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
@@ -391,9 +393,9 @@
 	loop.h lomount.h xmalloc.h sundries.h realpath.h $(fallback)
 
 losetup_CPPFLAGS = -DMAIN $(AM_CPPFLAGS)
-mount_LDADD = $(LDADD_common) $(am__append_7)
-umount_LDADD = $(LDADD_common)
-swapon_LDADD = $(LDADD_common)
+mount_LDADD = $(LDADD_common) $(UUID_LIBS) $(am__append_7)
+umount_LDADD = $(LDADD_common) $(UUID_LIBS)
+swapon_LDADD = $(LDADD_common) $(UUID_LIBS)
 LDADD_common = $(am__append_5) $(am__append_10)
 LDADD_common_static = $(am__append_6) $(am__append_11)
 @HAVE_STATIC_MOUNT_TRUE@mount_static_SOURCES = $(mount_SOURCES)
diff -ur util-linux-ng-2.14.1-rc2-orig/partx/Makefile.in util-linux-ng-2.14.1-rc2/partx/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/partx/Makefile.in	2008-08-22 13:08:19.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/partx/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -147,6 +147,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/schedutils/Makefile.in util-linux-ng-2.14.1-rc2/schedutils/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/schedutils/Makefile.in	2008-08-22 13:08:19.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/schedutils/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -148,6 +148,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/sys-utils/Makefile.in util-linux-ng-2.14.1-rc2/sys-utils/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/sys-utils/Makefile.in	2008-08-22 13:08:19.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/sys-utils/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -241,6 +241,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/tests/helpers/Makefile.in util-linux-ng-2.14.1-rc2/tests/helpers/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/tests/helpers/Makefile.in	2008-08-22 13:08:20.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/tests/helpers/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -131,6 +131,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/tests/Makefile.in util-linux-ng-2.14.1-rc2/tests/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/tests/Makefile.in	2008-08-22 13:08:20.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/tests/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -126,6 +126,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
diff -ur util-linux-ng-2.14.1-rc2-orig/text-utils/Makefile.in util-linux-ng-2.14.1-rc2/text-utils/Makefile.in
--- util-linux-ng-2.14.1-rc2-orig/text-utils/Makefile.in	2008-08-22 13:08:20.000000000 +0200
+++ util-linux-ng-2.14.1-rc2/text-utils/Makefile.in	2008-09-01 21:00:18.000000000 +0200
@@ -180,6 +180,7 @@
 SUID_CFLAGS = @SUID_CFLAGS@
 SUID_LDFLAGS = @SUID_LDFLAGS@
 USE_NLS = @USE_NLS@
+UUID_LIBS = @UUID_LIBS@
 VERSION = @VERSION@
 VOLID = @VOLID@
 VOLUMEID_LIBS = @VOLUMEID_LIBS@
