Subject: [patch] remove hand-made -m486 option
From: Juergen Beisert <j.beisert@pengutronix.de>

Don't use hand made -m<cpu> options.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---

diff -urN setmixer-27DEC94ds1.orig/Makefile setmixer-27DEC94ds1.orig-ptx/Makefile
--- setmixer-27DEC94ds1.orig/Makefile	1999-06-27 10:49:28.000000000 +0200
+++ setmixer-27DEC94ds1.orig-ptx/Makefile	2006-01-02 16:27:16.000000000 +0100
@@ -1,8 +1,8 @@
 DESTDIR =
 INSTALL = install
 CC=gcc
-CFLAGS= -O6 -fomit-frame-pointer -Wall -pipe -m486
-LDFLAGS= -Xlinker -N -s
+CFLAGS= -O6 -fomit-frame-pointer -Wall -pipe 
+LDFLAGS= -lgcc_s
 PROG=setmixer
 
 $(PROG): setmixer.o
