From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 7 Jul 2015 09:58:21 +0200
Subject: [PATCH] HACK: configure.ac: disable NLS to avoid unnecessary
 dependencies during autogen.sh

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure.ac | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/configure.ac b/configure.ac
index c7537aafc0c9..c8fa808bc4d3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -60,6 +60,7 @@ SET_ARCH(MIPS, mips*)
 SET_ARCH(ARM, arm*)
 SET_ARCH(AARCH64, aarch64*)
 
+m4_ifdef([AM_NLS__DISABLED], [
 # i18n stuff for the PolicyKit policy files, heck whether intltool can be found, disable NLS otherwise
 AC_CHECK_PROG(intltool_found, [intltool-merge], [yes], [no])
 AS_IF([test x"$intltool_found" != xyes],
@@ -82,6 +83,12 @@ AS_IF([test -z "$INTLTOOL_POLICY_RULE"], [
     INTLTOOL_POLICY_RULE='%.policy: %.policy.in ; @echo "  ITMRG   " $@ && echo "*** intltool support required to build target $@" && false'
     AC_SUBST(INTLTOOL_POLICY_RULE)
 ])
+],[
+touch `dirname $0`/po/Makefile.in.in
+echo -e 'all:\ninstall:\n' > `dirname $0`/po/Makefile
+INTLTOOL_POLICY_RULE=
+AC_SUBST(INTLTOOL_POLICY_RULE)
+])
 
 GETTEXT_PACKAGE=systemd
 AC_SUBST(GETTEXT_PACKAGE)
