From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 19 Nov 2015 16:52:47 +0100
Subject: [PATCH] missing: add more btrfs defines

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/basic/missing.h | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/src/basic/missing.h b/src/basic/missing.h
index fae5329663b2..975d6054c5ce 100644
--- a/src/basic/missing.h
+++ b/src/basic/missing.h
@@ -331,6 +331,23 @@ struct btrfs_ioctl_quota_ctl_args {
         __u64 cmd;
         __u64 status;
 };
+
+struct btrfs_ioctl_quota_rescan_args {
+        __u64 flags;
+        __u64 progress;
+        __u64 reserved[6];
+};
+
+struct btrfs_ioctl_qgroup_assign_args {
+        __u64 assign;
+        __u64 src;
+        __u64 dst;
+};
+
+struct btrfs_ioctl_qgroup_create_args {
+        __u64 create;
+        __u64 qgroupid;
+};
 #endif
 
 #ifndef BTRFS_IOC_DEFRAG
@@ -405,11 +422,31 @@ struct btrfs_ioctl_quota_ctl_args {
                                struct btrfs_ioctl_quota_ctl_args)
 #endif
 
+#ifndef BTRFS_IOC_QGROUP_ASSIGN
+#define BTRFS_IOC_QGROUP_ASSIGN _IOW(BTRFS_IOCTL_MAGIC, 41, \
+                               struct btrfs_ioctl_qgroup_assign_args)
+#endif
+
+#ifndef BTRFS_IOC_QGROUP_CREATE
+#define BTRFS_IOC_QGROUP_CREATE _IOW(BTRFS_IOCTL_MAGIC, 42, \
+                               struct btrfs_ioctl_qgroup_create_args)
+#endif
+
 #ifndef BTRFS_IOC_QGROUP_LIMIT
 #define BTRFS_IOC_QGROUP_LIMIT _IOR(BTRFS_IOCTL_MAGIC, 43, \
                                struct btrfs_ioctl_qgroup_limit_args)
 #endif
 
+#ifndef BTRFS_IOC_QUOTA_RESCAN
+#define BTRFS_IOC_QUOTA_RESCAN _IOW(BTRFS_IOCTL_MAGIC, 44, \
+                               struct btrfs_ioctl_quota_rescan_args)
+#endif
+
+#ifndef BTRFS_IOC_QUOTA_RESCAN_STATUS
+#define BTRFS_IOC_QUOTA_RESCAN_STATUS _IOR(BTRFS_IOCTL_MAGIC, 45, \
+                               struct btrfs_ioctl_quota_rescan_args)
+#endif
+
 #ifndef BTRFS_IOC_QUOTA_RESCAN_WAIT
 #define BTRFS_IOC_QUOTA_RESCAN_WAIT _IO(BTRFS_IOCTL_MAGIC, 46)
 #endif
