Subject: busybox 1.9.0 fixes
From: http://busybox.net/downloads/fixes-1.9.0

---
 Makefile.help           |    1 -
 applets/applet_tables.c |    2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

Index: busybox-1.9.0/Makefile.help
===================================================================
--- busybox-1.9.0.orig/Makefile.help
+++ busybox-1.9.0/Makefile.help
@@ -16,7 +16,6 @@ help:
 	@echo 'Configuration:'
 	@echo '  allnoconfig		- disable all symbols in .config'
 	@echo '  allyesconfig		- enable all symbols in .config (see defconfig)'
-	@echo '  allbareconfig		- enable all applets without any sub-features'
 	@echo '  config		- text based configurator (of last resort)'
 	@echo '  defconfig		- set .config to largest generic configuration'
 	@echo '  menuconfig		- interactive curses-based configurator'
Index: busybox-1.9.0/applets/applet_tables.c
===================================================================
--- busybox-1.9.0.orig/applets/applet_tables.c
+++ busybox-1.9.0/applets/applet_tables.c
@@ -71,7 +71,7 @@ int main(int argc, char **argv)
 
 	puts("/* This is a generated file, don't edit */");
 
-	puts("const char applet_names[] ALIGN1 =");
+	puts("const char applet_names[] ALIGN1 = \"\" \n");
 	for (i = 0; i < NUM_APPLETS; i++) {
 		printf("\"%s\" \"\\0\"\n", applets[i].name);
 	}
