Subject: busybox 1.9.0 fixes
From: http://busybox.net/downloads/fixes-1.9.0

---
 e2fsprogs/fsck.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: busybox-1.9.0/e2fsprogs/fsck.c
===================================================================
--- busybox-1.9.0.orig/e2fsprogs/fsck.c
+++ busybox-1.9.0/e2fsprogs/fsck.c
@@ -665,6 +665,12 @@ static void execute(const char *type, co
 	for (i = num_args+1; i < argc; i++)
 		free(argv[i]);
 
+	/* No pid, so don't record an instance */
+	if (pid < 0) {
+		free(inst);
+		return;
+	}
+
 	inst->pid = pid;
 	inst->prog = argv[0];
 	inst->type = xstrdup(type);
