From 0390735daaefeda85dd1f0bcfdcb6dbcd1910aec Mon Sep 17 00:00:00 2001
From: the debian project <info@debian.org>
Date: Thu, 11 Mar 2010 01:11:11 +0100
Subject: [PATCH 1/6] 01_add_toplevel_makefile.diff

---
 Makefile |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)
 create mode 100644 Makefile

diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..2a93406
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,7 @@
+SUBDIRS = elfls  elftoc rebind  sstrip
+
+all:
+	set -e; for d in $(SUBDIRS); do cd $$d ; CFLAGS="$(CFLAGS)" CC=$(CC) $(MAKE) ; cd - ; done
+
+clean:
+	set -e; for d in $(SUBDIRS); do cd $$d ; $(MAKE) clean; cd - ; done
-- 
1.7.0

