diff -x CVS -urN FixesExt-20041103-1/autogen-rs.sh FixesExt-ptx/autogen-rs.sh
--- FixesExt-20041103-1/autogen-rs.sh	1970-01-01 01:00:00.000000000 +0100
+++ FixesExt-ptx/autogen-rs.sh	2004-11-03 07:40:50.000000000 +0100
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+aclocal-1.7 -I /usr/share/aclocal/
+#autoheader
+autoconf
+automake-1.7 --add-missing --copy
+
diff -x CVS -urN FixesExt-20041103-1/configure.ac FixesExt-ptx/configure.ac
--- FixesExt-20041103-1/configure.ac	2004-11-03 09:20:50.000000000 +0100
+++ FixesExt-ptx/configure.ac	2004-11-03 08:11:10.000000000 +0100
@@ -25,7 +25,7 @@
 
 AC_PREREQ([2.57])
 AC_INIT([fixesext], [3.0.0], [keithp@keithp.com], fixesext)
-AM_INIT_AUTOMAKE([dist-bzip2])
+AM_INIT_AUTOMAKE(FixesExt, 20041103-1, [dist-bzip2])
 AM_MAINTAINER_MODE
 
 PKG_CHECK_MODULES(XEXTENSIONS, xextensions)
diff -x CVS -urN FixesExt-20041103-1/Makefile.am FixesExt-ptx/Makefile.am
--- FixesExt-20041103-1/Makefile.am	2004-11-03 09:20:50.000000000 +0100
+++ FixesExt-ptx/Makefile.am	2004-11-03 08:10:54.000000000 +0100
@@ -33,5 +33,5 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = fixesext.pc
 
-EXTRA_DIST = autogen.sh fixesext.pc.in $(fixesextdoc_DATA)
+EXTRA_DIST = autogen.sh autogen-rs.sh fixesext.pc.in $(fixesextdoc_DATA)
 
