Fix the taboo list to avoid .dpkg-old, etc.



Index: logrotate-3.7.1/config.c
===================================================================
--- logrotate-3.7.1.orig/config.c	2006-04-08 20:46:40.147476352 +0100
+++ logrotate-3.7.1/config.c	2006-04-08 20:48:26.323326011 +0100
@@ -30,6 +30,8 @@
 #endif
 
 static char * defTabooExts[] = { ".rpmsave", ".rpmorig", "~", ",v",
+				 ".disabled", ".dpkg-old", ".dpkg-dist",
+				 ".dpkg-new",
 				 ".rpmnew", ".swp" };
 static int defTabooCount = sizeof(defTabooExts) / sizeof(char *);
 
