---
 logrotate.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: logrotate-3.7.1/logrotate.c
===================================================================
--- logrotate-3.7.1.orig/logrotate.c
+++ logrotate-3.7.1/logrotate.c
@@ -209,7 +209,7 @@ static int compressLogFile(char * name, 
 static int mailLog(char * logFile, char * mailCommand, char * uncompressCommand, 
 		   char * address, char * subject) {
     int mailInput;
-    pid_t mailChild, uncompressChild;
+    pid_t mailChild, uncompressChild=-1;
     int mailStatus, uncompressStatus;
     int uncompressPipe[2];
     char * mailArgv[] = { mailCommand, "-s", subject, address, NULL };
