---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: elektra-0.7.0rc5/configure.ac
===================================================================
--- elektra-0.7.0rc5.orig/configure.ac
+++ elektra-0.7.0rc5/configure.ac
@@ -287,7 +287,7 @@ AC_PROG_LIBTOOL
 # xsltproc
 AC_PATH_PROG(xsltproc, xsltproc, "no")
 # Check if have style-sheet too
-if test "x$default_docbook" != "xno"; then
+if test "x$default_docbook" != "xno" && test "x$cross_compiling" = "xno"; then
 	AC_CHECK_FILE([$default_docbook/manpages/docbook.xsl],
 		      [dbroot=$default_docbook],
 		      [dbroot="no"])
