diff -urN mii-diag-2.09.orig/mii-diag.c mii-diag-2.09.orig-ptx/mii-diag.c
--- mii-diag-2.09.orig/mii-diag.c	2006-01-02 13:51:33.000000000 +0100
+++ mii-diag-2.09.orig-ptx/mii-diag.c	2006-01-02 14:45:34.000000000 +0100
@@ -170,6 +170,8 @@
 static int parse_advertise(const char *capabilities);
 static void monitor_status(long ioaddr, int phy_id);
 
+int  show_mii_details(long ioaddr, int phy_id) __attribute__((weak));
+void monitor_mii(long ioaddr, int phy_id) __attribute__((weak));
 
 int
 main(int argc, char **argv)
@@ -632,9 +634,6 @@
 
 /* These functions are not used if we are linked with libmii. */
 
-int  show_mii_details(long ioaddr, int phy_id) __attribute__((weak));
-void monitor_mii(long ioaddr, int phy_id) __attribute__((weak));
-
 int  show_mii_details(long ioaddr, int phy_id)
 {
 	int mii_reg, mii_val;
