Index: libXt-1.0.4/util/Makefile.in
===================================================================
--- libXt-1.0.4.orig/util/Makefile.in	2006-11-08 15:16:24.000000000 +0100
+++ libXt-1.0.4/util/Makefile.in	2007-10-17 17:29:45.000000000 +0200
@@ -96,11 +96,13 @@
 AUTOHEADER = @AUTOHEADER@
 AUTOMAKE = @AUTOMAKE@
 AWK = @AWK@
-CC = @CC@
+CC = $(CC_FOR_BUILD)
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+
+# evil hack
+CFLAGS = $(CFLAGS_FOR_BUILD)
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = $(CPPFLAGS_FOR_BUILD)
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
@@ -127,10 +129,10 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = $(LDFLAGS_FOR_BUILD)
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
-LIBTOOL = @LIBTOOL@
+LIBTOOL = @LIBTOOL@ --tag=CC
 LIB_MAN_DIR = @LIB_MAN_DIR@
 LIB_MAN_SUFFIX = @LIB_MAN_SUFFIX@
 LN_S = @LN_S@
@@ -214,7 +216,9 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 EXTRA_DIST = Shell.ht StrDefs.ct StrDefs.ht string.list $(appman_PRE)
-makestrs_CFLAGS = $(XT_CFLAGS)
+makestrs_CFLAGS = $(CFLAGS_FOR_BUILD) $(XT_CFLAGS)
+makestrs_CPPLAGS = $(CPPFLAGS_FOR_BUILD)
+makestrs_LDFLAGS = $(LDFLAGS_FOR_BUILD)
 makestrs_SOURCES = makestrs.c
 
 # Man page
