From: unknown author <unknown.author@example.com>
Date: Wed, 3 Aug 2011 17:25:10 +0200
Subject: [PATCH] 0018-dont-add-rpaths.diff

---
 setup.py |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 2156621..036063c 100644
--- a/setup.py
+++ b/setup.py
@@ -1051,7 +1051,6 @@ class PyBuildExt(build_ext):
                                   include_dirs=["Modules/_sqlite",
                                                 sqlite_incdir],
                                   library_dirs=sqlite_libdir,
-                                  runtime_library_dirs=sqlite_libdir,
                                   extra_link_args=sqlite_extra_link_args,
                                   libraries=["sqlite3",]))
         else:
@@ -1113,7 +1112,6 @@ class PyBuildExt(build_ext):
                         print("building dbm using bdb")
                         dbmext = Extension('_dbm', ['_dbmmodule.c'],
                                            library_dirs=dblib_dir,
-                                           runtime_library_dirs=dblib_dir,
                                            include_dirs=db_incs,
                                            define_macros=[
                                                ('HAVE_BERKDB_H', None),
