From: Markus Pargmann <mpa@pengutronix.de>
Date: Thu, 13 Nov 2014 15:34:56 +0100
Subject: [PATCH] configure: Add missing conditionals

Signed-off-by: Markus Pargmann <mpa@pengutronix.de>
---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index 85f0454987f4..5acbcd193e61 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1622,6 +1622,9 @@ fi
 
 AM_CONDITIONAL(ENABLE_MAN, test x$enable_man != xno)
 
+AM_CONDITIONAL(HAVE_INTROSPECTION, test x$have_introspection == xyes)
+AM_CONDITIONAL(HAVE_GTK_DOC, test x$have_gtk_doc == xyes)
+
 
 ##################################################
 # Output commands
