From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 10 May 2016 11:45:47 +0200
Subject: [PATCH] fix parallel building

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index fb31707c979b..c2873aa7e167 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -116,7 +116,7 @@ ELFOBJS = @PROGRAM_ELFOBJECTS@
 
 SHELL = /bin/sh
 
-DOT_O_DEPS = sllimits.h _slang.h $(CONFIG_H)
+DOT_O_DEPS = sllimits.h _slang.h $(CONFIG_H) $(OBJDIR) $(ELFDIR)
 sltoken_O_DEP = keywhash.c
 slarith_O_DEP = slarith.inc slarith2.inc
 slarrfun_O_DEP = slarrfun.inc
