From: Ron Lee <ron@debian.org>
Date: Fri, 11 Jan 2013 12:13:50 +0100
Subject: [PATCH] speex_header

from: http://patch-tracker.debian.org/package/speex/1.2~rc1-7

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 libspeex/speex_header.c |   17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/libspeex/speex_header.c b/libspeex/speex_header.c
index b743059..92cc0d4 100644
--- a/libspeex/speex_header.c
+++ b/libspeex/speex_header.c
@@ -147,20 +147,21 @@ EXPORT SpeexHeader *speex_packet_to_header(char *packet, int size)
    int i;
    SpeexHeader *le_header;
    const char *h = "Speex   ";
-   for (i=0;i<8;i++)
-      if (packet[i]!=h[i])
-      {
-         speex_notify("This doesn't look like a Speex file");
-         return NULL;
-      }
-   
+
    /*FIXME: Do we allow larger headers?*/
    if (size < (int)sizeof(SpeexHeader))
    {
       speex_notify("Speex header too small");
       return NULL;
    }
-   
+
+   for (i=0;i<8;i++)
+      if (packet[i]!=h[i])
+      {
+         speex_notify("This doesn't look like a Speex file");
+         return NULL;
+      }
+
    le_header = (SpeexHeader*)speex_alloc(sizeof(SpeexHeader));
    
    SPEEX_COPY(le_header, (SpeexHeader*)packet, 1);
