---
 Makefile |    2 +-
 fax      |   14 +++++++-------
 2 files changed, 8 insertions(+), 8 deletions(-)

--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ CC=gcc
 # Compile/load options. Add -DNO_STRERROR to CFLAGS if _strerror
 # is undefined
 
-CFLAGS=
+CFLAGS= $(RPM_OPT_FLAGS)
 LDFLAGS=
 
 # Change the following to the destination directories for
--- a/fax
+++ b/fax
@@ -17,9 +17,9 @@
 # The names of the fax script, efax and efix, including full path
 # if necessary.
 
-FAX=fax
-EFAX=efax
-EFIX=efix
+FAX=/usr/bin/fax
+EFAX=/usr/bin/efax
+EFIX=/usr/bin/efix
 
 # The device to which the fax modem is connected (e.g. ttya for
 # /dev/ttya).  Use a dial-out (cua) device if available.  If
@@ -28,16 +28,16 @@ EFIX=efix
 # /dev/modem is a link to /dev/cua1, then getty, uucp, kermit,
 # pppd, dip, etc. must *all* use either /dev/modem or /dev/cua1.
 
-DEV=cua1
+DEV=modem
 
 # Your fax number in international format, 20 characters maximum.
 # Use only digits, spaces, and the "+" character.
 
-FROM="+1 800 555 5555"
+FROM="+0 000 000 0000"
 
 # Your name as it should appear on the page header.
 
-NAME="Put Your Name Here"
+NAME="from a Red Hat Linux system"
 
 # The preferred page size for creating and printing faxes.
 # Allowed values are "letter", "legal", and "a4".
@@ -67,7 +67,7 @@ VIEWCMD="xloadimage stdin"		# best
 # The name of the Ghostscript executable including full path if
 # necessary.  Only required if faxing Postscript files.
 
-GS=gs
+GS=/usr/bin/gs
 
 # Dial string prefix and suffix such as T for tone dialing, P for
 # pulse dialing, 9 to get an external line, commas for delays or
