---
 efax.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/efax.c
+++ b/efax.c
@@ -759,7 +759,7 @@ int readfaxruns ( TFILE *f, DECODER *d, 
 	c = tgetd ( f, TO_CHAR ) ;
 
 	rd_state = ( rd_state & rd_allowed[c] ) ?
-	  ( ( rd_state & rd_nexts[c] ) ? rd_state <<= 1 : rd_state ) : 
+	  ( ( rd_state & rd_nexts[c] ) ? rd_state << 1 : rd_state ) : 
 	  RD_BEGIN ;
 
 	if ( rd_state == RD_END )
