From: Marc Kleine-Budde <mkl@pengutronix.de>
Subject: disable D_DEBUG_AT, undef symbol "wimpl" with enabled dicrectfb debugging

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>

---
 gdk/directfb/gdkwindow-directfb.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: gtk+-2.14.7/gdk/directfb/gdkwindow-directfb.c
===================================================================
--- gtk+-2.14.7.orig/gdk/directfb/gdkwindow-directfb.c
+++ gtk+-2.14.7/gdk/directfb/gdkwindow-directfb.c
@@ -106,16 +106,20 @@ gdk_window_directfb_process_all_updates 
 
       if (private->update_freeze_count)
         {
+/*
           D_DEBUG_AT( GDKDFB_Updates, "  -> %p frozen [%4d,%4d-%4dx%4d] (%d boxes)\n",
                       private, DFB_RECTANGLE_VALS_FROM_REGION( &wimpl->flips.bounding ),
                       wimpl->flips.num_regions );
+*/
           update_windows = g_slist_prepend (update_windows, private);
         }
       else
         {
+/*
           D_DEBUG_AT( GDKDFB_Updates, "  -> %p update [%4d,%4d-%4dx%4d] (%d boxes)\n",
                       private, DFB_RECTANGLE_VALS_FROM_REGION( &wimpl->flips.bounding ),
                       wimpl->flips.num_regions );
+*/
           gdk_window_process_updates(tmp_list->data,TRUE);
         }
 
