From: unknown author <unknown.author@example.com>
Date: Fri, 24 Jan 2014 16:45:55 +0100
Subject: [PATCH] mktime-718332

===================================================================
---
 logrotate.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/logrotate.c b/logrotate.c
index 755f395..526750b 100644
--- a/logrotate.c
+++ b/logrotate.c
@@ -2115,12 +2115,14 @@ static int readState(char *stateFilename)
 		return 1;
 	}
 
+	memset(&st->lastRotated, 0, sizeof(st->lastRotated));
 	st->lastRotated.tm_year = year;
 	st->lastRotated.tm_mon = month;
 	st->lastRotated.tm_mday = day;
 	st->lastRotated.tm_hour = hour;
 	st->lastRotated.tm_min = minute;
 	st->lastRotated.tm_sec = second;
+	st->lastRotated.tm_isdst = -1;
 
 	/* fill in the rest of the st->lastRotated fields */
 	lr_time = mktime(&st->lastRotated);
