Index: xorg-server-1.4/hw/xfree86/xaa/Makefile.am
===================================================================
--- xorg-server-1.4.orig/hw/xfree86/xaa/Makefile.am	2007-10-17 21:16:59.000000000 +0200
+++ xorg-server-1.4/hw/xfree86/xaa/Makefile.am	2007-10-17 21:17:24.000000000 +0200
@@ -9,7 +9,6 @@
 POLYSEG = s-xaaLine.c s-xaaDashLine.c
 
 libxaa_la_LDFLAGS = -avoid-version
-libxaa_la_LIBADD = $(top_builddir)/miext/cw/libcw.la
 
 module_LTLIBRARIES = libxaa.la
 libxaa_la_SOURCES = xaaInit.c xaaGC.c xaaInitAccel.c xaaFallback.c \
